<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Differential abundance testing of small RNAs</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: Differential abundance testing of small RNAs" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/srna/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Differential abundance testing of small RNAs",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "transcriptomics / srna / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Differential abundance testing of small RNAs' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "Differential abundance testing of small RNAs",
    "description": "Differential abundance testing of small RNAs",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/transcriptomics/tutorials/srna/tutorial.html",
  "timeRequired": "PT3H",
  "description": "Differential abundance testing of small RNAs\\nThe questions this  addresses are:\n - What small RNAs are expressed?\n - What RNA features have significantly different numbers of small RNAs targeting them between two conditions?\n\n\\nThe objectives are:\n - Process small RNA-seq datasets to determine quality and reproducibility.\n - Filter out contaminants (e.g. rRNA reads) in small RNA-seq datasets.\n - Differentiate between subclasses of small RNAs based on their characteristics.\n - Identify differently abundant small RNAs and their targets.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/826906",
      "name": "Training data for Differential abundance testing of small RNAs tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Mallory Freeberg"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Mallory Freeberg"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://galaxyproject.github.io//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Differential abundance testing of small RNAs</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#malloryfreeberg" class="contributor-badge"><img src="https://avatars.githubusercontent.com/malloryfreeberg" alt="Mallory Freeberg">Mallory Freeberg</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>What small RNAs are expressed?</li>
            
            <li>What RNA features have significantly different numbers of small RNAs targeting them between two conditions?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Process small RNA-seq datasets to determine quality and reproducibility.</li>
            
            <li>Filter out contaminants (e.g. rRNA reads) in small RNA-seq datasets.</li>
            
            <li>Differentiate between subclasses of small RNAs based on their characteristics.</li>
            
            <li>Identify differently abundant small RNAs and their targets.</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 3 hours</p>
            

            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    
                    <div>
                        <a class="topic-icon" href="/training-material/topics/transcriptomics/slides/introduction.html" title="Introduction slides">
                            <i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Introduction Slides
                        </a>
                    </div>
                    

                    
                    <div>
                        
<a class="topic-icon" href="https://zenodo.org/record/826906">
    <i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>


                    </div>
                    

                    
                    <div>
                        
    <a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/srna/workflows/" title="Workflows" alt="Differential abundance testing of small RNAs workflows">
        <i class="fa fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>


                    </div>
                    
                    
                    <div>
                        
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/srna/tours/">
    <i class="fa fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>


                    </div>
                    


                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Small, noncoding RNA (sRNA) molecules, typically 18-40nt in length, are key features of post-transcriptional regulatory mechanisms governing gene expression. Through interactions with protein cofactors, these tiny sRNAs typically function by perfectly or imperfectly basepairing with substrate RNA molecules, and then eliciting downstream effects such as translation inhibition or RNA degradation. Different subclasses of sRNAs - <em>e.g.</em> microRNAs (miRNAs), Piwi-interaction RNAs (piRNAs), and endogenous short interferring RNAs (siRNAs) - exhibit unique characteristics, and their relative abundances in biological contexts can indicate whether they are active or not. In this tutorial, we will examine expression of the piRNA subclass of sRNAs and their targets in <em>Drosophila melanogaster</em>.</p>

<p>The data used in this tutorial are from polyphosphatase-treated sRNA sequencing (sRNA-seq) experiments in <em>Drosophila</em>. The goal of this study was to determine how siRNA expression changes in flies treated with RNA interference (RNAi) to knock down Symplekin, which is a component of the core cleaveage completx. To that end, in addition to sRNA-seq, mRNA-seq experiments were performed to determine whether targets of differentially expressed siRNAs were also differentially expressed. The original published study can be found <a href="https://www.ncbi.nlm.nih.gov/pubmed/28415970">here</a>. Because of the long processing time for the large original files - which contained 7-22 million reads each - we have downsampled the original input data to include only a subset of usable reads.</p>

<h1 id="analysis-strategy">Analysis strategy</h1>

<p>In this exercise we will identify what small RNAs are present in <em>Drosophila</em> Dmel-2 tissue culture cells treated with either RNAi against  core cleavage complex component Symplekin or blank (control) RNAi (published data available in GEO at <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE82128">GSE82128</a>). In this study, biological triplicate sRNA- and mRNA-seq libraries were sequenced for both RNAi conditions. After removing contaminant ribosomal RNA (rRNA) and miRNA reads, we will identify and quantify endogenous siRNAs from sequenced reads and test for differential abundance. We will follow a popular small RNA analysis pipeline developed for piRNAs by the Phillip Zamore Lab and the ZLab (Zhiping Weng) at UMass Med School called <a href="https://github.com/bowhan/piPipes">PiPipes</a>. Although PiPipes was developed for analysis of piRNAs, many of the basical principles can be applied to other classes of small RNAs.</p>

<p>It is of note that this tutorial uses datasets that have been de-multiplexed so that the input data files are a single FASTQ formatted file for each sample. Because sRNAs are typically much smaller than fragments generated for RNA-seq or other types of deep sequencing experiments, single-end sequencing strategies are almost always used to sequence sRNAs. This tutorial uses the <em>Collections</em> feature of Galaxy to orgainze each set of replicates into a single group, making tool form submission easier and ensuring reproducibility.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#analysis-strategy" id="markdown-toc-analysis-strategy">Analysis strategy</a>    <ol>
      <li><a href="#data-upload-and-organization" id="markdown-toc-data-upload-and-organization">Data upload and organization</a></li>
      <li><a href="#read-quality-checking" id="markdown-toc-read-quality-checking">Read quality checking</a></li>
      <li><a href="#adaptor-trimming" id="markdown-toc-adaptor-trimming">Adaptor trimming</a></li>
      <li><a href="#hierarchical-read-alignment-to-remove-rrnamirna-reads" id="markdown-toc-hierarchical-read-alignment-to-remove-rrnamirna-reads">Hierarchical read alignment to remove rRNA/miRNA reads</a></li>
      <li><a href="#small-rna-subclass-distinction" id="markdown-toc-small-rna-subclass-distinction">Small RNA subclass distinction</a></li>
      <li><a href="#sirna-abundance-estimation" id="markdown-toc-sirna-abundance-estimation">siRNA abundance estimation</a></li>
      <li><a href="#sirna-differential-abundance-testing" id="markdown-toc-sirna-differential-abundance-testing">siRNA differential abundance testing</a></li>
      <li><a href="#visualization" id="markdown-toc-visualization">Visualization</a></li>
      <li><a href="#small-rna-analysis-general-pipeline" id="markdown-toc-small-rna-analysis-general-pipeline">Small RNA analysis (general) pipeline</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h2 id="data-upload-and-organization">Data upload and organization</h2>

<p>Due to the large size of the original sRNA-seq datasets, we have downsampled them to only inlcude a subset of reads. These datasets are avaialble at <a href="https://doi.org/10.5281/zenodo.826906"><code class="highlighter-rouge">Zenodo</code></a> where you can find the FASTQ files corresponding to replicate sRNA-seq experiments and additional annotation files for the <em>Drosophila melanogaster</em> genome version dm3.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-and-organization"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload and organization</h3>

  <ol>
    <li>
      <p>Create a new history and name it something meaningful (<em>e.g.</em> sRNA-seq tutorial)</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import the 3 <code class="highlighter-rouge">Blank_RNAi_sRNA-seq</code> and 3 <code class="highlighter-rouge">Symp_RNAi_sRNA-seq</code> FASTQ files from <a href="https://zenodo.org/record/1324070">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/826906/files/Blank_RNAi_sRNA-seq_rep1_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Blank_RNAi_sRNA-seq_rep2_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Blank_RNAi_sRNA-seq_rep3_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Symp_RNAi_sRNA-seq_rep1_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Symp_RNAi_sRNA-seq_rep2_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Symp_RNAi_sRNA-seq_rep3_downsampled.fastqsanger.gz
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the Galaxy Upload Manager (<i class="fa fa fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, Galaxy uses the URL as the name, so rename the files with a more useful name.</p>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>

      <p>Set the datatype of the read (.fastqsanger) files to <strong>fastq</strong>
As default, Galaxy takes the link as name, so rename them.</p>
    </li>
    <li>
      <p>Build a dataset list for both sets (<code class="highlighter-rouge">Blank</code> and <code class="highlighter-rouge">Symp</code>) of replicate FASTQ files</p>

      <blockquote class="tip">

        <h3 id="tip-tip-build-a-dataset-list"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Build a dataset list</h3>
        <ul>
          <li>Click on <i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Operations on multiple datasets</strong> at the top of the history panel</li>
          <li>Check the boxes next the datasets to select</li>
          <li>In <strong>For all selected…</strong>, choose <strong>Build dataset list</strong></li>
          <li>Ensure that only the wanted samples are selected</li>
          <li>Enter a name for the new collection</li>
          <li>Click on <strong>Create list</strong></li>
        </ul>
      </blockquote>

      <p><img src="../../images/sRNA/Fig3_dataset_lists_made.png" alt="History section with all the loaded datasets" /></p>
    </li>
    <li>
      <p>Import the remaining files from <a href="https://zenodo.org/record/1324070">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/826906/files/dm3_transcriptome_Tx2Gene_downsampled.tab.gz
https://zenodo.org/record/826906/files/dm3_transcriptome_sequences_downsampled.fa.gz
https://zenodo.org/record/826906/files/dm3_miRNA_hairpin_sequences.fa.gz
https://zenodo.org/record/826906/files/dm3_rRNA_sequences.fa.gz
</code></pre></div>      </div>

      <ul>
        <li>Set the datatype of the annotation file to <strong>tab</strong> and assign the Genome as <strong>dm3</strong></li>
        <li>Set the datatype of the reference files to <strong>fasta</strong> and assign the Genome as <strong>dm3</strong></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="read-quality-checking">Read quality checking</h2>

<p>Read quality scores (phred scores) in FASTQ-formatted data can be encoded by one of a few different encoding schemes. Most Galaxy tools assume that input FASTQ files are using the Sanger/Illumina 1.9 encoding scheme. If the input FASTQ files are using an alternate encoding scheme, then some tools will not interpret the quality score encodings correctly. It is good practice to confirm the quality encoding scheme of your data and then convert to Sanger/Illumina 1.9, if necessary. We can check the quality encoding scheme using the <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> tool (further described in the <a href="/training-material/topics/sequence-analysis">NGS-QC tutorial</a>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-checking"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality checking</h3>

  <ol>
    <li>Run <strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the <code class="highlighter-rouge">Blank</code> collection
      <ul>
        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Short read data from your current history”</em>: <code class="highlighter-rouge">Blank</code> collection</li>
      </ul>

      <blockquote class="tip">

        <h3 id="tip-tip-selecting-a-dataset-collection-as-input"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Selecting a dataset collection as input</h3>

        <ol>
          <li>Click on <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in
front of the input parameter you want to supply the collection to.</li>
          <li>Select the collection you want to use from the list</li>
        </ol>
      </blockquote>
    </li>
    <li>Run <strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for the <em>Symplekin</em> RNAi dataset collection</li>
    <li>
      <p>Inspect the <strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> reports</p>

      <blockquote class="question">
        <h3 id="question-questions"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What quality score encoding scheme is being used for each sample?</li>
          <li>What is the read length for each sample?</li>
          <li>
            <p>What does the base/read quality look like for each sample?</p>

            <p><img src="../../images/sRNA/Fig5a_fastqc_result_stats_boxplot.png" alt="The output of FastQC as a boxplot" /></p>
          </li>
          <li>
            <p>Are there any adaptors present in these reads? Which one(s)?</p>

            <p><img src="../../images/sRNA/Fig5b_fastqc_result_stats_adapter.png" alt="The output of FastQC stats adapted" /></p>
          </li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>All samples use the Illumina 1.9 quality encoding scheme, so we do <strong>not</strong> need to convert.</li>
            <li>All samples have a read length of 50 nt.</li>
            <li>The base quality across the entire length of the reads is good (phred score &gt; 28 for the most part).</li>
            <li>Yes, the “Illumina Universal Adapter” is present.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-formatting"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Formatting</h3>

  <p>If your data are not in <strong>Sanger / Illumina 1.9 format</strong>:</p>
  <ol>
    <li><strong>FASTQ Groomer</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection
      <ul>
        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“File to groom”</em>: input collection</li>
        <li><em>“Input FASTQ quality scores type”</em>: <code class="highlighter-rouge">Illumina 1.3-1.7</code></li>
      </ul>
    </li>
    <li>Check that the format is <strong>fastqsanger</strong> instead of <strong>fastq</strong></li>
  </ol>

  <p>If your data are already in <strong>Sanger / Illumina 1.9 format</strong>:</p>
  <ol>
    <li>
      <p>Change the datatype to <code class="highlighter-rouge">fastqsanger</code></p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fa fa fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="highlighter-rouge">fastqsanger</code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>Go back to the <strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and scroll down to the “Adapter Content” section. You can see that Illumina Universal Adapters are present in <strong>&gt;80%</strong> of the reads. The next step is to remove these artificial adaptors because they are not part of the biological sRNAs. If a different adapter is present, you can update the <strong>Adapter sequence to be trimmed off</strong> in the <strong>Trim Galore!</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> step described below.</p>

<h2 id="adaptor-trimming">Adaptor trimming</h2>

<p>sRNA-seq library preparation involves adding an artificial adaptor sequence to both the 5’ and 3’ ends of the small RNAs. While the 5’ adaptor anchors reads to the sequencing surface and thus are not sequenced, the 3’ adaptor is typically sequenced immediately following the sRNA sequence. In the example datasets here, the 3’ adaptor sequence is identified as the Illumina Universal Adapter, and needs to be removed from each read before aligning to a reference. We will be using the Galaxy tool <strong>Trim Galore!</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> which implements the <a href="https://cutadapt.readthedocs.io/en/stable/"><code class="highlighter-rouge">cutadapt</code></a> tool for adapter trimming.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-adaptor-trimming"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adaptor trimming</h3>

  <ol>
    <li><strong>Trim Galore!</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of FASTQ read files to remove Illumina adapters from the 3’ ends of reads with the following parameters:
      <ul>
        <li><em>“Is this library paired- or single-end”</em>: <code class="highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Reads in FASTQ format?”</em>: <code class="highlighter-rouge">Blank</code> collection</li>
            <li><em>“Adapter sequence to be trimmed”</em>: <code class="highlighter-rouge">Illumina universal</code></li>
          </ul>
        </li>
        <li><em>“Trim Galore! advanced settings”</em>: <code class="highlighter-rouge">Full parameter list</code>
          <ul>
            <li><em>“Trim low-quality ends from reads in addition to adapter removal”</em>: <code class="highlighter-rouge">0</code></li>
            <li><em>“Overlap with adapter sequence required to trim a sequence”</em>: <code class="highlighter-rouge">6</code></li>
            <li><em>“Discard reads that became shorter than length N”</em>: <code class="highlighter-rouge">12</code></li>
            <li><em>“Generate a report file”</em>: <code class="highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Trim Galore!</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> RNAi dataset collection</li>
  </ol>

</blockquote>

<p>We don’t want to trim for quality because the adapter-trimmed sequences represent a full small RNA molecule, and we want to maintain the integrity of the entire molecule. We increase the minimum read length required to keep a read because small RNAs can potentially be shorter than 20 nt (the default value). We can check out the generated report file for any sample and see the command for the tool, a summary of the total reads processed and number of reads with an adapter identified, and histogram data of the length of adaptor trimmed. We also see that a very small percentage of low-quality bases have been trimmed</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-adaptor-trimming-check"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adaptor trimming check</h3>

  <ol>
    <li><strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of trimmed FASTQ read files to assess whether adapters were successfully removed.</li>
    <li>
      <p>Inspect the <strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> reports</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is the read length?</li>
        </ol>

        <p><img src="../../images/sRNA/Fig8a_fastqc_post_trimming_result.png" alt="FastQC output after trimming" /></p>

        <ol>
          <li>Are there any adaptors present in these reads? Which one(s)?</li>
        </ol>

        <p><img src="../../images/sRNA/Fig8b_fastqc_post_trimming_lengths.png" alt="FastQC output after trimming" /></p>
        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>The read lengths range from 12 to 50 nt after trimming.</li>
            <li>No, Illumina Universal Adaptors are no longer present. No other adapters are present.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>An interesting thing to note from our <strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> results is the <em>Sequence Length Distribution</em> results. While many RNA-seq experiments have normal distribution of read lengths, an unusual spike at 22nt is observed in our data. This spike represents the large set of endogenous siRNAs that occur in the cell line used in this study, and is actually confirmation that our dataset captures the biological molecule we are interested in.</p>

<p>Now that we have converted to <em>fastqsanger</em> format and trimmed our reads of the Illumina Universal Adaptors, we will align our trimmed reads to reference <em>Drosophila</em> rRNA and miRNA sequences (dm3) to remove these artifacts. Interestingly, <em>Drosophila</em> have a short 2S rRNA sequence that is 30nt long and typically co-migrates with sRNA populations during gel electrophoresis. rRNAs make up a very large proportion of all non-coding RNAs, and thus need to be removed. Oftentimes, experimental approaches can be utilized to deplete or avoid capture of rRNAs, but these methods are not always 100% efficient. We also want to remove any miRNA sequences, as these are not relevant to our analysis. After removing rRNA and miRNA reads, we will analyze the remaining reads, which may be siRNA or piRNA sequences.</p>

<h2 id="hierarchical-read-alignment-to-remove-rrnamirna-reads">Hierarchical read alignment to remove rRNA/miRNA reads</h2>

<p>To first identify rRNA-originating reads (which we are not interested in in this case), we will align the reads to a reference set of rRNA sequences using <a href="https://ccb.jhu.edu/software/hisat2/index.shtml"><strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></a>, an accurate and fast tool for aligning reads to a reference.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-hierarchical-alignment-to-rrna-and-mirna-reference-sequences"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Hierarchical alignment to rRNA and miRNA reference sequences</h3>

  <ol>
    <li><strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of trimmed reads to align to reference rRNA sequences with the following parameters
      <ul>
        <li><em>“Source for the reference genome to align against”</em>: <code class="highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select the reference genome”</em>: <code class="highlighter-rouge">Dmel_rRNA_sequences.fa</code></li>
          </ul>
        </li>
        <li><em>“Is this a single or paired library?”</em>: <code class="highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTA/Q file”</em>: output of <strong>Trim Galore!</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="highlighter-rouge">Blank</code> collection</li>
            <li><em>“Specify strand-specific information”</em>: <code class="highlighter-rouge">Forward (F)</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> RNAi dataset collection</li>
  </ol>
</blockquote>

<p>We now need to extract the <em>unaligned</em> reads from the output BAM file for aligning to reference miRNA sequences. We can do this by using the <code class="highlighter-rouge">Filter SAM or BAM, output SAM or BAM</code> tool to obtain reads with a bit flag = 4 (meaning the read is unaligned) and then converting the filtered BAM file to FASTQ format with the <code class="highlighter-rouge">Convert from BAM to FastQ</code> tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-hierarchical-alignment-to-rrna-and-mirna-reference-sequences-2"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Hierarchical alignment to rRNA and miRNA reference sequences (2)</h3>
  <ol>
    <li><strong>Filter SAM or BAM, output SAM or BAM</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on collection of HISAT2 output BAM files with the following parameters
      <ul>
        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“SAM or BAM file to filter”</em>: output of <strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="highlighter-rouge">Blank</code> collection</li>
        <li><em>“Filter on bitwise flag”</em>: <code class="highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Only output alignments with all of these flag bits set”</em>: <code class="highlighter-rouge">The read in unmapped</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Filter SAM or BAM, output SAM or BAM</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> RNAi dataset collection</li>
    <li><strong>Convert from BAM to FastQ</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of filtered HISAT2 output BAM files with the following parameters:
      <ul>
        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Convert the following BAM file to FASTQ”</em>: output of <strong>Filter SAM or BAM</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="highlighter-rouge">Blank</code> collection</li>
      </ul>
    </li>
    <li><strong>Convert from BAM to FastQ</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> RNAi dataset collection</li>
  </ol>
</blockquote>

<p>Next we will align the non-rRNA reads to a known set of miRNA hairpin sequences to identify miRNA reads.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-hierarchical-alignment-to-rrna-and-mirna-reference-sequences-3"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Hierarchical alignment to rRNA and miRNA reference sequences (3)</h3>
  <ol>
    <li><strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of filtered HISAT2 output FASTQ files to align non-rRNA reads to reference miRNA hairpin sequences using the following parameters:
      <ul>
        <li><em>“Source for the reference genome to align against”</em>: <code class="highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select the reference genome”</em>: <code class="highlighter-rouge">Dmel_miRNA_sequences.fa</code></li>
          </ul>
        </li>
        <li><em>“Is this a single or paired library?”</em>: <code class="highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTA/Q file”</em>: output of <strong>Convert from BAM to FastQ</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="highlighter-rouge">Blank</code> collection</li>
            <li><em>“Specify strand-specific information”</em>: <code class="highlighter-rouge">Forward (F)</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> RNAi dataset collection</li>
  </ol>
</blockquote>

<p>For this tutorial we are not interested in miRNA reads, so we need to extract <em>unaligned</em> reads from the output BAM files. To do this, repeat the <strong>Filter SAM or BAM, output SAM or BAM</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> steps for each dataset collection. Finally, rename the converted FASTQ files something meaningful (<em>e.g.</em> “non-r/miRNA control RNAi sRNA-seq”).</p>

<p>In some instances, miRNA-aligned reads are the desired output for downstream analyses, for example, if we are investigating how loss of key miRNA pathway components affect levels of pre-miRNA and mature miRNAs. In this case, the BAM output of HISAT2 can directly be used in subsequent tools as it contains the subset of reads aligned to miRNA sequences.</p>

<h2 id="small-rna-subclass-distinction">Small RNA subclass distinction</h2>

<p>In <em>Drosophila</em>, non-miRNA small RNAs are typically divided into two major groups: endogenous siRNAs which are 20-22nt long and piRNAs which are 23-29nt long. We want to analyze these sRNA subclasses independently, so next we are going to filter the non-r/miRNA reads based on length using the <strong>Manipulate FASTQ</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-subclasses"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract subclasses</h3>

  <ol>
    <li><strong>Manipulate FASTQ</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of non-r/miRNA reads to identify siRNAs (20-22nt) using the following parameters.
      <ul>
        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ File”</em>: output of last <strong>HISAT</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="highlighter-rouge">Blank</code> collection</li>
        <li>Click on <em>“Insert Match Reads”</em>
          <ul>
            <li>In “1: Match Reads”
              <ul>
                <li><em>“Match Reads by”</em>: <code class="highlighter-rouge">Sequence Content</code>
                  <ul>
                    <li><em>“Match by”</em>: <code class="highlighter-rouge">^.{12,19}$|^.{23,50}$</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Click on <em>“Manipulate Reads”</em>
          <ul>
            <li>In “1: Manipulate Reads”
              <ul>
                <li><em>“Manipulate Reads on”</em>: <code class="highlighter-rouge">Miscellaneous actions</code>
                  <ul>
                    <li><em>“Miscellaneous Manipulation Type”</em>: <code class="highlighter-rouge">Remove Read</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Manipulate FASTQ</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> RNAi dataset collection</li>
    <li>Rename each resulting dataset collection something meaningful (<em>i.e.</em> “blank RNAi - siRNA reads (20-22nt)”)</li>
  </ol>
</blockquote>

<p>The regular expression in the <strong>Match by</strong> parameter tells the tool to identify sequences that are length 12-19 or 23-50 (inclusive), and the <strong>Miscellaneous Manipulation Type</strong> parameter tells the tool to remove these sequences. What remains are sequences of length 20-22nt. We will now repeat these steps to identify sequences in the size-range of piRNAs (23-29nt).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-subclasses-1"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract subclasses</h3>
  <ol>
    <li><strong>Manipulate FASTQ</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="highlighter-rouge">Manipulate FASTQ</code> on each collection of non-r/miRNA reads to identify sequences in the size-range of piRNAs (23-29nt)
      <ul>
        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ File”</em>: output of last <strong>HISAT</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="highlighter-rouge">Blank</code> collection</li>
        <li>Click on <em>“Insert Match Reads”</em>
          <ul>
            <li>In “1: Match Reads”
              <ul>
                <li><em>“Match Reads by”</em>: <code class="highlighter-rouge">Sequence Content</code>
                  <ul>
                    <li><em>“Match by”</em>: <code class="highlighter-rouge">^.{12,22}$|^.{30,50}$</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Click on <em>“Manipulate Reads”</em>
          <ul>
            <li>In “1: Manipulate Reads”
              <ul>
                <li><em>“Manipulate Reads on”</em>: <code class="highlighter-rouge">Miscellaneous actions</code>
                  <ul>
                    <li><em>“Miscellaneous Manipulation Type”</em>: <code class="highlighter-rouge">Remove Read</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Manipulate FASTQ</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> RNAi dataset collection</li>
    <li>Rename each resulting dataset collection something meaningful (<em>i.e.</em> “blank RNAi - piRNA reads (23-29nt)”)</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-check-the-read-length"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check the read length</h3>
  <ol>
    <li><strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="highlighter-rouge">FastQC</code> on each collection of siRNA and piRNA read files to confirm the correct read lengths.</li>
  </ol>
</blockquote>

<p><img src="../../images/sRNA/Fig13_FastQC_piRNA_siRNA_result_Blank_rep1.png" alt="FastQC result on both collections" /></p>

<p>We see in the above image (for Blank RNAi, replicate 3) that we have ~32k reads that are 20-22nt long in our Reads_20-22_nt_length_siRNAs file and ~16k reads that are 23-29nt long in our Reads_23-29_nt_length_piRNAs file, as expected. Given that we had ~268k trimmed reads in the Blank RNAi replicate 3 file, siRNAs represent ~12.1% of the reads and piRNAs represent ~6.2% of the reads in this experiment. There is no distinct peak in the piRNA range (you can find the example of the real piRNA profile in Figure 1 from <a href="http://dx.doi.org/10.1016/j.cell.2009.03.040">Colin D. Malone et al, 2009</a>.</p>

<p>The next step in our analysis pipeline is to identify which RNA features - e.g. protein-coding mRNAs, transposable elements - the siRNAs and piRNAs align to. Most fly piRNAs originate from and thus align to transposable elements (TEs) but some also originate from genome piRNA clusters or protein-coding genes. siRNAs are thought to inhibit TE mobility and potentially target mRNAs for degradation via an RNA-interference (RNAi) mechanism. Ultimately, we want to know whether an RNA feature (TE, piRNA cluster, mRNA, etc.) has significantly different numbers of siRNAs or piRNAs targeting it. To determine this, we will use <strong>Salmon</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to simultaneously align and quantify siRNA reads against known target sequences to estimate siRNA abundance per target. For the remainder of the tutorial we will be focusing on siRNAs, but similar approaches can be done for piRNAs, miRNAs, or any other subclass of small, non-coding RNAs.</p>

<h2 id="sirna-abundance-estimation">siRNA abundance estimation</h2>

<p>We want to identify which siRNAs are differentially abundance between the blank and <em>Symplekin</em> RNAi conditions. To do this we will implement a counting approach using <strong>Salmon</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to quantify siRNAs per RNA feature. Specifically, we will analyzing mRNA and TE elements by counting siRNAs against a FASTA file of transcript sequences, not the reference genome. This approach is especially useful in this case because we are interested in TEs, which occur at many copies (100s - 1000s) in the genome, and thus will result in multiple alignments if we align to the genome with a tool like <strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. Further, we will be counting abundance of siRNAs that align <em>sense</em> or <em>antisense</em> to an RNA feature independently, as opposite sense alignments are correlated with unique downstream silencing effects. Then, we will provide this information to <strong>DESeq2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to generate normalized counts and significance testing for differential abundance of siRNAs per feature.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-sirna-abundance-estimation"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: siRNA abundance estimation</h3>

  <ol>
    <li><strong>Salmon</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of siRNA reads (20-22nt) to quantify the abundance of <em>antisense</em> siRNAs at relevant targets
      <ul>
        <li><em>“Select a reference transcriptome from your history or use a built-in index?”</em>: <code class="highlighter-rouge">Use one from the history</code>
          <ul>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select the reference transcriptome”</em>: <code class="highlighter-rouge">the reference mRNA and TE fasta file</code></li>
            <li><em>“The size should be odd number”</em>: <code class="highlighter-rouge">19</code></li>
          </ul>
        </li>
        <li><em>“Is this library mate-paired?”</em>: <code class="highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ/FASTA file”</em>: collection with the Blank RNAi siRNA (20-22nt) reads</li>
            <li><em>“Specify the strandedness of the reads”</em>: <code class="highlighter-rouge">read 1 (or single-end read) comes from the reverse strand (SR)</code></li>
          </ul>
        </li>
        <li>In <em>“Additional Options”</em>
          <ul>
            <li><em>“Incompatible Prior”</em>: <code class="highlighter-rouge">0</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Salmon</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> RNAi siRNAs (20-22nt) reads dataset collection</li>
    <li><strong>Salmon</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat on each collection of siRNA reads (20-22nt) to quantify the abundance of <em>sense</em> siRNAs at relevant targets by changing the following parameters:
      <ul>
        <li><em>“Specify the strandedness of the reads”</em>: <code class="highlighter-rouge">read 1 (or single-end read) comes from the forward strand (SF)</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The output of <strong>Salmon</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> includes a table of RNA features, estimated counts, transcripts per million, and feature length. We will be using the TPM quantification column as input to <strong>DESeq2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in the next section.</p>

<h2 id="sirna-differential-abundance-testing">siRNA differential abundance testing</h2>

<p><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html"><strong>DESeq2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></a> is a great tool for differential expression analysis, but we also employ it here for estimation of abundance of reads targeting each of our RNA features. As input, <strong>DESeq2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> can take <em>t</em>ranscripts <em>p</em>er <em>m</em>illion (TPM) counts produced by <strong>Salmon</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for each feature. TPMs are estimates of the relative abundance of a given transcript in units, and analogously can be used here to quantify siRNAs that align either sense or antisense to transcripts.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-sirna-differential-abundance-testing"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: siRNA differential abundance testing</h3>

  <ol>
    <li><strong>DESeq2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to test for differential abundance of <em>antisense</em> siRNAs at mRNA and TE features using the following parameters:
      <ul>
        <li>In <em>“Factor”</em>:
          <ul>
            <li>In “1: Factor”
              <ul>
                <li><em>“Specify a factor name”</em>: <code class="highlighter-rouge">RNAi</code></li>
                <li>In <em>“Factor level”</em>:
                  <ul>
                    <li>In <em>“1: Factor level”</em>:
                      <ul>
                        <li><em>“Specify a factor level”</em>: <code class="highlighter-rouge">Symplekin</code></li>
                        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Counts file(s)”</em>: the Symplekin RNAi siRNA counts from reverse strand (SR) collection</li>
                      </ul>
                    </li>
                    <li>In <em>“2: Factor level”</em>:
                      <ul>
                        <li><em>“Specify a factor level”</em>: <code class="highlighter-rouge">RNAi</code></li>
                        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Counts file(s)”</em>: the Symplekin RNAi siRNA counts from reverse strand (SR)</li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Choice of Input data”</em>: <code class="highlighter-rouge">TPM values (e.g. from sailfish and salmon)</code>
          <ul>
            <li><em>“Program used to generate TPMs”</em>: <code class="highlighter-rouge">Salmon</code></li>
            <li><em>“Gene mapping format”</em>: <code class="highlighter-rouge">Transcript-ID and Gene-ID mapping file</code>
              <ul>
                <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular file with Transcript - Gene mapping”</em>: <code class="highlighter-rouge">reference tx2g (transcript to gene) file</code> (available through zenodo link)</li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Output normalized counts table”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li><strong>DESeq2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for the <em>sense</em> siRNAs at mRNA and TE features:
      <ul>
        <li>In <em>“Factor”</em>:
          <ul>
            <li>In “1: Factor”
              <ul>
                <li><em>“Specify a factor name”</em>: <code class="highlighter-rouge">RNAi</code></li>
                <li>In <em>“Factor level”</em>:
                  <ul>
                    <li>In <em>“1: Factor level”</em>:
                      <ul>
                        <li><em>“Specify a factor level”</em>: <code class="highlighter-rouge">Symplekin</code></li>
                        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Counts file(s)”</em>: the Symplekin RNAi siRNA counts from forward strand (SF)</li>
                      </ul>
                    </li>
                    <li>In <em>“2: Factor level”</em>:
                      <ul>
                        <li><em>“Specify a factor level”</em>: <code class="highlighter-rouge">RNAi</code></li>
                        <li><i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Counts file(s)”</em>: the Symplekin RNAi siRNA counts from forward strand (SF)</li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Choice of Input data”</em>: <code class="highlighter-rouge">TPM values (e.g. from sailfish and salmon)</code>
          <ul>
            <li><em>“Program used to generate TPMs”</em>: <code class="highlighter-rouge">Salmon</code></li>
            <li><em>“Gene mapping format”</em>: <code class="highlighter-rouge">Transcript-ID and Gene-ID mapping file</code>
              <ul>
                <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular file with Transcript - Gene mapping”</em>: <code class="highlighter-rouge">reference tx2g (transcript to gene) file</code> (available through zenodo link)</li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Output normalized counts table”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li><strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to extract features with a significantly different <em>antisense</em> siRNA abundance (adjusted <em>p</em>-value less than 0.05).
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: Select the <code class="highlighter-rouge">DESeq2</code> result file from testing <em>antisense</em> (SR) siRNA abundances</li>
        <li><em>“With following condition”</em>: <code class="highlighter-rouge">c7&lt;0.05</code></li>
      </ul>
    </li>
    <li><strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for the <em>sense</em> siRNAs at mRNA and TE features:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: Select the <code class="highlighter-rouge">DESeq2</code> result file from testing <em>sense</em> (SF) siRNA abundances</li>
        <li><em>“With following condition”</em>: <code class="highlighter-rouge">c7&lt;0.05</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-question"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>How many features have a significant difference in <em>antisense</em> and <em>sense</em> siRNA abundance in the <em>Symplekin</em> RNAi condition?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>There are 87 <em>antisense</em> and 15 <em>sense</em> features that have significanlty different siRNA abundances.</p>
  </blockquote>
</blockquote>

<p>For more information about <code class="highlighter-rouge">DESeq2</code> and its outputs, have a look at <a href="https://www.bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf"><code class="highlighter-rouge">DESeq2</code> documentation</a>.</p>

<h2 id="visualization">Visualization</h2>

<p>Coming soon!</p>

<h2 class="no_toc" id="conclusion">Conclusion</h2>

<p>Analysis of small RNAs is a complicated and intricate process due to the diversity in characteristics, functionality, and nuances of small RNA subclasses. The goal of this tutorial is to introduce you to a common small RNA workflow that specifically identifies changes in endogenous siRNA abundances that target protein-coding mRNAs and transposable elements. The steps presented here can be rearranged and modified based on small RNA features of specific systems and the needs of the user.</p>

<h2 id="small-rna-analysis-general-pipeline">Small RNA analysis (general) pipeline</h2>

<p><img src="../../images/sRNA/schematic_for_sRNAseq_tutorial.png" alt="sRNA tutorial workflow" /></p>



                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>Analysis of small RNAs is complex due to the diversity of small RNA subclasses.</li>
                            
                            <li>Both alignment to references and transcript quantification approaches are useful for small RNA-seq analyses.</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

        <hr>
        <br>

        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Differential abundance testing of small RNAs (Transcriptomics)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=Differential abundance testing of small RNAs (Transcriptomics)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Mallory Freeberg)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/srna/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
