<!DOCTYPE html>
<html>









  <head>
    <meta charset="utf-8">
    <title>Plates, Batches, and Barcodes</title>
    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-45719423-18' , 'auto');
        ga('send', 'pageview');
    </script>
    
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css" id="theme">
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

    
    
    
    
    <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
    <meta property="og:title" content="Galaxy Training: Plates, Batches, and Barcodes" />
    <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
    <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    <script type="application/ld+json">
      


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Plates, Batches, and Barcodes",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "transcriptomics / scrna-plates-batches-barcodes / slides",
  "learningResourceType": "slides",
  "name": "Slides for 'Plates, Batches, and Barcodes' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "Plates, Batches, and Barcodes",
    "description": "Plates, Batches, and Barcodes",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/transcriptomics/tutorials/scrna-plates-batches-barcodes/slides.html",
  "timeRequired": "PT1H",
  "keywords": "single-cell",
  "description": "Plates, Batches, and Barcodes\\nThe questions this  addresses are:\n - What are barcodes and how are they applied to batches?\n - What is the difference between a plate and a batch?\n - Are batches and lanes the same?\n - Why are plating setups important to know?\n - Is it necessary to check barcodes that don't exist in a batch?\n\n\\nThe objectives are:\n - How to describe a plating setup\n - Proper naming conventions when dealing with multiple batches\n - Checking for false positives across batches\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "",
      "name": "Training data for Plates, Batches, and Barcodes tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Mehmet Tekman"
    },
    {
      "@type": "Person",
      "name": "Alex Ostrovsky"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Mehmet Tekman"
    },
    {
      "@type": "Person",
      "name": "Alex Ostrovsky"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://galaxyproject.github.io//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>
  </head>
  <body>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-header"><span>
<a href="/training-material/topics/transcriptomics" title="Return to topic page" ><i class="fa fa-level-up" aria-hidden="true"></i></a>

<a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/scrna-plates-batches-barcodes/slides.html"><i class="fa fa-pencil" aria-hidden="true"></i></a>

</span></div>

<div class="my-footer"><span>

<img src="/training-material/assets/images/GTN-60px.png" alt="Galaxy Training Network" style="height: 40px;"/>

</span></div>

---

# Plates, Batches, and Barcodes



---

## Requirements

Before diving into this slide deck, we recommend you to have a look at:



    
        
        
            
        
- [Introduction to Galaxy Analyses](/training-material/topics/introduction)
    

    
        
        
            
        
- [Sequence analysis](/training-material/topics/sequence-analysis)
    - Quality Control: [<i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides](/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html) - [<i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on](/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html)
    - Mapping: [<i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides](/training-material/topics/sequence-analysis/tutorials/mapping/slides.html) - [<i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on](/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html)
    






.footnote[Tip: press `P` to view the presenter notes]

???
Presenter notes contain extra information which might be useful if you
intend to use these slides for teaching.

Press `P` again to switch presenter notes off

---





### &lt;i class=&quot;fa fa-question-circle&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;question&lt;/span&gt; Questions


- What are barcodes and how are they applied to batches?

- What is the difference between a plate and a batch?

- Are batches and lanes the same?

- Why are plating setups important to know?

- Is it necessary to check barcodes that don't exist in a batch?


---


### &lt;i class=&quot;fa fa-bullseye&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;objectives&lt;/span&gt; Objectives


- How to describe a plating setup

- Proper naming conventions when dealing with multiple batches

- Checking for false positives across batches


---


### Sorting Plates


.image-100[![slide5](../../images/scrna-plates-batches-barcodes_slide5.svg)]

.left[Plates are *N x M* arrays of wells that cells are sorted, to then be individually amplified and sequenced.]

---

### Sorting Plates

.image-100[![slide6](../../images/scrna-plates-batches-barcodes_slide6.svg)]

.left[Not all wells need to be occupied by cells, some wells are left empty.]
.center[Why is this the case?]

---

### Sorting Plates

.image-100[![slide7](../../images/scrna-plates-batches-barcodes_slide7.svg)]

.center[Library preparation is not always perfect]
.center[Sample **contamination** from neighbouring wells]

---

### Plates and Lanes


.image-50[![slide5](../../images/scrna-plates-batches-barcodes_slide5.svg)]

.center[Sorting Plates are divided into *lanes*:]


.image-50[![slide9](../../images/scrna-plates-batches-barcodes_slide9.svg)]

???
Plates are $$N \times M$$ arrays or wells that cells are sorted into and then individually amplified and sequenced. The way these slot are filled depends entirely on the protocol, but usually not all slots are filled. The reason for this will become clear momentarily.

* Lanes demarcate evenly-sized *n x m* rectangular regions on a plate (where *n &lt; N and m &lt; M*).
* Multiple lanes can exist on plate, lanes ideally do not overlap within a plate.

---

### Plates and Lanes

.image-100[![slide9](../../images/scrna-plates-batches-barcodes_slide9.svg)]

* All slots within a lane are sequenced at the same time.
* All lanes within a plate *may or may not* be sequenced at the same time.
* Lanes can therefore be thought of as different *batches*.

---

### Lanes and Batches

.image-100[![slide10](../../images/scrna-plates-batches-barcodes_slide10.svg)]


???
* Lanes and batches describe the same spaces, but not the same samples.
* For this reason, batch numbers should *not* persist across different plates.

---

### Distinguishing cells in a plate

.image-100[![slide11](../../images/scrna-plates-batches-barcodes_slide11.svg)]


* Cells are selected from a plate by their *barcodes*
* Barcodes must be unique
  + e.g. *4 x 12  = 48* slots in the plate, need 48 unique cell barcodes

???
The way these slots are filled depends entirely on the protocol, but usually not all slots are filled. The reason for this will become clear momentarily

---

### What are Cell Barcodes?

.image-100[![Add Barcodes](../../images/scrna_pbb_barcodes_add.svg)]

* Barcodes are added to *all* transcripts of a *specific* cell
* Transcripts with different cell barcodes originate from different cells

???


---

### What are Cell Barcodes?

.image-100[![Barcodes in Plate](../../images/scrna_pbb_barcodes_overview.svg)]

* Many different cell barcodes are used across many different cells
* Each well in a plate contains a cell, indexed by its cell barcode

---

### Questions about Cell Barcodes


.image-75[![slide14](../../images/scrna-plates-batches-barcodes_slide14.svg)]

.left[Assuming a lane is a 20x5 array:]

1. How many cell barcodes are needed for a single lane?

1. How many cell barcodes are needed for a plate with 10 lanes?

1. What would be the minimum length of the barcodes for each of the previous questions?
--
.footnote[
1. *20 x 5 = 100 unique barcodes per lane*
1. *20 x 5 x 10 = 1,000 unique barcodes per plate*
]
---

### Questions about Cell Barcodes

.left[&lt;small&gt;3) What would be the minimum length of the barcodes for each of the previous questions?&lt;/small&gt;]

.pull-left[

A single lane?

 * 100 barcodes, 4 nucleotides for each base of a barcode

| Barcodes | Result |
|---------|------|
| $$4^2 =  16$$ | &lt;small&gt;No, 2 bases is not enough |
| $$4^3  = 64$$ | &lt;small&gt;No, 3 bases is not enough |
| $$4^4 = 256$$ | &lt;small&gt;Yes, 4 bases is enough to cover 100 barcodes (and more!) |

]
--
.pull-right[

A plate with 10 lanes?

 * 1000 barcodes, 4 nucleotides for each base of a barcode

| Barcodes | Result |
|---------|------|
|$$4^4 = 256$$ | &lt;small&gt;No, 4 bases is not enough |
|$$4^5 = 1024$$ | &lt;small&gt;Yes, 5 bases is enough to cover 1,000 barcodes (just barely!) |

]

---

### Barcode Safeguarding

&lt;!-- TODO:
 * Find a way to get &lt;style&gt;.MathJax_Display { display: inline; }&lt;/style&gt; working so that MathJax can render inline
 * The MathJax selector seems to apply directly to the element, which is impossible to override!
 --&gt;

.pull-left[
* Is 5 nucleotides really enough to capture 1000 cells?

* What could go wrong if all barcodes are separated by 1 bp, as shown?
]

.pull-right[
&lt;small&gt;*Edit distance = 1bp*

    AAAAA AAAAC AAAAG AAAAT AAACA AAAGA AAATA ····
    CCCCC CCCCA CCCCG CCCCT CCCAC CCCGC CCCTC ····
                        ·
                        ·
                        ·
]



???
If we assume that every barcode is separated from every other barcode by 1 bp, then the answer is 'yes'

That's right, sequencing errors can mislabel a read to a completely different cell than from where the transcript originated from.

---

### Guarding against Sequencing Errors

.pull-left[
* Sequencing errors can change the cell barcode

* To guard against a 1 bp sequencing error, need to space barcodes by 2 bp

* How many barcodes for a cell barcode of length 5 bp (ED = 2)?
]
.pull-right[
&lt;small&gt;*Edit Distance = 2 bp*


    AAAAA AAACC AAAGG AAATT AACCA AAGGA AATTA ····
    CCCCC CCCAA CCCGG CCCTT CCCAA CCGGC CCTTC ····
                        ·
                        ·
                        ·

]

--
$$ 4^{5-1} = 512 $$

---

### Edit Distance : General Principle

.pull-left[
e.g. For barcodes of length N=3:

* Edit Distance of E=1:
.center[`AAA AAC AAG AAT ACA ACC` ...]
.center[&lt;small&gt;64 barcodes&lt;/small&gt;]

* Edit Distance of E=2:
.center[`AAA ACC AGG ATT CAA CCC` ...]
.center[&lt;small&gt;16 barcodes&lt;/small&gt;]

* Edit Distance of E=3:
.center[`AAA CCC GGG TTT`]
.center[&lt;small&gt;4 barcodes&lt;/small&gt;]
]
--
.pull-right[
Number of Barcodes :

$$ 4^{N-(E-1)} $$

.center[For barcodes of length *N*,
and Edit Distance of *E*.]

]

---

### How many available barcodes are there?

.pull-left[
* Barcodes typically limited to 4 main bases {A,C,T,G}.

* Number of available barcodes depends on:

  1. The length of the barcode

  2. The edit distance between adjacent barcodes

* Must take sequencing errors into account

]
--
.pull-right[
* Availability is balance between *barcode size* vs. *sequencing errors*

* Typically between 4-10 bases, range of cells that can be labelled:

   $$[4^4, 4^{10}] = [256, 1048576]$$


* True range is lower due to edit distance.
]

???
* The availability of barcodes and the number of cells they can label is a design choice, where the technician must balance two opposing forces: barcode size vs sequencing errors.

* For the first, this means that for a barcode $$N$$ bases long, there will $$4^N$$ barcodes available. Typically barcodes tend to span 4-10 bases ($$4^4 = 256$$ to $$4^{10} = 1048576$$), since longer barcodes tend to be more subjectable to sequencing errors.

* The true number of barcodes used is actually smaller than $$4^N$$ due to the measures used to space barcodes apart from one another to reduce sequencing errors.

---

### Design Factors in Barcodes

Is it better to use Longer or Shorter Barcodes?

  | Longer | Shorter |
  |--------|---------|
  | Offer larger range of barcodes | Offer smaller range of barcodes |
  | Prone to more sequencing errors | More robust against sequencing errors |
  | Must increase edit distance significantly | A smaller edit distance is more acceptable |
  | Can accommodate large edit distances | *Cannot* accommodate large edit distances |

---

### Cell Barcodes: Summary

.pull-left[

* A single barcode sequence indexes a single cell

* Every transcript in a specific cell has the same cell barcode

* Barcodes are *designed* and are not random oligonucleotides

* Barcode use is *limited* by length and plate size

]
--
&lt;hline&gt;
.pull-right[#### Need to balance:

&lt;small&gt;

  | | | |
|-|-|-|
| Edit Distance | vs. | Barcode Length |
| Number of Barcodes | vs. | Plate and Lane Size |

&lt;/small&gt;
]
--
.pull-right[* How?]


---

### Intelligent Plating Strategies

* The main contending questions are:

 1. How large is each batch?

???
How many slots in a batch that we need to barcode for

--

 2. How many batches on a plate?

--

 3. Should each batch use the same barcodes?

--

 4. What **constraints** are there on the plate?



???
A technician always has to balance quality against cost, and this is illustrated in the following examples:

---

### Example Setup

.pull-left[
* Barcodes:
  * 24 unique barcodes, with an edit distance of E=2:

             AAA ACC AGG TTT TAA TCC
             ATT CCC CAA TGG NAA NCC
             CGG CTT GGG NGG NTT ANN
             GAA GCC GTT CNN GNN TNN

]
--
.pull-right[
* Plates and Lanes:
  * 12 slots per lane (3x4)
  * 4 lanes per plate

* Constraints
  * Only 2 lanes sequenced at the same time
    * i.e. half the plate is sequenced at the same time
]

???
Here we use N as an extra base just for example purposes, but you do sometimes see this in other barcodes.
2 lanes at a time, only half

---

### Example 1: Single Plate with a Single Lane

.bottom-info-box[
Available Barcodes

    AAA ACC AGG TTT TAA TCC
    ATT CCC CAA TGG NAA NCC
    CGG CTT GGG NGG NTT ANN
    GAA GCC GTT CNN GNN TNN

]
--

.image-75[![slide36](../../images/scrna-plates-batches-barcodes_slide36.svg)]

.pull-left[
* 12 total slots in Plate 1

* 12 slots in Lane 1

* All slots filled
]

--
.pull-right[
* We only need to use half of our barcodes

* Why is this wasteful?
]


???
* Half of the barcodes used for that lane, and the other half we can ignore.
* Wasteful because we are not getting full use of our 24 barcodes in a single sequencing run

---

### Example 2: Single Plate with 2 Lanes

.bottom-info-box[
Available Barcodes

    AAA ACC AGG TTT TAA TCC
    ATT CCC CAA TGG NAA NCC
    CGG CTT GGG NGG NTT ANN
    GAA GCC GTT CNN GNN TNN

]

* 24 total slots in Plate 1
* 12 slots in Lane 1, and 12 slots in Lane 2
* All slots filled

.image-75[![slide38](../../images/scrna-plates-batches-barcodes_slide38.svg)]

--
* We can use all of our barcodes
* Maximum number of cells can be sequenced in a single run
* Why might this be too optimal?


???
Here we use all barcodes since these lanes will be sequenced at the same time
Let's look at one final example to see why using all our barcodes on a plate might not be optimal.

---

### Example 3: Single Plate with 2 Lanes, only 1 active

.bottom-info-box[
Available Barcodes

    AAA ACC AGG TTT TAA TCC
    ATT CCC CAA TGG NAA NCC
    CGG CTT GGG NGG NTT ANN
    GAA GCC GTT CNN GNN TNN

]

* 24 total slots in Plate 1
* 12 slots in Lane 1, and 12 slots in Lane 2
* Only slots in Lane 1 filled

.image-75[![slide40](../../images/scrna-plates-batches-barcodes_slide40.svg)]


--
* Why are we still using all barcodes?
* Why have we not filled in Lane 2?


???
All barcodes used, why leave one lane empty?

---

### Example 3: Single Plate with 2 Lanes, only 1 active

.bottom-info-box[
Available Barcodes

    AAA ACC AGG TTT TAA TCC
    ATT CCC CAA TGG NAA NCC
    CGG CTT GGG NGG NTT ANN
    GAA GCC GTT CNN GNN TNN

]

* What would it mean if we sequenced reads with {TTT,TAA,...,TNN} as their Cell Barcodes?

.image-75[![slide40](../../images/scrna-plates-batches-barcodes_slide40.svg)]


--
* **Cross-Contamination**
  * There should be *no cells in that lane*!
  * These are contaminants from Lane 1
  * We can ignore these reads and select for the {AAA,ACC,...,GTT}


???
If we see any reads in the Plate which contain barcodes {TTT,TAA,TCC, etc} then we know that some contamination has occurred *because there should be no cells there*. One reason is that the second lane was not completely cleaned before being used.

---

### Example 4: 2 Plates with 2 Lanes, only 1 active (alternate)

.bottom-info-box[
Available Barcodes

    AAA ACC AGG TTT TAA TCC
    ATT CCC CAA TGG NAA NCC
    CGG CTT GGG NGG NTT ANN
    GAA GCC GTT CNN GNN TNN

]


* Same as before but extra plate. A single lane used to check for contaminants.
* Alternate barcodes used for each plate

.image-50[![slide44](../../images/scrna-plates-batches-barcodes_slide44.svg)]


&lt;small&gt;
* Why alternate active lanes across plates?
&lt;/small&gt;

--
&lt;small&gt;
  * Plate 1 and Plate 2 are sometimes the *same plate*
  * Contaminants might carry over *undetected* if same lane is used!
&lt;/small&gt;

???
* Here we have repeated previous example, but with an extra plate. In the first plate, the first half of the barcodes are used, and in the second plate, the second half of the barcodes are used.

* Why alternate the barcodes between plates? The full set of barcodes does not change, so why not keep the same format?

* Loaded at different times, washed clean, re-used.

* Again, the answer is to reduce cross-contamination. Plate2 will be loaded after Plate1 (and perhaps Plate2 and Plate1 will use the same plate!) If we see any reads in Plate2 that should not be there, we can now surmise where they came from. We also have the added benefit of protecting the cells in Plate2 from those that may have been used in Plate1, since they are in completely different positions across plates.

---

### Example 5: 1 Plate with 2 lanes, both active

.bottom-info-box[
Available Barcodes

    AAA ACC AGG TTT TAA TCC
    ATT CCC CAA TGG NAA NCC
    CGG CTT GGG NGG NTT ANN
    GAA GCC GTT CNN GNN TNN

]

* Both lanes filled. *All barcodes* are applied individually to *each* lane

.image-100[![slide46](../../images/scrna-plates-batches-barcodes_slide46.svg)]

.pull-left[

* Why check all barcodes against each lane?

* Why not separate lanes across different plates?

]
--
.pull-right[
* If &lt;small&gt;{TTT,TAA,...,TNN}&lt;/small&gt; is detected in Lane 1, or vice versa → **Contamination!**

* *Benefit of* **detecting cross-contamination** *whilst still* **maximising plate usage**
]
???
1. Why apply the full set of barcodes to each lane, when only half will actually label?
2. What benefit does this serve, instead of separating them over different plates as in the previous example?

A1. This setup is actually the same as example 4, but with the two plates merged. Here we can check for cross-contamination in each lane by measuring the real cell labels against the false barcodes. If in lane 1, we detect a significant number of reads with cell barcodes of `TAA` or `ANN`, we can assume that some cross-contamination has occurred since we should not be able to detect these barcodes in that lane. The converse is also true of lane 2.

A2. We have the benefit of detecting cross-contamination with the same advantages as example 4, but with the cost advantage of sequencing two batches at the same time.

---

# Summary

.pull-left[

* Cell Barcodes are sequences attached to transcripts to indicate what cell a transcript came from

* Cell Barcodes are *designed* to the Plate/Lane setup

]
--
.pull-right[

* Indicate what cell a transcript came from.

* Reduce sequencing errors

* Check for cross contamination

]




---
### &lt;i class=&quot;fa fa-key&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;keypoints&lt;/span&gt; Key points


- Plates are split into Lanes

- Batches are Lanes, but names are not reused

- Multiple batches can exist in many Plates

- Checking all barcodes across Batches eliminates false positives / cross-contamination

- Intelligent plating strategies can guard against cross-contamination





---
### &lt;i class=&quot;fa fa-graduation-cap&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;curriculum&lt;/span&gt; Do you want to extend your knowledge? 


Follow one of our recommended follow-up trainings:



    
        
        
            
        
- [Transcriptomics](/training-material/topics/transcriptomics)
    - Pre-processing of Single-Cell RNA Data: [&lt;i class=&quot;fa fa-laptop&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;tutorial&lt;/span&gt; hands-on](/training-material/topics/transcriptomics/tutorials/scrna_preprocessing/tutorial.html)
    





---

## Thank you!

This material is the result of a collaborative work. Thanks to the [Galaxy Training Network](https://wiki.galaxyproject.org/Teach/GTN) and all the contributors!


<div class="contributors-line">


<a href="/training-material/hall-of-fame#mtekman" class="contributor-badge"><img src="https://avatars.githubusercontent.com/mtekman" alt="Mehmet Tekman">Mehmet Tekman</a>, <a href="/training-material/hall-of-fame#astrovsky01" class="contributor-badge"><img src="https://avatars.githubusercontent.com/astrovsky01" alt="Alex Ostrovsky">Alex Ostrovsky</a>


</div>



<img src="/training-material/assets/images/GTN.png" alt="Galaxy Training Network" style="height: 100px;"/>



    </textarea>
    <script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript">
    </script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
    </script>
  </body>
</html>
