<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Proteogenomics 3: Novel peptide analysis</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:title" content="Galaxy Training: Proteogenomics 3: Novel peptide analysis" />
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Proteomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-novel-peptide-analysis%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-novel-peptide-analysis%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-novel-peptide-analysis%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-novel-peptide-analysis%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-novel-peptide-analysis%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Proteogenomics 3: Novel peptide analysis",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "proteomics / proteogenomics-novel-peptide-analysis / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Proteogenomics 3: Novel peptide analysis' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "Proteogenomics 3: Novel peptide analysis",
    "description": "Proteogenomics 3: Novel peptide analysis",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html",
  "timeRequired": "PT30M",
  "description": "Proteogenomics 3: Novel peptide analysis\\nThe questions this  addresses are:\n - How to verify the spectra of novel proteoforms?\n - How to assign genomic allocation to these novel proteoforms?\n\n\\nThe objectives are:\n - How to assign and visualize the genomic localization of these identified novel proteoforms?\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.1489208",
      "name": "Training data for Proteogenomics 3: Novel peptide analysis tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "James Johnson"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "James Johnson"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://galaxyproject.github.io//training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Proteogenomics 3: Novel peptide analysis</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#subinamehta" class="contributor-badge"><img src="https://avatars.githubusercontent.com/subinamehta" alt="Subina Mehta">Subina Mehta</a>, <a href="/training-material/hall-of-fame#timothygriffin" class="contributor-badge"><img src="https://avatars.githubusercontent.com/timothygriffin" alt="Timothy J. Griffin">Timothy J. Griffin</a>, <a href="/training-material/hall-of-fame#pratikdjagtap" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pratikdjagtap" alt="Pratik Jagtap">Pratik Jagtap</a>, <a href="/training-material/hall-of-fame#jraysajulga" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jraysajulga" alt="Ray Sajulga">Ray Sajulga</a>, <a href="/training-material/hall-of-fame#jj-umn" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jj-umn" alt="James Johnson">James Johnson</a>, <a href="/training-material/hall-of-fame#pravs3683" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pravs3683" alt="Praveen Kumar">Praveen Kumar</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>How to verify the spectra of novel proteoforms?</li>
            
            <li>How to assign genomic allocation to these novel proteoforms?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>How to assign and visualize the genomic localization of these identified novel proteoforms?</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 30 minutes</p>
            

            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    

                    
                    <div>
                        
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1489208">
    <i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>


                    </div>
                    

                    
                    <div>
                        
    <a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/workflows/" title="Workflows" alt="Proteogenomics 3: Novel peptide analysis workflows">
        <i class="fa fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>


                    </div>
                    
                    


                    
                    
                    <div>
                        

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">instances</span> Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fa fa-ship" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>



                    </div>
                    
                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>The third and the last workflow in the proteogenomics tutorial is to identifying the “<strong>Novel peptides</strong>” using BlastP and to localize the peptides to its genomic coordinates. Inputs from both workflow 1 and 2 will be used in this workflow.</p>

<p><img src="../../images/Third_workflow.png" alt="Workflow" /></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>
<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a></li>
  <li><a href="#peptide-selection" id="markdown-toc-peptide-selection">Peptide Selection</a></li>
  <li><a href="#multiomics-visualization-platform-mvp" id="markdown-toc-multiomics-visualization-platform-mvp">Multiomics Visualization Platform (MVP)</a></li>
  <li><a href="#obtain-peptide-genomic-coordinates" id="markdown-toc-obtain-peptide-genomic-coordinates">Obtain Peptide genomic Coordinates</a></li>
  <li><a href="#classify-peptides" id="markdown-toc-classify-peptides">Classify Peptides</a></li>
  <li><a href="#final-summary-output" id="markdown-toc-final-summary-output">Final Summary Output</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<p class="no_toc">All the files to run this workflow can be obtained from the <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html">second tutorial</a> output. Once the tabular output is generated, we convert this tabular report into a FASTA file. This can be achieved by using the Tabular to FASTA convertion tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-organization"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: data organization</h3>

  <ol>
    <li>The inputs for this workflow are:
      <ul>
        <li><strong>Tabular file</strong> – “<strong>Peptides from BlastP analysis</strong>”</li>
        <li><strong>Tabular file</strong> – “<strong>PeptideShaker_PSM</strong>”</li>
        <li><strong>Mz to sqlite</strong></li>
        <li><strong>Genomic mapping sqlite</strong></li>
      </ul>
    </li>
  </ol>

  <p>If you do not have these files from the previous tutorials in this series, you can import them from Zenodo:</p>
  <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1489208/files/Peptides_for_Blast-P_analysis.tabular
https://zenodo.org/record/1489208/files/PeptideShaker_PSM.tabular
https://zenodo.org/record/1489208/files/mz_to_sqlite.mz.sqlite
https://zenodo.org/record/1489208/files/genomic_mapping_sqlite.sqlite
</code></pre></div>  </div>
</blockquote>

<h1 id="peptide-selection">Peptide Selection</h1>

<p>Once Blast-P search is performed, it provides a tabular output containing “<strong>Novel peptides</strong>”. Now this output is further processed by comparing the Novel Peptide output with the PSM report for selecting only distinct peptides which meet the criteria.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-query-tabular"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Query Tabular</h3>

  <ol>
    <li><strong>Query Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong>
          <ul>
            <li>Section <strong>Table Options</strong>
              <ul>
                <li><em>“Specify Name for Table”</em>: <code class="highlighter-rouge">blast</code></li>
                <li><em>“Use first line as column names”</em> : <code class="highlighter-rouge">No</code></li>
                <li><em>“Specify Column Names (comma-separated list)”</em>:
<code class="highlighter-rouge">qseqid,sseqid,pident,length,mismatch,gapopen,qstart,qend,sstart,send,evalue,bitscore,sallseqid,score,nident,positive,gaps,ppos,qframe,sframe,qseq,sseq,qlen,slen,salltitles</code></li>
                <li><em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">Yes</code></li>
                <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Table Index</strong>
                  <ul>
                    <li><em>“Table Index”</em>: <code class="highlighter-rouge">No</code></li>
                    <li><em>“Index on Columns”</em>: <code class="highlighter-rouge">id</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong>
          <ul>
            <li>Section <strong>Filter Dataset Input</strong>
              <ul>
                <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Filter Tabular Input Lines</strong>
                  <ul>
                    <li><em>“Filter by”</em>:  <code class="highlighter-rouge">skip leading lines</code></li>
                    <li><em>“Skip lines”</em>: <code class="highlighter-rouge">1</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Section <strong>Table Options</strong>
              <ul>
                <li><em>“Specify Name for Table”</em>: <code class="highlighter-rouge">psm</code></li>
                <li><em>“Use first line as column names”</em> : <code class="highlighter-rouge">No</code></li>
                <li><em>“Specify Column Names (comma-separated list)”</em>: <code class="highlighter-rouge">ID,Proteins,Sequence,AAs_Before,AAs_After,Position,Modified_Sequence,Variable_Modifications,Fixed_Modifications,Spectrum_File,Spectrum_Title,Spectrum_Scan_Number,RT,mz,Measured_Charge,Identification_Charge,Theoretical_Mass,Isotope_Number,Precursor_mz_Error_ppm,Localization_Confidence,Probabilistic_PTM_score,Dscore,Confidence,Validation</code></li>
                <li><em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>
          <p><em>“Save the sqlite database in your history”</em>: <code class="highlighter-rouge">No</code></p>

          <blockquote class="comment">
            <h3 id="comment-querying-an-sqlite-database"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Querying an SQLite Database</h3>

            <p><strong>Query Tabular</strong> can also use an existing SQLite database. Activating <code class="highlighter-rouge">Save the sqlite database in your history</code>
will store the created database in the history, allowing to reuse it directly.</p>

          </blockquote>
        </li>
        <li><em>“SQL Query to generate tabular output”</em>:
          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT distinct psm.*
FROM psm join blast ON psm.Sequence = blast.qseqid
WHERE blast.pident &lt; 100 OR blast.gapopen &gt;= 1 OR blast.length &lt; blast.qlen
ORDER BY psm.Sequence, psm.ID
</code></pre></div>          </div>
        </li>
        <li><em>“Include query result column headers”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong> and inspect the query results file after it turned green.</li>
  </ol>

</blockquote>

<p>Once this step is completed, a tabular output containing novel proteoforms are displayed. These novel proteforms fulfill our criteria of not being present in the existing NCBI repository. The next step is to remove any duplicate sequences. For this, we use the Query tabular tool again to select distinct sequences from the tabular output.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-query-tabular-1"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Query Tabular</h3>

  <ol>
    <li><strong>Query Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: with the following parameters:
      <ul>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong>
          <ul>
            <li>Section <strong>Filter Dataset Input</strong>
              <ul>
                <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
                  <ul>
                    <li><em>“Filter by”</em>:  <code class="highlighter-rouge">skip leading lines</code></li>
                    <li><em>“Skip lines”</em>: <code class="highlighter-rouge">1</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Section <strong>Table Options</strong>
              <ul>
                <li><em>“Specify Name for Table”</em>: <code class="highlighter-rouge">psm</code></li>
                <li><em>“Use first line as column names”</em> : <code class="highlighter-rouge">No</code></li>
                <li><em>“Specify Column Names (comma-separated list)”</em>:<code class="highlighter-rouge">ID,Proteins,Sequence</code></li>
                <li><em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“SQL Query to generate tabular output”</em>:
          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT distinct Sequence from psm
</code></pre></div>          </div>
        </li>
        <li><em>“include query result column headers”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong> and inspect the query results file after it turned green.</li>
  </ol>

</blockquote>

<h1 id="multiomics-visualization-platform-mvp">Multiomics Visualization Platform (MVP)</h1>

<p>The Multiomics Visualization Platform is a galaxy visualization plugin that allows the user to browse the selected proteomics data. It uses the SQlite database which allows the data to be filtered and aggregated in a user defined manner. It allows various features such as; the PSM can be displayed with a lorikeet spectral view, the selected peptide can be displayed in a protein view and an IGV browser is also available for the selected protein. The step by step guide shown below will provide a walkthrough on how to use this plugin.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-guide-to-mvp"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Guide to MVP</h3>

  <p>The spectra belonging to these “Novel peptides” can be viewed using MVP,this can be achieved by selecting the output from the mz to sqlite tool.
Here is a step by step guide to obtain the proteogenomic view of the “Novel peptides”.</p>

  <p>1) Click on the <strong>Visualize in MVP application</strong>, it will open up a new window for MVP.</p>

  <p><img src="../../images/mz2sqlite.png" alt="mz to sqlite" width="20%" /></p>

  <p>2) Click on <strong>Load from Galaxy</strong>.</p>

  <p><img src="../../images/load_from_Galaxy.png" alt="load from galaxy" width="50%" /></p>

  <p>3) Select <strong>Novel Peptides</strong> from the right hand side.</p>

  <p><img src="../../images/novel_peptides_view.png" alt="novel peptides view" width="50%" /></p>

  <p>4) Select any peptide, For eg: <code class="highlighter-rouge">ESSREALVEPTSESPRPALAR</code>, and then click on <strong>Selected Peptide PSMs</strong>.</p>

  <p><img src="../../images/select_pep_PSM.png" alt="select_pep_PSM" width="50%" /></p>

  <p>5) If you scroll down, the PSM associated with the peptide will be displayed. By clicking on the PSM, the <em>Lorikeet</em>
values will be shown. The Lorikeet visualization is interactive, i.e the user can change the values or select any
parameter and click on Update button to view these changes.</p>

  <p><img src="../../images/Psm.png" alt="PSM" width="40%" /></p>

  <p><img src="../../images/lorikeet.png" alt="lorikeet" width="70%" /></p>

  <p>6) For a Protein centric view, click on <strong>View in Protein</strong> , it will open up all the proteins associate with the
peptides. For eg: Select the <code class="highlighter-rouge">ESSREALVEPTSESPRPALAR</code> peptide and click on the first protein. The chromosome location
of the peptide will be displayed.</p>

  <p><img src="../../images/view_in_prot.png" alt="view in protein" width="30%" /></p>

  <p>Once you click on protein it will show the list of proteins the belongs to the peptides.</p>

  <p><img src="../../images/select_protein.png" alt="select protein" width="60%" /></p>

  <p>Once you select the protein that you want to visualize you can click on the protein view.</p>

  <p><img src="../../images/PRoteinview.png" alt="protein view" width="50%" /></p>

  <p>7) Clicking on the arrow marks will open up the IGV(js) visualization tool, where-in the genomic localization of the
peptide will be displayed.</p>

  <p><img src="../../images/select_IGV.png" alt="select IGV" width="80%" /></p>

  <p>8) To add tracks to your IGV viewer, click on <strong>Add Track</strong>. This will open up a list of tracks that are compatible
to view in your IGV viewer. For eg. Select the <code class="highlighter-rouge">Pep_gen_coordinate.bed</code> file and then click on <strong>Load Track</strong>.
This will open up the bed will below the nucleotide sequence.</p>

  <p><img src="../../images/track_load.png" alt="track load" width="40%" /></p>

  <p>9) By clicking the wheel, you can select the <strong>three frame translate</strong> which will show the three frame translated
region of your sequence.</p>

  <p><img src="../../images/IGV_viewer.png" alt="IGV viewer" width="40%" /></p>

  <p>10) The IGV is inbuilt in the MVP viewer and is very interactive, you could also load more tracks such as the aligned
proBAM file (from HISAT) or the identified probam file (one of the input file).
MVP has many useful features beyond those covered in this workshop and is under active development.</p>

  <p><img src="../../images/tracks_align.png" alt="tracks align" width="70%" /></p>

</blockquote>

<p>The next tool in the workflow is the Peptide genomic coordinate tool which takes the “novel peptides” as the input along with the mztosqlite file and the genomic mapping sqlite file (obtained during creation of the database). This tool helps create a bed file with the genomic coordinate information of the peptides based on the sqlite files.</p>

<h1 id="obtain-peptide-genomic-coordinates">Obtain Peptide genomic Coordinates</h1>

<p>Gets genomic coordinate of peptides based on the information in mzsqlite and genomic mapping sqlite files. This program
loads two sqlite databases (mzsqlite and genomic mapping sqlite files) and calculates the genomic coordinates of the
peptides provided as input. This outputs bed file for peptides.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-peptide-genomic-coordinate"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Peptide genomic Coordinate</h3>

  <ol>
    <li>Run <strong>Peptide genomic Coordinate</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Input”</em>: <code class="highlighter-rouge">Peptide list file</code>, <code class="highlighter-rouge">mzsqlite sqlite DB file</code>, and <code class="highlighter-rouge">genomic mapping sqlite DB file</code></li>
        <li>
          <p><em>“Output”</em>: <code class="highlighter-rouge">Tabular BED file with all the columns</code></p>

          <p><img src="../../images/Output_PGC.png" alt="Output PGC" width="50%" /></p>
        </li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong> and inspect the resulting files</li>
  </ol>

</blockquote>

<h1 id="classify-peptides">Classify Peptides</h1>

<p>Given chromosomal locations of peptides in a BED file, PepPointer classifies them as CDS, UTR, exon, intron, or intergene.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-peppointer"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Peppointer</h3>

  <ol>
    <li><strong>Peppointer</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Choose the source of the GTF file”</em> - <code class="highlighter-rouge">Locally Installed</code></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GTF file with the genome of interest”</em> - <code class="highlighter-rouge">Mus_Musculus_GRCm38.90_Ensembl_GTF</code></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BED file with chromosomal coordinates of peptides”</em>: <code class="highlighter-rouge">Bed file from Peptide genomic coordinate tool</code></li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong> and inspect the query results file after it turned green.</li>
  </ol>

  <p>This tool provides a bed output with the classification of the genomic location of the peptides.</p>

  <p><img src="../../images/Output_PP.png" alt="Output PP" width="50%" /></p>

</blockquote>

<p>The final tool for this workflow generates a tabular output that summarizes the information after running these workflows. The final summary output consists of the Peptide sequence, the spectra associated with the peptides, the protein accession number, chromosome number, Start and Stop of the genomic coordinate, the annotation, the genomic coordinate entry for viewing in Integrative Genomics Viewer (IGV), MVP or UCSC genome browser and the URL for viewing it on UCSC genome browser. This summary is created with the help of the query tabular tool.</p>

<h1 id="final-summary-output">Final Summary Output</h1>

<blockquote>
  <h3 id="hands_on-hands-on-query-tabular-2"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Query Tabular</h3>

  <ol>
    <li><strong>Query Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span>  <strong>Insert Database Table</strong>
          <ul>
            <li>Section <strong>Table Options</strong>:
              <ul>
                <li><em>“Specify Name for Table”</em>: <code class="highlighter-rouge">bed_pep_pointer</code></li>
                <li><em>“Use first line as column names”</em> : <code class="highlighter-rouge">No</code></li>
                <li><em>“Specify Column Names (comma-separated list)”</em>:<code class="highlighter-rouge">chrom,start,end,peptide,score,strand,annot</code></li>
                <li><em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong>
          <ul>
            <li>Section <strong>Filter Dataset Input</strong>
              <ul>
                <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
                  <ul>
                    <li><em>“Filter by”</em>:  <code class="highlighter-rouge">skip leading lines</code></li>
                    <li><em>“Skip lines”</em>: <code class="highlighter-rouge">1</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Section <strong>Table Options</strong>
              <ul>
                <li><em>“Specify Name for Table”</em>: <code class="highlighter-rouge">psm</code></li>
                <li><em>“Use first line as column names”</em> : <code class="highlighter-rouge">No</code></li>
                <li><em>“Specify Column Names (comma-separated list)”</em>: <code class="highlighter-rouge">ID,Proteins,Sequence,AAs_Before,AAs_After,Position,Modified_Sequence,Variable_Modifications,Fixed_Modifications,Spectrum_File,Spectrum_Title,Spectrum_Scan_Number,RT,mz,Measured_Charge,Identification_Charge,Theoretical_Mass,Isotope_Number,Precursor_mz_Error_ppm,Localization_Confidence,Probabilistic_PTM_score,Dscore,Confidence,Validation</code></li>
                <li><em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“SQL Query to generate tabular output”</em>:
          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT psm.Sequence AS PeptideSequence,
count(psm.Sequence) AS SpectralCount,
psm.Proteins AS Proteins,
bed_pep_pointer.chrom AS Chromosome,
bed_pep_pointer.start AS Start,
bed_pep_pointer.end AS End,
bed_pep_pointer.strand AS Strand,
bed_pep_pointer.annot AS Annotation,
bed_pep_pointer.chrom||':'||bed_pep_pointer.start||'-'||bed_pep_pointer.end AS GenomeCoordinate,
'https://genome.ucsc.edu/cgi-bin/hgTracks?db=mm10&amp;position='||bed_pep_pointer.chrom||'%3A'||bed_pep_pointer.start||'-'||bed_pep_pointer.end AS UCSC_Genome_Browser
FROM psm
INNER JOIN bed_pep_pointer on bed_pep_pointer.peptide = psm.Sequence
GROUP BY psm.Sequence
</code></pre></div>          </div>
        </li>
        <li><em>“include query result column headers”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Execute</strong> and inspect the query results file after it turned green. If everything went well, it should look similiar:</p>

      <p><img src="../../images/final_summary.png" alt="Final Summary" width="100%" /></p>
    </li>
  </ol>

  <p>The Final summary displays a tabular output containing the list of novel peptides and its corresponding protein. It also provides the users with the chromosomal location of the novel proteoform along with the peptide’s start and end position.	The output also features the strand information, gene	annotation and the genomic coordinates in a specific format that could be used on IGV or UCSC browser. It also provides the user with a	UCSC Genome Browser link, which the user can directly copy and paste it on a web browser to learn more about the novel proteoform. Here we are demonstrating the use of proteogenomics workflow on an example trimmed mouse dataset. This study explores the possibilities for downstream biological /functional analysis of peptides corresponding to novel proteoforms.</p>
</blockquote>

<h3 class="hands_on" id="whats-next">What’s next?</h3>

<p>This completes the proteogenomics workflow analysis. This training workflow uses mouse data. For any other organism the data, tool paramters and the workflow will need to be modified accordingly.</p>

<p>Please look at the following tutorials in this proteogenomics series:</p>
<ol>
  <li><a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html">Proteogenomics database creation</a></li>
  <li><a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html">Proteogenomics database search</a></li>
</ol>

<p>This workflow is also available at <a href="http://galaxyp-proteogenomics.duckdns.org/">Proteogenomics gateway</a>.</p>

<p>This workflow was developed by the Galaxy-P team at the University of Minnesota.
For more information about Galaxy-P or our ongoing work, please visit us at <a href="http://galaxyp.org">galaxyp.org</a></p>




                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>Learning how to visualize proteomic data and to perform its genomic allocation</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                    

                    

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
        <blockquote class="agenda follow-up">
            <strong class="follow-up"><i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
            <ul>
                
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Proteogenomics 1: Database Creation:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Proteogenomics 2: Database Search:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
        </blockquote>
        

        <hr>
        <br>

        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Proteogenomics 3: Novel peptide analysis (Proteomics)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=Proteogenomics 3: Novel peptide analysis (Proteomics)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
