<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>EWAS data analysis of 450k data</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:title" content="Galaxy Training: EWAS data analysis of 450k data" />
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Epigenetics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/epigenetics/tutorials/ewas-suite/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "EWAS data analysis of 450k data",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "epigenetics / ewas-suite / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'EWAS data analysis of 450k data' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "EWAS data analysis of 450k data",
    "description": "EWAS data analysis of 450k data",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html",
  "timeRequired": "PT1H",
  "description": "EWAS data analysis of 450k data\\nThe questions this  addresses are:\n - What is a epigenome-wide association studies?\n - Why is a EWAS analysis useful?\n - Why 450k data?\n\n\\nThe objectives are:\n - Learn how to perform reproducible EWAS analysis\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/1251211#.WwREQ1Mvz-Y",
      "name": "Training data for EWAS data analysis of 450k data tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Katarzyna Murat"
    },
    {
      "@type": "Person",
      "name": "Krzysztof Poterlowicz"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Katarzyna Murat"
    },
    {
      "@type": "Person",
      "name": "Krzysztof Poterlowicz"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://galaxyproject.github.io//training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>EWAS data analysis of 450k data</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#kpbioteam" class="contributor-badge"><img src="https://avatars.githubusercontent.com/kpbioteam" alt="Katarzyna Murat">Katarzyna Murat</a>, <a href="/training-material/hall-of-fame#kpoterlowicz" class="contributor-badge"><img src="https://avatars.githubusercontent.com/kpoterlowicz" alt="Krzysztof Poterlowicz">Krzysztof Poterlowicz</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>What is a epigenome-wide association studies?</li>
            
            <li>Why is a EWAS analysis useful?</li>
            
            <li>Why 450k data?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Learn how to perform reproducible EWAS analysis</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour</p>
            

            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    
                    <div>
                        <a class="topic-icon" href="/training-material/topics/epigenetics/slides/introduction.html" title="Introduction slides">
                            <i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Introduction Slides
                        </a>
                    </div>
                    

                    

                    
                    


                    
                    
                    <div>
                        

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">instances</span> Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/docker" title="Docker image for this tutorial">
                <i class="fa fa-ship" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>



                    </div>
                    
                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>The field of cancer genomics has demonstrated the power of massively parallel sequencing techniques to inform on the genes and specific alterations that drive tumor onset and progression. Although large comprehensive sequence data sets continue to be made increasingly available, data analysis remains an ongoing challenge, particularly for laboratories lacking dedicated resources and bioinformatics expertise. To address this, we have provide training based on  Galaxy tools ewas suite that represent many popular algorithms for detecting somatic genetic alterations from cancer genome and exome data.</p>

<figure id="figure-1"><img src="../../images/Epigenetic_mechanisms.jpg" alt="epimechanism" /><figcaption><span class="figcaption-prefix">Figure 1:</span> How epigenetics mechanism can effect health (adapted from https://commonfund.nih.gov/epigenomics/figure)</figcaption></figure>

<p>This exercise uses datasets from the Cell publication by Hugo, Willy, et al., 2015 <a class="citation" href="#Hugo2015">Hugo <i>et al.</i> 2015</a>. with the goal to identify differentially methylated regions and positions associated with treatment resistant melanomas.
Datasets include Illumina 450K Methylation array performed in melanoma tumors in sample of patients pre and post MAPKi and BRAFi treatment with  different outcomes (sensitive and resistant). For each sample there is raw green and red colour arrays containing the summarized bead information generated bythe Illumina 450k scanner, i.e. there are 8 files in total:</p>

<p>The Infinium Methylation Assay 450k uses two different bead types to detect changes in DNA methylation levels. In the figure we can see M - methylated and U - unmethylated bead types. In our study unmethylated and methylated bead signals are reported as green and red colors respectively.</p>

<figure id="figure-2"><img src="../../images/methylationassay.png" alt="methassay" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Infinium Methylation Assay Overview (adapted from https://emea.illumina.com/science/technology/beadarray-technology/infinium-methylation-assay.html?langsel=/gb/)</figcaption></figure>

<table>
  <thead>
    <tr>
      <th>Accession</th>
      <th>Sensitivity</th>
      <th>Treatment</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>GSM1588704</td>
      <td>baseline</td>
      <td>pre-treatment</td>
    </tr>
    <tr>
      <td>GSM1588705</td>
      <td>baseline</td>
      <td>pre-treatment</td>
    </tr>
    <tr>
      <td>GSM1588706</td>
      <td>resistant</td>
      <td>BRAFi</td>
    </tr>
    <tr>
      <td>GSM1588707</td>
      <td>resistant</td>
      <td>BRAFi</td>
    </tr>
  </tbody>
</table>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

<ol id="markdown-toc">
  <li><a href="#450k-array-data-upload" id="markdown-toc-450k-array-data-upload">450K Array Data Upload</a></li>
  <li><a href="#preprocessing-and--quality-assessment" id="markdown-toc-preprocessing-and--quality-assessment">Preprocessing and  Quality Assessment</a></li>
  <li><a href="#removing-probes-affected-by-genetic-variation" id="markdown-toc-removing-probes-affected-by-genetic-variation">Removing probes affected by genetic variation</a></li>
  <li><a href="#dmps-and-dmrs-identification" id="markdown-toc-dmps-and-dmrs-identification">DMPs and DMRs Identification</a></li>
  <li><a href="#annotation-and-visualization" id="markdown-toc-annotation-and-visualization">Annotation and Visualization</a></li>
</ol>

</blockquote>

<h1 id="450k-array-data-upload">450K Array Data Upload</h1>

<p>The first step of EWAS data anylalysis is the raw methylation data loading (intensity information files for each two color micro array)</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-loading"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data Loading</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>Import the following IDAT files from <a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y">Zenodo</a> or from the data library (ask your instructor)
      <ul>
        <li><code class="highlighter-rouge">GSM1588704_8795207135_R01C02_Red.idat</code></li>
        <li><code class="highlighter-rouge">GSM1588705_8795207119_R05C02_Red.idat</code></li>
        <li><code class="highlighter-rouge">GSM1588706_8795207135_R02C02_Red.idat</code></li>
        <li><code class="highlighter-rouge">GSM1588707_8795207119_R06C02_Red.idat</code></li>
        <li><code class="highlighter-rouge">GSM1588704_8795207135_R01C02_Grn.idat</code></li>
        <li><code class="highlighter-rouge">GSM1588705_8795207119_R05C02_Grn.idat</code></li>
        <li><code class="highlighter-rouge">GSM1588706_8795207135_R02C02_Grn.idat</code></li>
        <li><code class="highlighter-rouge">GSM1588707_8795207119_R06C02_Grn.idat</code></li>
      </ul>

      <blockquote class="details">
        <h3 id="details-list-of-zenodo-urls"><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> List of Zenodo URLs</h3>
        <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1251211/files/GSM1588704_8795207135_R01C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588706_8795207135_R02C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588705_8795207119_R05C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588707_8795207119_R06C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588704_8795207135_R01C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588706_8795207135_R02C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588705_8795207119_R05C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588707_8795207119_R06C02_Grn.idat
</code></pre></div>        </div>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the Galaxy Upload Manager (<i class="fa fa fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, Galaxy uses the URL as the name, so rename the files with a more useful name.</p>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>
    </li>
    <li>Run <strong>minfi read450k</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“red channel files”</em>: all files ending in <code class="highlighter-rouge">_Red</code></li>
        <li><i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“green channel files”</em>: all files ending in <code class="highlighter-rouge">Grn</code></li>
      </ul>
    </li>
    <li>Inspect generated set of data</li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How are the Green and Red signals stored?</p>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>Green and Red micro arrays are built up into <code class="highlighter-rouge">RGChannelSet</code></p>
  </blockquote>

</blockquote>

<h1 id="preprocessing-and--quality-assessment">Preprocessing and  Quality Assessment</h1>

<p>Preprocessing and data quality assurance is an important step in Infinium Methylation Assay analysis.</p>

<p><code class="highlighter-rouge">RGChannelSet</code> represents two color data with a green and a red channel and can be converted into methylated and unmethylated signals assigned to <code class="highlighter-rouge">MethylSet</code> or into Beta values build in <code class="highlighter-rouge">RatioSet</code>. User can convert from <code class="highlighter-rouge">RGChannelSet</code> into <code class="highlighter-rouge">MethylSet</code> using the <strong>minfi mset</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> or compute Beta values using <strong>minfi set</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. The <strong>minfi qc</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool extracts and plots the quality control data frame with two columns mMed and uMed which are the medians of <code class="highlighter-rouge">MethylSet</code> signals (Meth and Unmeth).Comparing them against one another allows user to detect and remove low-quality samples.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-preprocessing"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Preprocessing</h3>
  <ol>
    <li>Run <strong>minfi mset</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the output from minfi_read450K to create <code class="highlighter-rouge">MethylSet</code> object</li>
    <li>Run <strong>minfi qc</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>  on the ‘MethylSet’ object created in the previous step to estimate sample-specific quality control. Once it is done click on the <i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon next to the QC filename in your history to view QC image.</li>
    <li>Once you have inspected the quality control run <strong>minfi rset</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the  ‘MethylSet’output from minfi_read450K  to convert methylation data from the <code class="highlighter-rouge">MethylSet</code> to ratio of the methylated probe intensity and the overall intensity (sum of methylated and unmethylated probe intensities)</li>
    <li>Run <strong>minfi maptogenome</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the output from the previous step to map ratio data to the genomic positions</li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-preprocess-and-normalize-data"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Preprocess and Normalize data</h3>
    <p>If your files require normalisation, you might prefer to use other of preprocessing tools provided in EWAS suite i.e. <strong>minfi ppfun</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> or <strong>minfi ppquantile</strong>  <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> look for recommendation at (ref).</p>

  </blockquote>

</blockquote>

<figure id="figure-3"><img src="../../images/qcplot.png" alt="Quality Control" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Quality control plot</figcaption></figure>

<h1 id="removing-probes-affected-by-genetic-variation">Removing probes affected by genetic variation</h1>

<p>Incomplete annotation of genetic variations such as single nucleotide polymorphism (SNP) may affect DNA measurements and interfere results from downstream analysis.</p>

<p>Hansen, J. P. 2014 highly recommended to remove the probes that contain either a SNP at the methylated loci interrogation or at the single nucleotide extension</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-removing-probes-affected-by-genetic-variation"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Removing probes affected by genetic variation</h3>
  <ol>
    <li>Run <strong>minfi dropsnp</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>  on the output of minfi_maptogenome to remove the probes that contain either a SNP at the methylated loci interrogation or at the single nucleotide extension</li>
  </ol>
</blockquote>

<h1 id="dmps-and-dmrs-identification">DMPs and DMRs Identification</h1>

<p>The main goal of the EWAS suite is to simplify the way differentially methylated loci sites are detected. The EWAS suite contains <strong>minfi dmp</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool detecting differentially methylated positions (DMPs) with respect to a phenotype covariate, and more complex <strong>minfi dmr</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> solution for finding differentially methylated regions (DMRs). Genomic regions that are differentially methylated between two conditions can be tracked using a bumphunting algorithm. The algorithm first implements a t-statistic at each methylated loci location, with optional smoothing, then groups probes into clusters with a maximum location gap and a cutoff size to refer the lowest possible value of genomic profile hunted by our tool.</p>

<p>In order the next we will need additional file that annotate illumina450K with genomic features, which we can obtain from UCSC.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-from-ucsc"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload from UCSC</h3>

  <ol>
    <li>
      <p>Search for <code class="highlighter-rouge">UCSC Main</code> in the tool search bar (top left)</p>
    </li>
    <li>
      <p>Click on <code class="highlighter-rouge">UCSC Main</code> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. You will be taken to the <strong>UCSC table browser</strong></p>
    </li>
    <li>Set the following options:
      <ul>
        <li><em>“clade”</em>: <code class="highlighter-rouge">Mammal</code></li>
        <li><em>“genome”</em>: <code class="highlighter-rouge">Human</code></li>
        <li><em>“assembly”</em>: <code class="highlighter-rouge">Feb. 2009 (GRCh37/hg19)</code></li>
        <li><em>“group”</em>: <code class="highlighter-rouge">Regulation</code></li>
        <li><em>“track”</em>: <code class="highlighter-rouge">HAIB Methyl450</code></li>
        <li><em>“table”</em>: <code class="highlighter-rouge">GM12878 (wgEncodeHaibMethyl450Gm12878SitesRep1)</code></li>
        <li><em>“region”</em>: <code class="highlighter-rouge">genome</code></li>
        <li><em>“output format”</em>: <code class="highlighter-rouge">GTF - gene transfer (limited)</code></li>
        <li><em>“Send output to”</em>: <code class="highlighter-rouge">Galaxy</code> (only)</li>
      </ul>
    </li>
    <li>Click on the <strong>get output</strong> button at the bottom of the screen</li>
    <li>On the next page, click on the <strong>Send Query to Galaxy</strong> button</li>
    <li>Wait for the upload to finish</li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-dmps-and-dmrs-identification"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: DMPs and DMRs Identification</h3>

  <ol>
    <li>Import <code class="highlighter-rouge">phenotypeTable.txt</code> from <a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y">Zenodo</a> or data library:
      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1251211/files/phenotypeTable.txt
</code></pre></div>      </div>
    </li>
    <li>Run <strong>minfi dmp</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the output of the minfi dropsnp tool adjusting the following parameters
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input set”</em>:<code class="highlighter-rouge">GenomicRatioSet</code></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Phenotype Table”</em>:<code class="highlighter-rouge">phenotypeTable.txt</code></li>
        <li><em>“Phenotype Type”</em>:<code class="highlighter-rouge">categorical</code></li>
        <li><em>“qCutoff Size”</em>:<code class="highlighter-rouge">0.5</code> (DMPs with an FDR q-value greater than this will not be returned)</li>
        <li><em>“Variance Shrinkage”</em>:<code class="highlighter-rouge"> TRUE</code> (This is recommended when sample sizes are small &lt;10}</li>
        <li><em>“Genome Table”</em>: <code class="highlighter-rouge">wgEncodeHaibMethyl450 ... </code></li>
      </ul>
    </li>
    <li>Run <strong>minfi dmr</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the output of minfi_dmp adjusting the following parameters
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input set”</em>:<code class="highlighter-rouge">GenomicRatioSet</code></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Phenotype Table”</em>:<code class="highlighter-rouge">phenotypeTable.txt</code></li>
        <li><em>“factor1”</em>: <code class="highlighter-rouge">sensitive</code></li>
        <li><em>“factor2”</em>: <code class="highlighter-rouge">resistant</code></li>
        <li><em>“maxGap Size”</em>:<code class="highlighter-rouge">250</code></li>
        <li><em>“coef Size”</em>:<code class="highlighter-rouge">2</code></li>
        <li><em>“Cutoff Size”</em>:<code class="highlighter-rouge">0.1</code></li>
        <li><em>“nullMethod”</em>:<code class="highlighter-rouge">permutation</code></li>
        <li><em>“verbose”</em>:<code class="highlighter-rouge">TRUE</code></li>
      </ul>
    </li>
    <li>Click on the <strong>minfi dmr</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> output in your history to expand it</li>
    <li>Set the database build of your dataset to <code class="highlighter-rouge">Human Feb. 2009 (GRCh37/hg19) (hg19)</code>
      <blockquote class="tip">

        <h3 id="tip-tip-changing-databasebuild-dbkey"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing Database/Build (dbkey)</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Database/Build</strong> field</li>
          <li>Select your desired database key from the dropdown list: <code class="highlighter-rouge">hg19</code></li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>Click on <code class="highlighter-rouge">display at UCSC</code> towards the bottom of the history item.
 This will launch UCSC Genome Browser with your Custom Track</li>
  </ol>
</blockquote>

<figure id="figure-4"><img src="../../images/ucsc.png" alt="Display at UCSC" /><figcaption><span class="figcaption-prefix">Figure 4:</span> UCSC genome track showing differentialy methylated regions located on chromosome 6</figcaption></figure>

<blockquote class="comment">
  <h3 id="comment-phenotype-table"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Phenotype table</h3>
  <p>Phenotype table can be in different size with different arguments only second column is required to contain phenotype covariate information for each sample.</p>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <p>How do we define phenotype covariate?</p>
  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>Phenotype covariate is the set of observable characteristics of an individual resulting from the gene-environment interactions</p>
  </blockquote>
</blockquote>

<h1 id="annotation-and-visualization">Annotation and Visualization</h1>

<p>In addition to downstream analysis users can annotate the differentially methylated loci to the promoter regions of genes with gene function description, and relationships between these concepts.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on--annotate-differentially-methylated-position"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on:  Annotate Differentially Methylated Position</h3>
  <ol>
    <li>Run <strong>chipeakanno annopeaks</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>on the output of minfi_dmp with the following parameters
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Differentialy methylated data”</em>: output of <strong>minfi dmp</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“bindingType”</em>: <code class="highlighter-rouge">StartSite</code></li>
        <li><em>“bindingRegionStart”</em>:<code class="highlighter-rouge">-5000</code></li>
        <li><em>“bindingRegionEnd”</em>:<code class="highlighter-rouge">3000</code></li>
        <li>
          <p><em>“Additional Column of Score”</em>:<code class="highlighter-rouge">8</code></p>

          <p>Position of column of score optional value if it is required</p>
        </li>
      </ul>
    </li>
    <li><strong>Cut</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the previous output adjusting the following parameters to cut “gene_name” column from table of annotated peaks and then get a list of geenes
      <ul>
        <li><em>“Cut columns”</em>: <code class="highlighter-rouge">c16</code></li>
        <li><em>“Delimited by”</em>: <code class="highlighter-rouge">Tab</code></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: output of <strong>chipeakanno annopeaks</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><strong>Remove beginning</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> of <code class="highlighter-rouge">Gene List</code> with the following parameters
      <ul>
        <li><em>“Remove first”</em>: <code class="highlighter-rouge">1</code></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“from”</em>: output of <strong>Cut</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li>Run  <strong>clusterProfiler bitr</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the previous output adjusting the following parameters to convert the list of genes to list of entrez ID
      <ul>
        <li><em>“Input Type Gene ID”</em>: <code class="highlighter-rouge">SYMBOL</code></li>
        <li><em>“Output Type Gene ID”</em>: <code class="highlighter-rouge">ENTREZID</code></li>
      </ul>
    </li>
    <li>Use the output of the  clusterProfiler bitr <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to run a GO Enrichment Analysis using <strong>clusterProfiler go</strong></li>
  </ol>
</blockquote>

<figure id="figure-5"><img src="../../images/funcann.jpg" alt="Functional annotations" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Results of GO enrichments analysis for DMPs</figcaption></figure>

<table>
  <thead>
    <tr>
      <th>ID</th>
      <th>Description</th>
      <th>pvalue</th>
      <th>qvalue</th>
      <th>geneID</th>
      <th>Count</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>GO:0048732</td>
      <td>gland development</td>
      <td>1.38E-58</td>
      <td>4.23E-55</td>
      <td>PTGS2 / KCNC1 / FZD1 /SLC22A18 /SLC22A3 (…)</td>
      <td>372</td>
    </tr>
    <tr>
      <td>GO:1901652</td>
      <td>response to peptide</td>
      <td>3.99E-57</td>
      <td>8.13E-54</td>
      <td>SULF1/ LAMA5/ MED1 /CFLAR/ MSX2 (…)</td>
      <td>359</td>
    </tr>
    <tr>
      <td>GO:0048545</td>
      <td>response to steroid hormone</td>
      <td>1.38EE-54</td>
      <td>2.11E-51</td>
      <td>HDAC9/ RAB10/ CFLAR/ WDTC1 (…)</td>
      <td>394</td>
    </tr>
  </tbody>
</table>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>Epigenetic aberrations which involve DNA modifications give researchers an interest to identify novel non-genetic factors responsible for complex human phenotypes such as height, weight, and disease. To identify methylation changes researchers need to perform complicated and  time consuming computational analysis. Here, the EWAS suite becomes a solution for this inconvenience and provides a simplified downstream analysis available as ready to run pipline in supplementary materials.</p>




                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>EWAS analyse changes in dna methylation in cohorts of different individuals effected by diseases</li>
                            
                            <li>DNA methylation changes can be triggered by environment (i.e. sun exposure, diet) and effect the way how our cells works and in long term cause cancer</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Hugo2015">Hugo, W., H. Shi, L. Sun, M. Piva, C. Song <i>et al.</i>, 2015 <b>Non-genomic and Immune Evolution of Melanoma Acquiring MAPKi Resistance</b>. Cell 162: 1271–1285.</span>
    
    
    <a href="https://doi.org/10.1016/j.cell.2015.07.061">10.1016/j.cell.2015.07.061</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li></ol>
                    

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

        <hr>
        <br>

        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=EWAS data analysis of 450k data (Epigenetics)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=EWAS data analysis of 450k data (Epigenetics)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Katarzyna Murat, Krzysztof Poterlowicz)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/tutorials/ewas-suite/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
