<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Proteogenomics 1: Database Creation</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:title" content="Galaxy Training: Proteogenomics 1: Database Creation" />
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Proteomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbcreation%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbcreation%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbcreation%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbcreation%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbcreation%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Proteogenomics 1: Database Creation",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "proteomics / proteogenomics-dbcreation / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Proteogenomics 1: Database Creation' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "Proteogenomics 1: Database Creation",
    "description": "Proteogenomics 1: Database Creation",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html",
  "timeRequired": "PT30M",
  "description": "Proteogenomics 1: Database Creation\\nThe questions this  addresses are:\n - How to create a customized Protein Database from RNAseq data?\n\n\\nThe objectives are:\n - Generating a customized Protein sequence database with variants, indels, splice junctions and known sequences.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.1489208",
      "name": "Training data for Proteogenomics 1: Database Creation tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "James Johnson"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "James Johnson"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://galaxyproject.github.io//training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Proteogenomics 1: Database Creation</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#subinamehta" class="contributor-badge"><img src="https://avatars.githubusercontent.com/subinamehta" alt="Subina Mehta">Subina Mehta</a>, <a href="/training-material/hall-of-fame#timothygriffin" class="contributor-badge"><img src="https://avatars.githubusercontent.com/timothygriffin" alt="Timothy J. Griffin">Timothy J. Griffin</a>, <a href="/training-material/hall-of-fame#pratikdjagtap" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pratikdjagtap" alt="Pratik Jagtap">Pratik Jagtap</a>, <a href="/training-material/hall-of-fame#jraysajulga" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jraysajulga" alt="Ray Sajulga">Ray Sajulga</a>, <a href="/training-material/hall-of-fame#jj-umn" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jj-umn" alt="James Johnson">James Johnson</a>, <a href="/training-material/hall-of-fame#pravs3683" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pravs3683" alt="Praveen Kumar">Praveen Kumar</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>How to create a customized Protein Database from RNAseq data?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Generating a customized Protein sequence database with variants, indels, splice junctions and known sequences.</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 30 minutes</p>
            

            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    

                    
                    <div>
                        
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1489208">
    <i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>


                    </div>
                    

                    
                    <div>
                        
    <a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/proteogenomics-dbcreation/workflows/" title="Workflows" alt="Proteogenomics 1: Database Creation workflows">
        <i class="fa fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>


                    </div>
                    
                    


                    
                    
                    <div>
                        

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">instances</span> Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fa fa-ship" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>



                    </div>
                    
                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p><strong>Proteogenomics</strong> involves the use of mass spectrometry (MS) based proteomics data against genomics and transcriptomics data to identify peptides and to understand protein-level evidence of gene expression. In the first section of the tutorial, we will create a protein database (FASTA) using RNA-sequencing files (FASTQ) and then perform sequence database searching using the resulting FASTA file with the MS data to identify peptides corresponding to novel proteoforms. Then, we will assign the genomic coordinates and annotations for these identified peptides and visualize the data for its spectral quality and genomic localization</p>

<p><img src="../../images/potential_novel_publication.png" alt="Novel Peptides" /></p>

<p>Proteogenomics integrates <strong>RNA-Seq</strong> data for generating customized protein sequence databases with mass spectrometry-based proteomics data, which are matched to these databases to identify protein sequence variants. (Cancer Res. (2017); 77(21):e43-e46. doi: <a target="_blank" href="https://doi.org/10.1158/0008-5472.CAN-17-0331">10.1158/0008-5472.CAN-17-0331</a>).</p>

<p><img src="../../images/workflow_objective1.png" alt="Proteogenomics" /></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>
<ol id="markdown-toc">
  <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
  <li><a href="#aligning-fastq-files-to-the-human-genome" id="markdown-toc-aligning-fastq-files-to-the-human-genome">Aligning FASTQ files to the human genome</a></li>
  <li><a href="#variant-analysis" id="markdown-toc-variant-analysis">Variant Analysis</a>    <ol>
      <li><a href="#variant-calling" id="markdown-toc-variant-calling">Variant Calling</a></li>
      <li><a href="#variant-annotation-and-genome-mapping" id="markdown-toc-variant-annotation-and-genome-mapping">Variant annotation and Genome Mapping</a></li>
    </ol>
  </li>
  <li><a href="#transcript-assembly" id="markdown-toc-transcript-assembly">Transcript Assembly</a>    <ol>
      <li><a href="#rna-seq-to-transcripts" id="markdown-toc-rna-seq-to-transcripts">RNA-seq to transcripts</a></li>
      <li><a href="#evaluate-the-assembly-with-annotated-transcripts" id="markdown-toc-evaluate-the-assembly-with-annotated-transcripts">Evaluate the assembly with annotated transcripts</a></li>
      <li><a href="#translate-transcripts" id="markdown-toc-translate-transcripts">Translate transcripts</a></li>
    </ol>
  </li>
  <li><a href="#creating-fasta-databases" id="markdown-toc-creating-fasta-databases">Creating FASTA Databases</a>    <ol>
      <li><a href="#genomic-mapping-database" id="markdown-toc-genomic-mapping-database">Genomic mapping database</a></li>
      <li><a href="#variant-annotations-database" id="markdown-toc-variant-annotations-database">Variant Annotations database</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 class="no_toc" id="overview">Overview</h1>

<p>This tutorial focuses on creating a <strong>FASTA</strong> database generated from RNA-seq data. There are two outputs from this workflow: (1) a <strong>sequence database</strong> consisting of variants and known reference sequences and (2) mapping files containing <strong>genomic</strong> and <strong>variant</strong> mapping data.</p>

<p><img src="../../images/database_creation.png" alt="Database creation" /></p>

<p>The first part of the tutorial deals with creating the FASTA file containing sequences with single amino acid variants (SAVs), insertions and deletions (indels). The second part of the workflow helps in creating a FASTA file with transcript assemblies (splicing variants).</p>

<h1 id="data-upload">Data upload</h1>

<p>In this tutorial, protein and the total RNA sample was obtained from the early development of B-cells from mice. It was obtained at two developmental stages of B-cells: <em>Ebf1</em> -/- pre-pro-B and <em>Rag2</em> -/- pro-B. Please refer to the original study  <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4276347/">Heydarian, M. et al.</a>for details.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-and-organization"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: data upload and organization</h3>

  <ol>
    <li>
      <p>Create a new history and name it something meaningful (e.g. <em>Proteogenomics DB creation</em>)</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>Import the FASTQ file and the GTF file from Zenodo <a href="https://doi.org/10.5281/zenodo.1489208"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.1489208.svg" alt="DOI" /></a>
      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1489208/files/FASTQ_ProB_22LIST.fastqsanger
https://zenodo.org/record/1489208/files/Mus_musculus.GRCm38.86.gtf
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the Galaxy Upload Manager (<i class="fa fa fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, Galaxy uses the URL as the name, so rename the files with a more useful name.</p>
      </blockquote>
    </li>
    <li>Rename the datasets with more descriptive names (strip off the url prefixes)</li>
  </ol>

</blockquote>

<h1 id="aligning-fastq-files-to-the-human-genome">Aligning FASTQ files to the human genome</h1>

<p>The first tool in the workflow is the <a href="http://ccb.jhu.edu/software/hisat"><strong>HISAT2</strong></a> alignment tool. It maps next-generation sequence (NGS) reads to the reference genome. This tool requires an RNA-seq file (.FASTQ) and a reference genome file in Gene transfer format (GTF). This .gtf file is obtained from the Ensembl database. When successful, this tool outputs a .bam file (binary version of a SAM: <strong>S</strong>equence <strong>A</strong>lignment/<strong>M</strong>ap).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-alignment-with-hisat2"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Alignment with HISAT2</h3>

  <ol>
    <li><strong>HISAT2</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Source for the reference genome”</em>: <code class="highlighter-rouge">Use a built-in genome</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Select a reference genome”</em>: <code class="highlighter-rouge">mm10</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Single-end or paired-end reads”</em>: <code class="highlighter-rouge">Single end</code></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input FASTQ files”</em>: <code class="highlighter-rouge">FASTQ_ProB_22LIST.fastqsanger</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Specify strand information”</em>: <code class="highlighter-rouge">Unstranded</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-note-on-strandedness"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note on strandedness</h3>
    <p>Note that if your reads are from a stranded library, you need to choose the appropriate
setting for “Specify strand information” mentioned above. For single-end reads, use <code class="highlighter-rouge">F</code> or <code class="highlighter-rouge">R</code>.
<code class="highlighter-rouge">F</code> means that a read corresponds to a forward transcript. <code class="highlighter-rouge">R</code> means that a read corresponds to the reverse
complemented counterpart of a transcript.</p>

    <p>For paired-end reads, use either <code class="highlighter-rouge">FR</code> or <code class="highlighter-rouge">RF</code>. With this
option being used, every read alignment will have an XS attribute tag: <code class="highlighter-rouge">+</code> means a read
belongs to a transcript on the positive <code class="highlighter-rouge">+</code> strand of the genome. <code class="highlighter-rouge">-</code> means a read belongs to a
transcript on the negative <code class="highlighter-rouge">-</code> strand of the genome. (TopHat has a similar option, <code class="highlighter-rouge">--library--type</code> option,
where fr - first strand corresponds to R and RF; fr - second strand corresponds to F and FR.)</p>
  </blockquote>

</blockquote>

<h1 id="variant-analysis">Variant Analysis</h1>

<h2 id="variant-calling">Variant Calling</h2>

<p><a href="https://github.com/ekg/freebayes">FreeBayes</a> is a Bayesian genetic variant detector designed to find small polymorphisms, specifically SNPs (single-nucleotide polymorphisms), indels (insertions and deletions), MNPs (multi-nucleotide polymorphisms), and complex events (composite insertion and substitution events) smaller than the length of a short-read sequencing alignment.</p>

<p><img src="../../images/variant_calling.png" alt="Variant calling" /></p>

<p>Provided with some BAM dataset(s) and a reference sequence, FreeBayes will generate
a VCF dataset describing SNPs, indels, and complex variants in samples in the input
alignments. By default, FreeBayes will consider variants supported by at least two observations
in a single sample (<code class="highlighter-rouge">-C</code>) and also by at least 20% of the reads from a single sample (<code class="highlighter-rouge">-F</code>). These
settings are suitable for low to high depth sequencing in haploid and diploid samples, but
users working with polyploid or pooled samples may adjust them depending on the characteristics
of their sequencing data.</p>

<p>FreeBayes is capable of calling variant haplotypes shorter than a read length where multiple
polymorphisms segregate on the same read. The maximum distance between polymorphisms phased in
this way is determined by the <code class="highlighter-rouge">--max-complex-gap</code>, which defaults to 3bp. In practice, this can
comfortably be set to half the read length. Ploidy may be set to any level (<code class="highlighter-rouge">-p</code>), but by default
all samples are assumed to be diploid. FreeBayes can model per-sample and per-region variation
in copy-number (<code class="highlighter-rouge">-A</code>) using a copy-number variation map.</p>

<p>FreeBayes can act as a frequency-based pooled caller and can describe variants and haplotypes in
terms of observation frequency rather than called genotypes. To do so, use –pooled-continuous
and set input filters to a suitable level. Allele observation counts will be described by AO
and RO fields in the VCF output.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-variant-calling-with-freebayes"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Variant Calling with FreeBayes</h3>

  <ol>
    <li><strong>FreeBayes</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Choose the source for the reference genome”</em>: <code class="highlighter-rouge">Locally cached file</code>
          <ul>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Run in batch mode?”</em>: <code class="highlighter-rouge">Run Individually</code></li>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span>  <em>“BAM dataset:</em>: <code class="highlighter-rouge">HISAT_Output.BAM</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span>  <em>“Using reference genome”</em>: <code class="highlighter-rouge">mm10</code></li>
          </ul>
        </li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Limit variant calling to a set of regions?”</em>: <code class="highlighter-rouge">Do not Limit</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Choose parameter selection level”</em>: <code class="highlighter-rouge">Simple diploid calling</code></li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong> and inspect the resulting files</li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-freebayes-options"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> FreeBayes options</h3>

  <p>Galaxy allows five levels of control over FreeBayes options, provided by the Choose
   parameter selection level menu option. These are:</p>
  <ol>
    <li>
      <p>Simple diploid calling: The simplest possible FreeBayes application. Equivalent to using
   FreeBayes with only a BAM input and no other parameter options.</p>
    </li>
    <li>
      <p>Simple diploid calling with filtering and coverage: Same as #1 plus two additional options:
    (1) <code class="highlighter-rouge">-0</code> (standard filters: <code class="highlighter-rouge">--min-mapping-quality 30 --min-base-quality 20 --min-supporting-allele
    -qsum 0 --genotype-variant-threshold 0</code>) and (2) <code class="highlighter-rouge">--min-coverage</code>.</p>
    </li>
    <li>
      <p>Frequency-based pooled calling: This is equivalent to using FreeBayes with the following
   options: <code class="highlighter-rouge">--haplotype-length 0 --min-alternate-count 1 --min-alternate-fraction 0 --pooled
   -continuous  --report- monomorphic</code>. This is the best choice for calling variants in mixtures
   such as viral, bacterial, or organellar genomes.</p>
    </li>
    <li>
      <p>Frequency-based pooled calling with filtering and coverage: Same as #3 but adds <code class="highlighter-rouge">-0</code> and
   <code class="highlighter-rouge">--min-coverage</code> like in #2.</p>
    </li>
  </ol>

  <p>Complete list of all options: Gives you full control by exposing all FreeBayes options as Galaxy parameters.</p>
</blockquote>

<h2 id="variant-annotation-and-genome-mapping">Variant annotation and Genome Mapping</h2>

<p><a href="http://dx.doi.org/10.1093/bioinformatics/btt543">CustomProDB</a> generates custom protein FASTA sequences files from exosome or transcriptome data. Once Freebayes creates the .vcf file, CustomProDB uses this file to generate a custom protein FASTA file from the transcriptome data. For this tool, we use Ensembl 89 mmusculus (GRm38.p5) (dbsnp142) as the genome annotation. We create three FASTA files from CustomProDB: (1) a variant FASTA file for short indels, (2) a Single Amino acid Variant (SAV) FASTA file, an SQLite database file (genome mapping and variant mapping) for mapping proteins to a genome and (3) an RData file for variant protein coding sequences.
The reference protein set can be filtered by transcript expression level (RPKM calculated from a BAM file), and variant protein forms can be predicted based on variant calls (SNPs and indels reported in a VCF file).</p>

<p>Annotations CustomProDB depends on a set of annotation files (in RData format) to
create reference and variant protein sequences. Galaxy administrators can use the CustomProDB
data manager to create these annotations to make them available for users.</p>

<p><img src="../../images/CustomProDB.png" alt="customproDB" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generate-protein-fastas-from-exosome-or-transcriptome-data"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generate protein FASTAs from exosome or transcriptome data</h3>

  <ol>
    <li><strong>CustomProDB</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span>  <em>“Will you select a genome annotation from your history or use a built-in annotation?”</em>: <code class="highlighter-rouge">Use built in genome annotation</code>
          <ul>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span>  <em>“Using reference genome”</em>: <code class="highlighter-rouge">Ensemble 89 mmusculus (GRm38.p5) (dbsnp142)</code></li>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BAM file”</em>: <code class="highlighter-rouge">HISAT_Output.BAM</code></li>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“VCF file”</em>: <code class="highlighter-rouge">Freebayes.vcf</code></li>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Annotate SNPs with rsid from dbSNP”</em>: <code class="highlighter-rouge">No</code></li>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Annotate somatic SNPs from COSMIC (human only)”</em>: <code class="highlighter-rouge">No</code></li>
          </ul>
        </li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Transcript Expression Cutoff (RPKM)”</em>: <code class="highlighter-rouge">1</code></li>
        <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Create a variant FASTA for short insertions and deletions”</em>: <code class="highlighter-rouge">Yes</code></li>
        <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Create SQLite files for mapping proteins to genome and summarizing variant proteins”</em>: <code class="highlighter-rouge">Yes</code></li>
        <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Create RData file of variant protein coding sequences”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Three FASTA files are generated through the CustomProDB tool:</p>
<ul>
  <li>a variant FASTA file for short indels,</li>
  <li>a Single Amino acid Variant (SAV) FASTA file</li>
  <li>an Sqlite file (genome mapping and variant mapping)</li>
</ul>

<p>for mapping proteins to genome and an RData file for variant protein coding sequences.
Similar to the genomic mapping, a variant mapping file is also generated from CustomProDB.
This SQLite file is also converted to tabular format and made SearchGUI-compatible. This
variant annotation file will be used to visualize the variants in the Multi-omics Visualization
Platform (in-house visualization platform developed by Galaxy-P senior developers).</p>

<h1 id="transcript-assembly">Transcript Assembly</h1>

<h2 id="rna-seq-to-transcripts">RNA-seq to transcripts</h2>

<p><a href="http://ccb.jhu.edu/software/stringtie/">StringTie</a> is a fast and highly efficient assembler of RNA-Seq alignments into potential transcripts. It uses a network flow algorithm as well as an optional <em>de novo</em> assembly step to assemble and quantitate full-length transcripts representing multiple splice variants for each gene locus.</p>

<p>Its input can include not only the alignments of raw reads used by other transcript assemblers, but also alignments of longer sequences that have been assembled from those reads. To identify differentially expressed genes between experiments, StringTie’s output can be processed by specialized software like Ballgown, Cuffdiff or other programs (DESeq2, edgeR, etc.).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-transcript-assembly-with-stringtie"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Transcript assembly with StringTie</h3>

  <ol>
    <li><strong>StringTie</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input mapped reads”</em>: <code class="highlighter-rouge">FASTQ_ProB_22LIST.BAM</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Specify strand information”</em>: <code class="highlighter-rouge">Unstranded</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Use a reference file to guide assembly?”</em>: <code class="highlighter-rouge">Use Reference GTF/GFF3</code>
          <ul>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Reference file”</em>: <code class="highlighter-rouge">Use file from History</code></li>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GTF/GFF3 dataset to guide assembly”</em>: <code class="highlighter-rouge">Mus_musculus.GRCm38.86.gtf</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Use Reference transcripts only?”</em>: <code class="highlighter-rouge">No</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Output files for differential expression?”</em>: <code class="highlighter-rouge">No additional output</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>StringTie accepts a BAM (or SAM) file of paired-end RNA-seq reads, which must be
sorted by genomic location (coordinate position). This file contains spliced read alignments
and can be produced directly by programs such as HISAT2. We recommend using HISAT2 as it is a
fast and accurate alignment program. Every spliced read alignment (i.e. an alignment across
at least one junction) in the input BAM file must contain the tag XS to indicate the genomic
strand that produced the RNA from which the read was sequenced. Alignments produced by HISAT2
(when run with the <code class="highlighter-rouge">--dta</code> option) already include this tag, but if you use a different read mapper
you should check that this XS tag is included for spliced alignments.</p>

<blockquote class="details">
  <h3 id="details-note-about-parameter-settings"><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Note about parameter settings</h3>
  <p>Be sure to run HISAT2 with the <code class="highlighter-rouge">--dta</code> option for alignment (under ‘Spliced alignment options’),
for the best results.</p>

  <p>Also note that if your reads are from a stranded library, you need to choose the appropriate
setting for ‘Specify strand information’ above. As, if Forward (FR) is selected, StringTie will
assume the reads are from a <code class="highlighter-rouge">--fr</code> library, while if Reverse (RF) is selected, StringTie will
assume the reads are from a <code class="highlighter-rouge">--rf</code> library, otherwise it is assumed that the reads are from an
unstranded library (the widely-used, although now deprecated, TopHat had a similar <code class="highlighter-rouge">--library-type</code>
option, where fr-firststrand corresponded to RF and fr-secondstrand corresponded to FR).
If you don’t know whether your reads are from a stranded library or not, you could use the
tool ‘RSeQC Infer Experiment’ to try to determine.</p>

  <p>As an option, a reference annotation file in GTF/GFF3 format can be provided to StringTie. In
this case, StringTie will prefer to use these “known” genes from the annotation file, and for
the ones that are expressed it will compute coverage, TPM and FPKM values. It will also produce
additional transcripts to account for RNA-seq data that aren’t covered by (or explained by) the
annotation. Note that if option <code class="highlighter-rouge">-e</code> is not used, the reference transcripts need to be fully covered
by reads in order to be included in StringTie’s output. In that case, other transcripts assembled
from the data by StringTie and not present in the reference file will be printed as well.</p>

  <p>We highly recommend that you provide annotation if you are analyzing a genome that is well
annotated, such as human, mouse, or other model organisms.</p>
</blockquote>

<h2 id="evaluate-the-assembly-with-annotated-transcripts">Evaluate the assembly with annotated transcripts</h2>

<p><a href="https://ccb.jhu.edu/software/stringtie/gffcompare.shtml">GffCompare</a> compares and evaluates the accuracy of RNA-Seq transcript assemblers (Cufflinks, Stringtie). * collapse (merge) duplicate transcripts from multiple GTF/GFF3 files (e.g. resulted from assembly of different samples) * classify transcripts from one or multiple GTF/GFF3 files as they relate to reference transcripts provided in a annotation file (also in GTF/GFF3 format)</p>

<p>The original form of this program is also distributed as part of the Cufflinks suite, under the name <a href="http://cole-trapnell-lab.github.io/cufflinks/cuffcompare/">CuffCompare</a>. Most of the options and parameters of CuffCompare are supported by GffCompare, while new features will likely be added to GffCompare in the future.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-compare-assembled-transcripts-against-a-reference-annotation"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: compare assembled transcripts against a reference annotation</h3>

  <ol>
    <li><strong>GffCompare</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GTF inputs for comparison”</em><code class="highlighter-rouge">Stringtie_outut.gtf</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Use Reference Annotation”</em>: <code class="highlighter-rouge">yes</code>
          <ul>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference Annotation”</em>: <code class="highlighter-rouge">Mus_musculus.GRCm38.86.gtf</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Ignore reference transcripts that are not overlapped by any input transfrags”</em>: <code class="highlighter-rouge">No</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Ignore input transcripts that are not overlapped by any reference transcripts”</em>: <code class="highlighter-rouge">No</code></li>
          </ul>
        </li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Use Sequence Data”</em>: <code class="highlighter-rouge">No</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“discard (ignore) single-exon transcripts”</em>: <code class="highlighter-rouge">No</code></li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Max. Distance for assessing exon accuracy”</em>: <code class="highlighter-rouge">100</code></li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Max distance for transcript grouping”</em>: <code class="highlighter-rouge">100</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“discard intron-redundant transfrags sharing 5’“</em>: <code class="highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="details">
  <h3 id="details-gffcompare-vs-cuffcompare"><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> GffCompare vs CuffCompare</h3>
  <p>A notable difference between GffCompare and CuffCompare is that when a single query GTF/GFF file is
given as input along with a reference annotation (<code class="highlighter-rouge">-r</code> option), GFFCompare switches into “annotation mode”
and it generates a <code class="highlighter-rouge">.annotated.gtf</code> file instead of the <code class="highlighter-rouge">.merged.gtf</code> produced by CuffCompare with the same
parameters. This file has the same general format as CuffCompare’s .merged.gtf file (with “class
codes” assigned to transcripts as per their relationship with the matching/overlapping reference,
transcript) but the original transcript IDs are preserved, so GffCompare can thus be used as a
simple way of annotating a set of transcripts.</p>

  <p>Another important difference is that the input transcripts are no longer discarded when they
are found to be “intron redundant”, i.e. contained within other, longer isoforms. CuffCompare
had the <code class="highlighter-rouge">-G</code> option to prevent collapsing of such intron redundant isoforms into their longer
“containers”, but GffCompare has made this the default mode of operation (hence the <code class="highlighter-rouge">-G</code> option is
no longer needed and is simply ignored when given).</p>

</blockquote>

<h2 id="translate-transcripts">Translate transcripts</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-translate-transcripts-to-protein-sequence"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Translate transcripts to protein sequence</h3>

  <p>First, we must convert the GffCompare annotated GTF file to BED format.</p>

  <ol>
    <li><strong>Convert gffCompare annotated GTF to BED</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GTF annotated by gffCompare”</em>: <code class="highlighter-rouge">output from gff compare</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“filter GffCompare class_codes to convert”</em>:
          <ul>
            <li><code class="highlighter-rouge">j : Potentially novel isoform (fragment): at least one splice junction is shared with a reference transcript</code></li>
            <li><code class="highlighter-rouge">e : Single exon transfrag overlapping a reference exon and at least 10 bp of a reference intron, indicating a possible   pre-mRNA fragment.</code></li>
            <li><code class="highlighter-rouge">i : A transfrag falling entirely within a reference intron</code></li>
            <li><code class="highlighter-rouge">p : Possible polymerase run-on fragment (within 2Kbases of a reference transcript)</code></li>
            <li><code class="highlighter-rouge">u : Unknown, intergenic transcript</code></li>
          </ul>
        </li>
      </ul>

      <p>Next, we translate transcripts from the input BED file into protein sequences.</p>
    </li>
    <li><strong>Translate BED transcripts cDNA in 3frames or CDS</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“A BED file with 12 columns”</em>: <code class="highlighter-rouge">Convert GffCompare-annotated GTF to BED</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Source for Genomic Sequence Data”</em>: <code class="highlighter-rouge">Locally cached File</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Select reference 2bit file”</em>: <code class="highlighter-rouge">mm10</code></li>
      </ul>

      <p>Finally, we convert a BED format file of the proteins from a proteomics search database into a tabular format for the Multiomics Visualization Platform (MVP).</p>
    </li>
    <li><strong>bed to protein map</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“A BED file with 12 columns, thickStart and thickEnd define protein coding region”</em>: <code class="highlighter-rouge">Translate cDNA_minus_CDS</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-note-on-data-formats"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note on data formats</h3>
  <ul>
    <li>The tabular output can be converted to a sqlite database using the Query_Tabular tool.</li>
    <li>The sqlite table should be named: feature_cds_map.</li>
    <li>The names for the columns should be: name, chrom, start, end, strand, cds_start, cds_end</li>
    <li>This SQL query will return the genomic location for a peptide sequence in a protein
(multiply the amino acid position by 3 for the cds location)</li>
  </ul>
</blockquote>

<h1 id="creating-fasta-databases">Creating FASTA Databases</h1>

<p>In this section we will perform the following tasks:</p>
<ul>
  <li>The <strong>Protein Database Downloader</strong> tool is used to download the FASTA database from UniProt and cRAP (<strong>c</strong>ommon <strong>R</strong>epository of <strong>A</strong>dventitious <strong>P</strong>roteins) database containing known/reference mouse proteins.</li>
  <li>The <strong>Regex Text Manipulation</strong> tool is used to manipulate the FASTA file to make it SearchGUI-compatible.</li>
  <li>The <strong>FASTA Merge Files and Filter Unique Sequences</strong> tool is used to merge the databases obtained from CustomProDB and “Translate BED” tool
along with the UniProt and cRAP databases.</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on</h3>

  <ol>
    <li><strong>FASTA Merge Files and Filter Unique Sequences</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Run in batch mode?”</em>: <code class="highlighter-rouge">Merge individual FASTAs (output collection if input is collection)</code></li>
        <li><i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input FASTA File(s)”</em> : <code class="highlighter-rouge">Input Custom ProDB Fasta File output</code>
          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 1.HISAT_Output.rpkm
 2.HISAT_Output.snv
 3.HISAT_Output.indel
</code></pre></div>          </div>
        </li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“How are sequences judged to be unique?”</em>: <code class="highlighter-rouge">Accession and Sequence</code></li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Accession Parsing Regular Expression”</em>: <code class="highlighter-rouge">^&gt;([^ |]+).*$</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-tool-parameters-explained"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Tool parameters explained</h3>
  <p>This tool concatenates FASTA database files together.</p>
  <ul>
    <li>If the uniqueness criterion is “Accession and Sequence”, only the first appearence of each unique sequence will appear in the output. Otherwise, duplicate sequences are allowed, but only the first appearance of each accession will appear in the output.</li>
    <li>The default accession parser will treat everything in the header before the first space as the accession.</li>
  </ul>
</blockquote>

<p><img src="../../images/Fasta_sequence.png" alt="Fasta sequence" /></p>

<p>For visualization purposes we also use the concatenate tool to concatenate the genomic mapping with the protein mapping dataset. This output will be used for visualization in MVP to view the genomic coordinates of the variant peptide.</p>

<p>An SQLite database containing the genomic mapping SQLite, variant annotation and information from the protein mapping file is concatenated to form a single genomic mapping SQLite database later used as an input for the ‘Peptide Genomic Coordinate’ tool. For that we need to follow the steps below:</p>

<h2 id="genomic-mapping-database">Genomic mapping database</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-database-for-genomic-mapping"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create Database for genomic mapping</h3>

  <ol>
    <li><strong>SQLite to tabular for SQL query</strong> <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“SQLite Database”</em>: <code class="highlighter-rouge">genomic_mapping.sqlite</code> from CustomProDB</li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Query”</em>:
          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  SELECT pro_name, chr_name, cds_chr_start - 1, cds_chr_end,strand,cds_start - 1, cds_end
  FROM genomic_mapping
  ORDER BY pro_name, cds_start, cds_end
</code></pre></div>          </div>
        </li>
      </ul>

      <p>The output is further processed so that the results are compatible with the Multiomics Visualization Platform.</p>
    </li>
    <li><strong>Column Regex Find And Replace</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select cells from”</em>: <code class="highlighter-rouge">genomic_mapping_sqlite' (tabular)</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Using”</em>: <code class="highlighter-rouge">column 1</code></li>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Check</strong>
          <ul>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em> : <code class="highlighter-rouge">^(ENS[^_]+_\d+:)([ACGTacgt]+)&gt;([ACGTacgt]+)\s*</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="highlighter-rouge">\1\2_\3</code></li>
          </ul>
        </li>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Check</strong>
          <ul>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>:  <code class="highlighter-rouge">,([A-Z]\d+[A-Z])\s*</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="highlighter-rouge">.\1</code></li>
          </ul>
        </li>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Check</strong>
          <ul>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="highlighter-rouge">^(ENS[^ |]*)\s*</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="highlighter-rouge">\1</code></li>
          </ul>
        </li>
      </ul>

      <p>This tool goes line by line through the specified input file and if the text in the selected column matches a specified regular expression pattern, it replaces the text with the specified replacement.</p>

      <p>Next, we will concatenate the output from this tool with the “Bed to protein map” output.</p>
    </li>
    <li><strong>Concatenate multiple datasets</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Concatenate Datasets”</em>: Select the output from the previous tool and the <code class="highlighter-rouge">Bed2protein_SJ_SAV_INDEL</code> output.</li>
      </ul>

      <p>Output will be the “Genomic_Protein_map”</p>

      <p>Now, we load this tabular datasets into an SQLite database.</p>
    </li>
    <li><strong>Query Tabular using SQLite</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong>
          <ul>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular Dataset for Table”</em>: <code class="highlighter-rouge">Genomic_Protein_map</code></li>
            <li>Section <strong>Table Options</strong>:
              <ul>
                <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Specify Name for Table”</em>: <code class="highlighter-rouge">feature_cds_map</code></li>
                <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Specify Column Names (comma-separated list)”</em>: <code class="highlighter-rouge">name,chrom,start,end,strand,cds_start,cds_end</code></li>
                <li><em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">No</code></li>
                <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Table Index</strong>:
                  <ul>
                    <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“This is a unique index”</em>: <code class="highlighter-rouge">No</code></li>
                    <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Index on columns”</em>: <code class="highlighter-rouge">name,cds_start,cds_end</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <strong>“genomic_mapping_sqlite”</strong></p>

      <p><img src="../../images/genomic_mapping_file.png" alt="genomic mapping" /></p>
    </li>
  </ol>

</blockquote>

<h2 id="variant-annotations-database">Variant Annotations database</h2>

<p>We will repeat the process for the variant annotations</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-database-for-variant-annotations"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create database for variant annotations</h3>

  <ol>
    <li><strong>SQLite to tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“SQLite Database”</em>: <code class="highlighter-rouge">variant_annotations.sqlite</code> from CustomProDB</li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Query”</em>:
          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT var_pro_name,pro_name,cigar,annotation
FROM variant_annotation
</code></pre></div>          </div>
        </li>
      </ul>

      <p>We will subject the output to text manipulation so that the results are compatible with the Multiomics Visualization Platform.</p>
    </li>
    <li><strong>Column Regex Find And Replace</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select cells from”</em>: <code class="highlighter-rouge">variant_annotations_sqlite' (tabular)</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Using:”</em> <code class="highlighter-rouge">column 1</code></li>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Check</strong>
          <ul>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="highlighter-rouge">^(ENS[^_]+_\d+:)([ACGTacgt]+)&gt;([ACGTacgt]+)\s*</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="highlighter-rouge">\1\2_\3</code></li>
          </ul>
        </li>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Check</strong>:
          <ul>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="highlighter-rouge">,([A-Z]\d+[A-Z])\s*</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="highlighter-rouge">.\1</code></li>
          </ul>
        </li>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Check</strong>:
          <ul>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="highlighter-rouge">^(ENS[^ |]*)\s*</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>:  <code class="highlighter-rouge">\1</code></li>
          </ul>
        </li>
      </ul>

      <p>Next, we will load this tabular dataset into an SQLite database.</p>
    </li>
    <li><strong>Query Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong>
          <ul>
            <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular Dataset for Table”</em>: <code class="highlighter-rouge">variant_annotation</code></li>
            <li>Section <strong>Table Options</strong>:
              <ul>
                <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Specify Name for Table”</em>: <code class="highlighter-rouge">variant_annotation</code></li>
                <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Specify Column Names (comma-separated list)”</em>: <code class="highlighter-rouge">name,reference,cigar,annotation</code></li>
                <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">No</code></li>
              </ul>
            </li>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Table Index</strong>
              <ul>
                <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“This is a unique index”</em>: <code class="highlighter-rouge">No</code></li>
                <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Index on columns”</em>: <code class="highlighter-rouge">name,cigar</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <strong>“Variant_annotation_sqlitedb”</strong></li>
  </ol>

</blockquote>

<p>These SQLite databases, which contain the genomic mapping SQLite and variant
annotation information from the protein mapping file, will be utilized
by MVP to visualize the genomic loci of any variant peptides.</p>

<p><img src="../../images/viewing_SNP_Variant_IGV.png" alt="SNP variant" /></p>




                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>Generating variant protein database</li>
                            
                            <li>Generating genomic and variant mapping files for visualization</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                    

                    

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
        <blockquote class="agenda follow-up">
            <strong class="follow-up"><i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
            <ul>
                
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Proteogenomics 3: Novel peptide analysis:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
        </blockquote>
        

        <hr>
        <br>

        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Proteogenomics 1: Database Creation (Proteomics)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=Proteogenomics 1: Database Creation (Proteomics)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
