<!DOCTYPE html>
<html>









  <head>
    <meta charset="utf-8">
    <title>Tool development and integration into Galaxy</title>
    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-45719423-18' , 'auto');
        ga('send', 'pageview');
    </script>
    
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css" id="theme">
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

    
    
    
    
    <meta name="description" content="Galaxy is an open-source project. Everyone can contribute..." />
    <meta property="og:title" content="Galaxy Training: Tool development and integration into Galaxy" />
    <meta property="og:description" content="Galaxy is an open-source project. Everyone can contribute..." />
    <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    <script type="application/ld+json">
      


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Tool development and integration into Galaxy",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "dev / tool-integration / slides",
  "learningResourceType": "slides",
  "name": "Slides for 'Tool development and integration into Galaxy' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "Tool development and integration into Galaxy",
    "description": "Tool development and integration into Galaxy",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/dev/tutorials/tool-integration/slides.html",
  "timeRequired": "PT6H",
  "description": "Tool development and integration into Galaxy\\nThe questions this  addresses are:\n - What is a tool for Galaxy?\n - How to build a tool/wrapper with the good practices?\n - How to deal with the tool environment?\n\n\\nThe objectives are:\n - Discover what is a wrapper and its structure\n - Use the Planemo utilities to develop a good wrapper\n - Deal with the dependencies\n - Write functional tests\n - Make a tool ready for publishing in a ToolShed\n\n",
  "hasPart": [

  ],
  "mentions": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    },
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Anthony Bretaudeau"
    },
    {
      "@type": "Person",
      "name": "John Chilton"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Björn Grüning"
    },
    {
      "@type": "Person",
      "name": "Gildas Le Corguillé"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    },
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Anthony Bretaudeau"
    },
    {
      "@type": "Person",
      "name": "John Chilton"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Björn Grüning"
    },
    {
      "@type": "Person",
      "name": "Gildas Le Corguillé"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Development in Galaxy",
      "description": "Galaxy is an open-source project. Everyone can contribute to its development with core Galaxy development, integration of softwares in Galaxy environment, ...",
      "url": "https://galaxyproject.github.io//training-material/topics/dev/"
    }
  ]
}
    </script>
  </head>
  <body>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-header"><span>
<a href="/training-material/topics/dev" title="Return to topic page" ><i class="fa fa-level-up" aria-hidden="true"></i></a>

<a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/dev/tutorials/tool-integration/slides.html"><i class="fa fa-pencil" aria-hidden="true"></i></a>

</span></div>

<div class="my-footer"><span>

<img src="/training-material/assets/images/GTN-60px.png" alt="Galaxy Training Network" style="height: 40px;"/>

</span></div>

---

# Tool development and integration into Galaxy



.footnote[Tip: press `P` to view the presenter notes]

???
Presenter notes contain extra information which might be useful if you
intend to use these slides for teaching.

Press `P` again to switch presenter notes off

---





### &lt;i class=&quot;fa fa-question-circle&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;question&lt;/span&gt; Questions


- What is a tool for Galaxy?

- How to build a tool/wrapper with the good practices?

- How to deal with the tool environment?


---


### &lt;i class=&quot;fa fa-bullseye&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;objectives&lt;/span&gt; Objectives


- Discover what is a wrapper and its structure

- Use the Planemo utilities to develop a good wrapper

- Deal with the dependencies

- Write functional tests

- Make a tool ready for publishing in a ToolShed


---


## A tool in Galaxy

---

### A tool in Galaxy

![](../../images/galaxy_instance_detailed_screenshot.png)

---

### Galaxy tool / wrapper

![](../../images/graphlan_screenshot.png)

```bash
graphlan.py --format &quot;png&quot; --size 7 input_tree.txt png_image.png
```

---

### Wrapper

- Description of the user interface
- Link between the underlying tool and the Galaxy instance
- How to invoke the tool
- Which files and options to pass
- Which files the tool will produce as output

---

### Wrapper

![](../../images/wrapper_layers.png)

???

1. `&lt;inputs&gt;` (datasets and parameters) specified in the tool XML are exposed in the Galaxy tool UI
2. When the user fills the form and click the `Execute` button, Galaxy fills the `&lt;command&gt;` template in the XML with the inputs entered by the user and execute the Cheetah code, producing a script as output
3. Galaxy creates a job for the generated script and executes it somewhere (bowtie is run in this case)
4. Some (not necessarily all) output files become new history datasets, as specified in the `&lt;outputs&gt;` XML tag set

---

### Wrapper

![](../../images/wrapper_big_picture_1.png)

???

- CDATA tags are used to prevent the interpretation of ampersands and less-than signs as XML special characters
- The tool name and description are combined in the left panel (tool menu), keep them short!

---

### Wrapper

![](../../images/wrapper_big_picture_2.png)

---

### Wrapper

![](../../images/wrapper_big_picture_3.png)

---

### Wrapper

Galaxy tool XML format is formally defined in a XML Schema Definition (XSD), used to generate the relative [online documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html)

---

### `command`
#### How to invoke the tool?

```xml
&lt;command&gt;&lt;![CDATA[
graphlan.py
--format $format
#if $dpi
    --dpi $dpi
#end if
--size $size
'$input_tree'
]]&gt;&lt;/command&gt;
```

&lt;small&gt;If the script is provided with the wrapper xml&lt;/small&gt;
```xml
&lt;command&gt;&lt;![CDATA[
python '$__tool_directory__/graphlan.py'
# command params...
]]&gt;&lt;/command&gt;
```

???

- In the first case, `graphlan.py` is expected to be on the PATH when the job executes
- In the second case, `$__tool_directory__` is a special variable which is substituted by Galaxy with the directory where the tool XML is
- The `#if ... #end if` syntax comes from the [Cheetah](https://cheetahtemplate.org/) template language, which has a Python-like syntax

---

### `command`
#### How to pass the parameters?

![](../../images/graphlan_screenshot.png)

---

### `inputs` &gt; `param` to `command`
#### How to pass the parameters?

```xml
&lt;inputs&gt;
    &lt;param name=&quot;input_tree&quot; type=&quot;data&quot; label=&quot;...&quot;/&gt;

    &lt;param argument=&quot;--dpi&quot; type=&quot;integer&quot; optional=&quot;true&quot; label=&quot;...&quot;
        help=&quot;For non vectorial formats&quot; /&gt;
&lt;/inputs&gt;
```

???

- `name` identifies a parameter
- Parameters can have different types (`data`, `data_collection`, `integer`, `float`, `text`, `select`, `boolean`, `color`, `data_column`,...)
- The content `argument` is automatically added at the end of the displayed help
- When `argument` is specified and `name` is not, `name` is derived from `argument` by removing the initial dashes

---

### `inputs` &gt; `param` to `command`
#### Directly linked to `&lt;command&gt;`

```xml
&lt;command&gt;&lt;![CDATA[
graphlan.py
...
#if $dpi
    --dpi $dpi
#end if
'$input_tree'
...
]]&gt;&lt;/command&gt;
```

---

### `inputs` &gt; `param` &gt; `data`

![](../../images/input_data.png)

```xml
&lt;param name=&quot;...&quot; type=&quot;data&quot; format=&quot;txt&quot; label=&quot;...&quot; help=&quot;...&quot; /&gt;
```

.footnote[[List of possible formats](https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample)]

---

### `inputs` &gt; `param` &gt; `integer` | `float`

![](../../images/input_integer.png)

```xml
&lt;param name=&quot;...&quot; type=&quot;integer&quot; value=&quot;7&quot; label=&quot;...&quot; help=&quot;...&quot;/&gt;
```

![](../../images/input_integer_range.png)

```xml
&lt;param name=&quot;...&quot; type=&quot;float&quot; min=&quot;0&quot; max=&quot;10&quot; value=&quot;1&quot; label=&quot;...&quot;
    help=&quot;...&quot;/&gt;
```

???

- In the first case, Galaxy creates a text box which accepts only integer values
- In the second case, since *both* `min` and `max` are specified, a slider is shown in addition

---

### `inputs` &gt; `param` &gt; `text`

![](../../images/input_text.png)

```xml
&lt;param name=&quot;...&quot; type=&quot;text&quot; value=&quot;...&quot; label=&quot;...&quot; help=&quot;...&quot;/&gt;
```

---

### `inputs` &gt; `param` &gt; `select`

![](../../images/input_select.png)

```xml
&lt;param name=&quot;...&quot; type=&quot;select&quot; label=&quot;...&quot; help=&quot;...&quot;&gt;
    &lt;option value=&quot;png&quot; selected=&quot;true&quot;&gt;PNG&lt;/option&gt;
    &lt;option value=&quot;pdf&quot;&gt;PDF&lt;/option&gt;
    &lt;option value=&quot;ps&quot;&gt;PS&lt;/option&gt;
    &lt;option value=&quot;eps&quot;&gt;EPS&lt;/option&gt;
    &lt;option value=&quot;svg&quot;&gt;SVG&lt;/option&gt;
&lt;/param&gt;
```

If no `option` has `selected=&quot;true&quot;`, the first one is selected by default.

---

### `inputs` &gt; `param` &gt; `select`

![](../../images/input_select_checkboxes.png)

```xml
&lt;param name=&quot;...&quot; type=&quot;select&quot; display=&quot;radio&quot; label=&quot;...&quot; help=&quot;...&quot;&gt;
    &lt;option value=&quot;min&quot; selected=&quot;true&quot;&gt;Minimum&lt;/option&gt;
    &lt;option value=&quot;mean&quot;&gt;Mean&lt;/option&gt;
    &lt;option value=&quot;max&quot;&gt;Max&lt;/option&gt;
    &lt;option value=&quot;sum&quot;&gt;Sum&lt;/option&gt;
&lt;/param&gt;
```

---

### `inputs` &gt; `param` &gt; `select`

![](../../images/input_select_multiple.png)

```xml
&lt;param name=&quot;...&quot; type=&quot;select&quot; multiple=&quot;true&quot; label=&quot;...&quot; help=&quot;...&quot;&gt;
    &lt;option value=&quot;ld&quot; selected=&quot;true&quot;&gt;Length distribution&lt;/option&gt;
    &lt;option value=&quot;gc&quot; selected=&quot;true&quot;&gt;GC content distribution&lt;/option&gt;
    ...
&lt;/param&gt;
```

---

### `inputs` &gt; `param` &gt; `boolean`

![](../../images/input_boolean.png)

```xml
&lt;param name=&quot;...&quot; type=&quot;boolean&quot; checked=&quot;false&quot; truevalue=&quot;--log&quot; falsevalue=&quot;&quot;
    label=&quot;...&quot; help=&quot;...&quot; /&gt;
```

---

class: packed

### `inputs` &gt; `param` &gt; `conditional`

.image-75[ ![](../../images/input_conditional.png) ]

```xml
&lt;command&gt;&lt;![CDATA[
#if $fastq_input.selector == 'paired':
    '$fastq_input.input1' '$fastq_input.input2'
#else:
    '$fastq_input.input'
#end if
]]&gt;&lt;/command&gt;
&lt;inputs&gt;
    &lt;conditional name=&quot;fastq_input&quot;&gt;
        &lt;param name=&quot;selector&quot; type=&quot;select&quot; label=&quot;Single or paired-end reads?&quot;&gt;
            &lt;option value=&quot;paired&quot;&gt;Paired-end&lt;/option&gt;
            &lt;option value=&quot;single&quot;&gt;Single-end&lt;/option&gt;
        &lt;/param&gt;
        &lt;when value=&quot;paired&quot;&gt;
            &lt;param name=&quot;input1&quot; type=&quot;data&quot; format=&quot;fastq&quot; label=&quot;Forward reads&quot; /&gt;
            &lt;param name=&quot;input2&quot; type=&quot;data&quot; format=&quot;fastq&quot; label=&quot;Reverse reads&quot; /&gt;
        &lt;/when&gt;
        &lt;when value=&quot;single&quot;&gt;
            &lt;param name=&quot;input&quot; type=&quot;data&quot; format=&quot;fastq&quot; label=&quot;Single reads&quot; /&gt;
        &lt;/when&gt;
    &lt;/conditional&gt;
&lt;/inputs&gt;
```

---

class: packed

### `inputs` &gt; `param` &gt; `repeat`

![](../../images/input_repeat.png)

```xml
&lt;command&gt;&lt;![CDATA[
#for $i, $s in enumerate( $series )
    rank_of_series=$i
    input_path=${s.input}
    x_column=${s.xcol}
#end for
]]&gt;&lt;/command&gt;

&lt;inputs&gt;
    &lt;repeat name=&quot;series&quot; title=&quot;Series&quot;&gt;
        &lt;param name=&quot;input&quot; type=&quot;data&quot; format=&quot;tabular&quot; label=&quot;Dataset&quot;/&gt;
        &lt;param name=&quot;xcol&quot; type=&quot;data_column&quot; data_ref=&quot;input&quot; label=&quot;Column for x axis&quot;/&gt;
    &lt;/repeat&gt;
&lt;/inputs&gt;
```

???

It makes sense to use a `&lt;repeat&gt;` block only if it contains multiple related parameters, otherwise adding `multiple=&quot;true&quot;` is preferable.

---

### `outputs`
#### Which files the tool will produce as output?

![](../../images/output.png)

---

### `outputs`

```xml
&lt;outputs&gt;
    &lt;data name=&quot;tree&quot; format=&quot;txt&quot; label=&quot;${tool.name} on ${on_string}: Tree&quot; /&gt;
    &lt;data name=&quot;annotation&quot; format=&quot;txt&quot;
        label=&quot;${tool.name} on ${on_string}: Annotation&quot; /&gt;
&lt;/outputs&gt;
```

???

`${tool.name} on ${on_string}` is the default output label, need to modify this if the tool generates more than 1 output

---

### `outputs` &gt; `filter`
#### Conditional

```xml
&lt;inputs&gt;
    &lt;param type=&quot;select&quot; name=&quot;format&quot; label=&quot;Output format&quot;&gt;
        &lt;option value=&quot;png&quot;&gt;PNG&lt;/option&gt;
        &lt;option value=&quot;pdf&quot;&gt;PDF&lt;/option&gt;
    &lt;/param&gt;
&lt;/inputs&gt;
&lt;outputs&gt;
    &lt;data name=&quot;png_output&quot; format=&quot;png&quot; label=&quot;${tool.name} on ${on_string}: PNG&quot;&gt;
        &lt;filter&gt;format == &quot;png&quot;&lt;/filter&gt;
    &lt;/data&gt;
    &lt;data name=&quot;pdf_output&quot; format=&quot;pdf&quot; label=&quot;${tool.name} on ${on_string}: PDF&quot;&gt;
        &lt;filter&gt;format == &quot;pdf&quot;&lt;/filter&gt;
    &lt;/data&gt;
&lt;/outputs&gt;
```

???

N.B. If the filter expression raises an Exception, the dataset will NOT be filtered out

---

### `stdio`

Error if tool exit code is not 0 (default):
```xml
&lt;command detect_errors=&quot;exit_code&quot;&gt; ... &lt;/command&gt;
```
Error if tool exit code is not 0 or either &quot;Exception:&quot; or &quot;Error:&quot; appears in standard error/output:
```xml
&lt;command detect_errors=&quot;aggressive&quot;&gt; ... &lt;/command&gt;
```
If you need more precision:
```xml
&lt;stdio&gt;
    &lt;exit_code range=&quot;:-2&quot; level=&quot;warning&quot; description=&quot;Low disk space&quot; /&gt;
    &lt;exit_code range=&quot;1:&quot; level=&quot;fatal&quot;  /&gt;
    &lt;regex match=&quot;Error:&quot;  level=&quot;fatal&quot; /&gt;
&lt;/stdio&gt;
&lt;command&gt; ... &lt;/command&gt;
```

&lt;small&gt;&quot;Warning&quot; level allows to add information to `stderr` without marking the dataset as failed&lt;/small&gt;

---

### `help`

![](../../images/help.png)

```xml
&lt;help&gt;&lt;![CDATA[
**What it does**
GraPhlAn is a software tool for producing high-quality circular
representations of taxonomic and phylogenetic trees. GraPhlAn focuses
on concise, integrative, informative, and publication-ready
representations of phylogenetically- and taxonomically-driven
investigation.

For more information, check the `user manual
&lt; https://bitbucket.org/nsegata/graphlan/overview &gt;`_.
]]&gt;&lt;/help&gt;
```

.footnote[[reStructuredText markup format](http://docutils.sourceforge.net/rst.html)]

---

### `citations`

![](../../images/citations.png)

```xml
&lt;citations&gt;
    &lt;citation type=&quot;doi&quot;&gt;10.1093/bioinformatics/bts611&lt;/citation&gt;
    &lt;citation type=&quot;doi&quot;&gt;10.1093/nar/gks1219&lt;/citation&gt;
    &lt;citation type=&quot;doi&quot;&gt;10.1093/nar/gks1005&lt;/citation&gt;
    &lt;citation type=&quot;doi&quot;&gt;10.1093/bioinformatics/btq461&lt;/citation&gt;
    &lt;citation type=&quot;doi&quot;&gt;10.1038/nbt.2198&lt;/citation&gt;
&lt;/citations&gt;
```

&lt;small&gt;If no DOI is available, a BibTeX citation can be specified with `type=&quot;bibtex&quot;`&lt;/small&gt;

---

### Tips and best practices

Always quote `text` and `data` parameters and output `data`

```xml
&lt;command&gt;&lt;![CDATA[
graphlan.py
...
'$input_tree'
'$png_output_image'
]]&gt;&lt;/command&gt;
```

- For security reasons
- Paths may contain spaces
- Prefer single quotes over double quotes

---

### Tips and best practices

Use `&amp;&amp;` to concatenate multiple commands

```xml
&lt;command&gt;&lt;![CDATA[
graphlan.py
--format '$format'
&amp;&amp;
echo &quot;Yeah it worked!&quot;
]]&gt;&lt;/command&gt;
```

&lt;small&gt;The job will exit on the first error encountered.&lt;/small&gt;

---

### Tips and best practices

Use the `argument` tag when a param name reflects the command line argument

```xml
&lt;param argument=&quot;--size&quot; type=&quot;integer&quot; value=&quot;7&quot; label=&quot;...&quot; help=&quot;...&quot;/&gt;
```

- It will be printed at the end of the param help section
- No need to write `name=&quot;size&quot;`, it is implicit

---

### Tips and best practices

Use sections to group related parameters

![](../../images/input_section.png)

```xml
&lt;section name=&quot;advanced&quot; title=&quot;Advanced options&quot; expanded=&quot;False&quot;&gt;
    &lt;param argument=&quot;--size&quot; type=&quot;integer&quot; value=&quot;7&quot; label=&quot;...&quot; help=&quot;...&quot;/&gt;
    ...
&lt;/section&gt;
```

---

![](../../images/planemo-logo.png)

&gt; Command-line utilities to assist in building and publishing Galaxy tools.

- [Documentation](https://planemo.readthedocs.io/en/latest/)
- [Tutorial](https://planemo.readthedocs.io/en/latest/writing_standalone.html)

---

###.image-25[![](../../images/planemo-logo.png)]

![](../../images/big_picture.png)

---

###.image-25[![](../../images/planemo-logo.png)]

#### `planemo tool_init`

Creates a skeleton of xml file

```bash
$ mkdir new_tool
$ cd new_tool
$ planemo tool_init --id 'some_short_id' --name 'My super tool'
```

Complicated version:

```bash
$ planemo tool_init --id 'samtools_sort' --name 'Samtools sort' \
          --description 'order of storing aligned sequences' \
          --requirement 'samtools@1.3.1' \
          --example_command &quot;samtools sort -o '1_sorted.bam' '1.bam'&quot; \
          --example_input 1.bam \
          --example_output 1_sorted.bam \
          --test_case \
          --version_command 'samtools --version | head -1' \
          --help_from_command 'samtools sort' \
          --doi '10.1093/bioinformatics/btp352'
```

---

class: packed

###.image-25[![](../../images/planemo-logo.png)]

#### `planemo lint`

Checks the syntax of a wrapper

```bash
$ planemo lint
```

```bash
Linting tool /opt/galaxy/tools/seqtk_seq.xml
Applying linter tests... CHECK
.. CHECK: 1 test(s) found.
Applying linter output... CHECK
.. INFO: 1 outputs found.
Applying linter inputs... CHECK
.. INFO: Found 1 input parameters.
Applying linter help... CHECK
.. CHECK: Tool contains help section.
.. CHECK: Help contains valid reStructuredText.
Applying linter general... CHECK
.. CHECK: Tool defines a version [0.1.0].
.. CHECK: Tool defines a name [Convert to FASTA (seqtk)].
.. CHECK: Tool defines an id [seqtk_seq].
Applying linter command... CHECK
.. INFO: Tool contains a command.
Applying linter citations... CHECK
.. CHECK: Found 1 likely valid citations.
```

---

###.image-25[![](../../images/planemo-logo.png)]

#### `planemo serve`

View your new tool in a local Galaxy instance

```bash
$ planemo serve
```

Open http://127.0.0.1:9090/ in your web browser to view your new tool

---

###.image-25[![](../../images/planemo-logo.png)]

#### Building Galaxy Tools

![](../../images/planemo_tool_building_lint.png)

---

## Functional tests

---

### Functional tests

- Functional testing is a quality assurance (QA) process.
- The tests comfort developers and users that the tools can run across different servers/architectures. And that the latest modifications don't break older features.
- Tools are tested by feeding them inputs and parameters and verifying the outputs (typically a diff)

---

### Functional tests
#### Good practices

1. Build tests using inputs/parameters/outputs
2. Run the tests --&gt; Failed --&gt; Coffee
3. Develop
4. Run the tests --&gt; Failed --&gt; Coffee
5. Develop
6. Run the tests --&gt; Succeed --&gt; Beer

---

### `tests`

```xml
&lt;tests&gt;
    &lt;test&gt;
        &lt;param name=&quot;input_tree&quot; value=&quot;input_tree.txt&quot;/&gt;
        &lt;param name=&quot;format&quot; value=&quot;png&quot;/&gt;
        &lt;param name=&quot;dpi&quot; value=&quot;100&quot;/&gt;
        &lt;param name=&quot;size&quot; value=&quot;7&quot;/&gt;
        &lt;param name=&quot;pad&quot; value=&quot;2&quot;/&gt;
        &lt;output name=&quot;png_output_image&quot; file=&quot;png_image.png&quot; /&gt;
    &lt;/test&gt;
&lt;/tests&gt;
```

&lt;small&gt;`input_tree.txt` and `png_image.png` must be in the `test-data/` directory&lt;/small&gt;

---

### Wrapper directory tree

```
graphlan/
├── graphlan.xml
├── graphlan.py
└── test-data/
    ├── input_tree.txt
    └── png_image.png

```

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
&lt;output name=&quot;out_file1&quot; file=&quot;cf_maf2fasta_concat.dat&quot; ftype=&quot;fasta&quot; /&gt;
```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test)]

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
&lt;output ... compare=&quot;diff|re_match|sim_size|contains|re_match_multiline&quot; ... /&gt;
```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output)]

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
&lt;output ... compare=&quot;diff&quot; lines_diff=&quot;4&quot; /&gt;
```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output)]

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
&lt;output ... compare=&quot;sim_size&quot; delta=&quot;1000&quot; /&gt;
```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output)]

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
&lt;output ... md5=&quot;68b329da9893e34099c7d8ad5cb9c940&quot; /&gt;
```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output)]

---

### `tests`
#### Comparison Methods: checking the content

```xml
&lt;output name=&quot;out_file1&quot;&gt;
    &lt;assert_contents&gt;
        &lt;has_text text=&quot;chr7&quot; /&gt;
        &lt;not_has_text text=&quot;chr8&quot; /&gt;
        &lt;has_text_matching expression=&quot;1274\d+53&quot; /&gt;
        &lt;has_line_matching expression=&quot;.*\s+127489808\s+127494553&quot; /&gt;
        &lt;!-- &amp;#009; is XML escape code for tab --&gt;
        &lt;has_line line=&quot;chr7&amp;#009;127471195&amp;#009;127489808&quot; /&gt;
        &lt;has_n_columns n=&quot;3&quot; /&gt;
    &lt;/assert_contents&gt;
&lt;/output&gt;
```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output-assert-contents)]

---

### `tests`
#### Comparison Methods: checking the stdout/stderr

```xml
&lt;assert_stdout&gt;
    &lt;has_text text=&quot;Step 1... determine cutoff point&quot; /&gt;
    &lt;has_text text=&quot;Step 2... estimate parameters of null distribution&quot; /&gt;
&lt;/assert_stdout&gt;
```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output-assert-contents)]

---

### `tests`
#### Tests with conditionals, repeats, sections

```xml
&lt;tests&gt;
    &lt;test&gt;
        &lt;section name=&quot;advanced&quot;&gt;
            &lt;repeat name=&quot;names&quot;&gt;
                &lt;param name=&quot;first&quot; value=&quot;Abraham&quot;/&gt;
                &lt;param name=&quot;last&quot; value=&quot;Lincoln&quot;/&gt;
            &lt;/repeat&gt;
            &lt;repeat name=&quot;names&quot;&gt;
                &lt;param name=&quot;first&quot; value=&quot;Donald&quot;/&gt;
                &lt;param name=&quot;last&quot; value=&quot;Trump&quot;/&gt;
            &lt;/repeat&gt;
            &lt;conditional name=&quot;image&quot;&gt;
                &lt;param name=&quot;output_image&quot; value=&quot;yes&quot;/&gt;
                &lt;param name=&quot;format&quot; value=&quot;png&quot;/&gt;
            &lt;/conditional&gt;
        &lt;/section&gt;
        ...
    &lt;/test&gt;
&lt;/tests&gt;
```

---

###.image-25[![](../../images/planemo-logo.png)]

#### `planemo test`

Runs all functional tests

```bash
$ planemo test
```

An HTML report is automatically created with logs in case of failing test

---

###.image-25[![](../../images/planemo-logo.png)]

#### Test Galaxy Tools

![](../../images/planemo_tool_building_test.png)

---

## Dependencies

---

### Dependencies
#### How Galaxy will deal with dependencies?

![](../../images/galaxy_instance.png)

---

### Dependencies
#### Embedded tool

```
graphlan/
├── graphlan.xml
└── graphlan.py

```

```xml
&lt;command&gt;&lt;![CDATA[
$__tool_directory__/graphlan.py
...
]]&gt;&lt;/command&gt;
```

---

### Dependencies
#### External tool

Definition of requirements in `.xml` file

```xml
&lt;requirements&gt;
    &lt;requirement type=&quot;package&quot; version=&quot;1.66&quot;&gt;biopython&lt;/requirement&gt;
    &lt;requirement type=&quot;package&quot; version=&quot;1.0.0&quot;&gt;graphlan&lt;/requirement&gt;
&lt;/requirements&gt;
```

Local installation using Conda packages

---

![](../../../../shared/images/conda_logo.png)

Package, dependency and environment management

---

.image-50[![](../../../../shared/images/conda_logo.png)]

- Based on recipes describing how to install the software which are then built for their distribution
- No compilation at installation: binaries with their dependencies, libraries...
- Not restricted to Galaxy

See [Tool Dependencies and Conda](../conda/slides.html)

---

![](../../images/planemo-logo.png)

&gt; Command-line utilities to assist in building and publishing Galaxy tools.

---

### &lt;i class=&quot;fa fa-pencil&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;hands_on&lt;/span&gt; Hands-on

![](../../images/exercise.png)

---

### &lt;i class=&quot;fa fa-pencil&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;hands_on&lt;/span&gt; Hands-on

#### The Goal

.left[ Developing a wrapper for Seqtk: ]

.left[ A tool for processing sequence data in FASTA and FASTQ files ]

---

### &lt;i class=&quot;fa fa-pencil&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;hands_on&lt;/span&gt; Hands-on

#### Steps

#### [http://planemo.readthedocs.io/en/latest/writing_standalone.html ](http://planemo.readthedocs.io/en/latest/writing_standalone.html)
1. [Launching the Appliance](http://planemo.readthedocs.io/en/latest/appliance.html#launching-the-appliance)
2. [The Basics](http://planemo.readthedocs.io/en/latest/writing_standalone.html#the-basics)
3. [Simple Parameters](http://planemo.readthedocs.io/en/latest/writing_standalone.html#simple-parameters)
4. [Conditional Parameters](http://planemo.readthedocs.io/en/latest/writing_standalone.html#conditional-parameters)

---

## Advanced features

---

### Other tags: `configfiles`

Like `command`, uses Cheetah code to write:

- A config file
- A whole script or a module
- Any text file needed for the tool execution

---

class: packed

### `configfiles`

```xml
&lt;command&gt;&lt;![CDATA[ mb $script_nexus ]]&gt;&lt;/command&gt;

&lt;configfiles&gt;
    &lt;configfile name=&quot;script_nexus&quot;&gt;&lt;![CDATA[
set autoclose = yes;
execute $input_data;
#if str( $data_type.type ) == &quot;nuc“
    lset nst=$data_type.lset_params.lset_Nst;
#end if
mcmcp ngen=$mcmcp_ngen;
mcmc;
quit
    ]]&gt;&lt;/configfile&gt;
&lt;/configfiles&gt;
```

```bash
set autoclose = yes;
execute dataset_42.dat;
lset nst=2 ;
mcmcp ngen=100000;
mcmc;
quit
```

---

### `macros`

![](../../images/macro.png)

.footnote[[Planemo documents about macros](https://planemo.readthedocs.io/en/latest/writing_advanced.html#macros-reusable-elements)]

---

class: packed

### `macros` &gt; `xml`

macros.xml

```xml
&lt;macros&gt;
    &lt;xml name=&quot;requirements&quot;&gt;
        &lt;requirements&gt;
            &lt;requirement type=&quot;package&quot; version=&quot;2.5.0&quot;&gt;blast&lt;/requirement&gt;
        &lt;/requirements&gt;
    &lt;/xml&gt;
    &lt;xml name=&quot;stdio&quot;&gt;
        &lt;stdio&gt;
            &lt;exit_code range=&quot;1&quot; level=&quot;fatal&quot; /&gt;
        &lt;/stdio&gt;
    &lt;/xml&gt;
&lt;/macros&gt;
```

ncbi_blastn_wrapper.xml

```xml
&lt;macros&gt;
    &lt;import&gt;macros.xml&lt;/import&gt;
&lt;/macros&gt;
&lt;expand macro=&quot;requirements&quot;/&gt;
&lt;expand macro=&quot;stdio&quot;/&gt;
```

---

### `macros` &gt; `token`

macros.xml

```xml
&lt;macros&gt;
    &lt;token name=&quot;@THREADS@&quot;&gt;-num_threads &quot;\${GALAXY_SLOTS:-8}&quot;&lt;/token&gt;
&lt;/macros&gt;
```

ncbi_blastn_wrapper.xml

```xml
&lt;command&gt;
blastn -query '$query' @THREADS@ [...]
&lt;/command&gt;
```

---

### `macros` &gt; `yield`

macros.xml

```xml
&lt;macros&gt;
    &lt;xml name=&quot;requirements&quot;&gt;
        &lt;requirements&gt;
            &lt;requirement type=&quot;package&quot; version=&quot;2.2.0&quot;&gt;trinity&lt;/requirement&gt;
            &lt;yield/&gt;
        &lt;/requirements&gt;
    &lt;/xml&gt;
&lt;/macros&gt;
```

trinity.xml

```xml
&lt;expand macro=&quot;requirements&quot;&gt;
    &lt;requirement type=&quot;package&quot; version=&quot;1.1.2&quot;&gt;bowtie&lt;/requirement&gt;
    &lt;requirement type=&quot;set_environment&quot;&gt;TRINITY_MEM_OPTIONS&lt;/requirement&gt;
&lt;/expand&gt;
```

---

### Tips and best practices

Use a `TOOL_VERSION` token

```xml
&lt;macros&gt;
    &lt;token name=&quot;@TOOL_VERSION@&quot;&gt;1.2&lt;/token&gt;
&lt;/macros&gt;
```

```xml
&lt;tool id=&quot;seqtk_seq&quot; name=&quot;Convert to FASTA (seqtk)&quot; version=&quot;@TOOL_VERSION@+galaxy3&quot;&gt;
    &lt;requirements&gt;
        &lt;requirement type=&quot;package&quot; version=&quot;@TOOL_VERSION@&quot;&gt;seqtk&lt;/requirement&gt;
    &lt;/requirements&gt;
```

This means: the 3rd revision of the Galaxy tool wrapper for Seqtk 1.2 .

---

### `command` &gt; Reserved Variables

```xml
&lt;command&gt;&lt;![CDATA[
# Email’s numeric ID (id column of galaxy_user table in the database)
echo $__user_id__

# User’s email address
echo $__user_email__

# The galaxy.app.UniverseApplication instance, gives access to all other configuration file variables.
# Should be used as a last resort, may go away in future releases.
echo $__app__.config.user_library_import_dir

# Check a dataset type
#if $input1.is_of_type('gff'):
    echo &quot;input1 type is ${input1.datatype}&quot;
#end if
]]&gt;&lt;/command&gt;
```
.footnote[[Reserved Variables List](https://docs.galaxyproject.org/en/latest/dev/schema.html#reserved-variables)]

---

### Multiple input

```xml
&lt;param name=&quot;...&quot; type=&quot;data&quot; format=&quot;txt&quot; label=&quot;...&quot; help=&quot;...&quot; /&gt;
```

![](../../images/input_data.png)

Possible to select multiple dataset:

![](../../images/input_data_multiple.png)

- Useful to launch the same tool on multiple datasets independently
- One job per dataset

---

### Multiple input

```xml
&lt;param name=&quot;...&quot; type=&quot;data&quot; format=&quot;txt&quot; multiple=&quot;true&quot; label=&quot;...&quot; help=&quot;...&quot; /&gt;
```

![](../../images/input_data_multiple2.png)

In the command:

```xml
&lt;command&gt;&lt;![CDATA[
...
#for $input in $inputs
    --input &quot;$input&quot;
#end for
]]&gt;&lt;/command&gt;
```

One job for all selected dataset

---

### Multiple output

```xml
&lt;outputs&gt;
    &lt;data name=&quot;output&quot; format=&quot;txt&quot;&gt;
        &lt;discover_datasets pattern=&quot;__designation_and_ext__&quot;
            directory=&quot;output_dir&quot; visible=&quot;true&quot; /&gt;
    &lt;/data&gt;
&lt;/outputs&gt;
```

- `__designation_and_ext__`: a predefined regexp,

- catches the dataset identifier + the datatype

If the output file extension is not present/usable:

```xml
&lt;outputs&gt;
    &lt;data name=&quot;output&quot; format=&quot;txt&quot;&gt;
        &lt;discover_datasets pattern=&quot;__designation__&quot; format=&quot;txt&quot;
            directory=&quot;output_dir&quot; visible=&quot;true&quot; /&gt;
    &lt;/data&gt;
&lt;/outputs&gt;
```

---

### Dataset collection

A dataset collection combines numerous datasets in a single entity that can be manipulated together

- `list`: a simple list of datasets
- `paired`: a pair of datasets, `forward` and `reverse` for NGS
- composite: e.g. `list:paired` for a list of dataset pairs

Usage

- Useful to launch a workflow on many samples
- Sample names are kept along the workflow: `element_identifier`
- Galaxy tools are available to manipulate collections

---

.center[### Dataset collection]

.center[Creation from history]

.pull-left[
![](../../images/historyWithCheckboxes.png)
]

.pull-right[
![](../../images/buildPairs.png)
]

---

### Dataset collection

Creation when uploading

![collectionPrefiltered](../../../galaxy-data-manipulation/images/collectionPrefiltered.png)

[Also possible with the rule-based uploader](/training-material/topics/galaxy-data-manipulation/tutorials/upload-rules/tutorial.html)

---

### Dataset collection &gt; input

```xml
&lt;param name=&quot;inputs&quot; type=&quot;data&quot; format=&quot;bam&quot; label=&quot;Input BAM(s)&quot; /&gt;
```

![](../../images/input_data_collection.png)

&lt;small&gt;Also accepted with `multiple=&quot;true&quot;`&lt;/small&gt;

Accept only collections:

```xml
&lt;param name=&quot;inputs&quot; type=&quot;data_collection&quot; collection_type=&quot;list|paired|list:paired|...&quot;
    format=&quot;bam&quot; label=&quot;Input BAM(s)&quot; /&gt;
```

```xml
&lt;command&gt;&lt;![CDATA[
...
#for $input in $inputs
    --input '$input'
    --sample_name '$input.element_identifier'
#end for
]]&gt;&lt;/command&gt;
```

---

### Dataset collection &gt; output

A single paired collection:

```xml
&lt;collection name=&quot;paired_output&quot; type=&quot;paired&quot; label=&quot;Split Pair&quot;&gt;
    &lt;data name=&quot;forward&quot; format=&quot;txt&quot; /&gt;
    &lt;data name=&quot;reverse&quot; format_source=&quot;input1&quot; from_work_dir=&quot;reverse.txt&quot; /&gt;
&lt;/collection&gt;
```

Unknown number of files:

```xml
&lt;collection name=&quot;output&quot; type=&quot;list&quot; label=&quot;Unknown number of files&quot;&gt;
    &lt;discover_datasets pattern=&quot;__name_and_ext__&quot; directory=&quot;outputs&quot; /&gt;
&lt;/collection&gt;
```


- `__name_and_ext__`: a predefined regexp,
- catches the dataset identifier + the datatype

---

## Some Config

---

### `job_conf.xml`

![](../../images/job_conf.xml_1.png)

8 is the default value if not set in destination

```bash
blastn -query foo_bar -num_threads 8
```

---

### `job_conf.xml`

![](../../images/job_conf.xml_2.png)

```bash
blastn -query foo_bar -num_threads 4
```

---

### Data tables

- They list all reference data used by tools
- e.g.: Blast databases, BWA indexes, Fasta files
- Stored in `.loc` files
- Populated by hand or using Data Managers
- Data Managers are dedicated kind of Galaxy tools

---

### Using a data table in a tool

![](../../images/tool_data_table_conf.xml.png)

---

### Datatypes

- Every Galaxy dataset is associated with a datatype
- Datatype can be detected or user specified
- Gain of usability

.footnote[[Documentation: Adding Datatypes](https://galaxyproject.org/admin/datatypes/adding-datatypes/)]

---

## Publishing tools

---

### Contributing to a community

Many tools developed by the community on GitHub repositories
- [Intergalactic Utilities Commission](https://github.com/galaxyproject/tools-iuc)
- [GalaxyP](https://github.com/galaxyproteomics/tools-galaxyp)
- ...

Added value:
- Easier development
- Easier contribution for user
- Avoid duplications of efforts
- Automated tests on each contribution
- Automated publishing to ToolShed
- Principle of many eyes: if something is visible to many people then, collectively, they are more likely to find errors in it
---

### IUC: Intergalactic Utilities Commission

.image-50[![IUC](../../images/iuc_logo.png)]

- A team maintaining high quality tools
- Establishing and following [best practices](https://galaxy-iuc-standards.readthedocs.io/) for tool development
- Open to contributions: bug fixes, new tools, ...

https://github.com/galaxyproject/tools-iuc

---

### How should I publish my tool?

Adding to an existing GitHub repository (IUC, GalaxyP, ...)

- Read the guidelines
- Open a pull request
- Respond to review comments

---

### How should I publish my tool?

Using your own GitHub repository

- Reasons: ownership, specific practices, exotic tools
- Follow the same structure
- Use `.travis.yml` to automate tests and publishing

---

### How should I publish my tool?

Using planemo by hand

- Ok for few tools
- Makes contributing harder

[Check out our tutorial to publish to the ToolShed using Planemo](/training-material/topics/dev/tutorials/toolshed/slides.html)

---

### Continuous Integration

.image-50[![](../../images/github_logo.png)]

.image-50[![](../../images/travis_ci_logo.png)]

.image-50[![](../../images/planemo-logo.png)]

.image-50[![](../../../../shared/images/conda_logo.png)]

---

### Continuous Integration

- Create a Pull Request on a GitHub repository
- Tests are automatically run on Travis
- Other contributors review your tool
- The Pull Request is accepted when all lights are green
- The tool is automatically uploaded to the ToolShed

---

### GitHub side

![](../../images/github_travis_github.png)

---

### Travis CI side

![](../../images/github_travis_travis.png)

---

### ToolShed

- Need to create a `.shed.yml` file in the tool directory of the GitHub repository:

```yml
categories: [Sequence Analysis]
description: Tandem Repeats Finder description
long_description: A long long description.
name: tandem_repeats_finder_2
owner: gandres
```

```bash
planemo shed_init --name=&quot;tandem_repeats_finder_2&quot;
                     --owner=&quot;gandres&quot;
                     --description=&quot;Tandem Repeats Finder description&quot;
                     --long_description=&quot;A long long description.&quot;
                     --category=&quot;Sequence Analysis&quot;
                     [--remote_repository_url=&lt;URL to .shed.yml on github&gt;]
                     [--homepage_url=&lt;Homepage for tool.&gt;]
```

---

### Tool suites

A tool suite is a group of related tools that can all be installed at once.

Defined in `.shed.yml`: implicitly define repositories for each individual tool in the directory and build a suite for those tools.

Example: `trinity/.shed.yml`

```yml
[...]
auto_tool_repositories:
    name_template: &quot;&quot;
    description_template: &quot; (from the Trinity tool suite)&quot;
suite:
    name: &quot;suite_trinity&quot;
    description: Trinity tools to assemble transcript sequences from Illumina RNA-Seq data.
```

---

### Check

```bash
planemo shed_lint --tools
```
```bash
Linting repository […]/tandem_repeats_finder
Applying linter expansion... CHECK
.. INFO: Included files all found.
Applying linter tool_dependencies_xsd... CHECK
.. INFO: tool_dependencies.xml found and appears to be valid XML
Applying linter tool_dependencies_actions... CHECK
.. INFO: Parsed tool dependencies.
Applying linter repository_dependencies... CHECK
.. INFO: No repository_dependencies.xml, skipping.
Applying linter shed_yaml... CHECK
.. INFO: .shed.yml found and appears to be valid YAML.
Applying linter readme... CHECK
.. INFO: No README found skipping.
+Linting tool […]/tandem_repeats_finder/tandem_repeats_finder_wrapper.xml
[…]
```

---

.image-50[![](../../../../shared/images/ansible_logo.png)]

.image-50[![](../../../../shared/images/conda_logo.png)]

.image-50[![](../../../../shared/images/docker_logo.png)]

---

### Docker Galaxy Flavors

![](../../images/galaxydocker_logo.png)

.footnote[[Docker Galaxy Stable](https://github.com/bgruening/docker-galaxy-stable)]

---

### Docker Galaxy Flavors
#### Dockerfile

```yaml
# Galaxy - My own flavour
# VERSION       0.1

FROM quay.io/bgruening/galaxy:16.07

MAINTAINER Björn A. Grüning, bjoern.gruening@gmail.com

ENV GALAXY_CONFIG_BRAND deepTools

# Adding the tool definitions to the container
ADD my_tool_list.yml $GALAXY_ROOT/my_tool_list.yml

# Install deepTools
RUN install-tools $GALAXY_ROOT/my_tool_list.yml
```

---

### Docker Galaxy Flavors
#### tool_list.yml

```yaml
api_key: admin
galaxy_instance: http://127.0.0.1:8080/
tools:
- name: fastqc
  owner: devteam
  tool_panel_section_id: cshl_library_information
  tool_shed_url: https://toolshed.g2.bx.psu.edu
  revisions:
    - '8c650f7f76e9'  # v0.62
    - 'd2cf2c0c8a11'  # v0.63
- name: suite_deeptools
  owner: bgruening
  tool_panel_section_label: 'deepTools'
  tool_shed_url: https://toolshed.g2.bx.psu.edu
  revisions:
    - '7f5562625ae2'
```

---

### Docker Galaxy Flavors
#### RUN

```bash
docker run -d -p 8080:80 galaxy-deeptools
```




---
### &lt;i class=&quot;fa fa-key&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;keypoints&lt;/span&gt; Key points


- Galaxy Tool Syntax

- Use Planemo

- Use Conda

- Use GitHub

- Use Travis

- No more excuse to develop crappy Galaxy tools





---
### &lt;i class=&quot;fa fa-graduation-cap&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;curriculum&lt;/span&gt; Do you want to extend your knowledge? 


Follow one of our recommended follow-up trainings:



    
        
        
            
        
- [Development in Galaxy](/training-material/topics/dev)
    - Tool Dependencies and Conda: [&lt;i class=&quot;fa fa-slideshare&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;slides&lt;/span&gt; slides](/training-material/topics/dev/tutorials/conda/slides.html) - [&lt;i class=&quot;fa fa-laptop&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;tutorial&lt;/span&gt; hands-on](http://planemo.readthedocs.io/en/latest/writing_standalone.html)
    





---

## Thank you!

This material is the result of a collaborative work. Thanks to the [Galaxy Training Network](https://wiki.galaxyproject.org/Teach/GTN) and all the contributors!


<div class="contributors-line">


<a href="/training-material/hall-of-fame#shiltemann" class="contributor-badge"><img src="https://avatars.githubusercontent.com/shiltemann" alt="Saskia Hiltemann">Saskia Hiltemann</a>, <a href="/training-material/hall-of-fame#bebatut" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bebatut" alt="Bérénice Batut">Bérénice Batut</a>, <a href="/training-material/hall-of-fame#abretaud" class="contributor-badge"><img src="https://avatars.githubusercontent.com/abretaud" alt="Anthony Bretaudeau">Anthony Bretaudeau</a>, <a href="/training-material/hall-of-fame#jmchilton" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jmchilton" alt="John Chilton">John Chilton</a>, <a href="/training-material/hall-of-fame#nsoranzo" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nsoranzo" alt="Nicola Soranzo">Nicola Soranzo</a>, <a href="/training-material/hall-of-fame#bgruening" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bgruening" alt="Björn Grüning">Björn Grüning</a>, <a href="/training-material/hall-of-fame#lecorguille" class="contributor-badge"><img src="https://avatars.githubusercontent.com/lecorguille" alt="Gildas Le Corguillé">Gildas Le Corguillé</a>


</div>



<img src="/training-material/assets/images/GTN.png" alt="Galaxy Training Network" style="height: 100px;"/>



    </textarea>
    <script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript">
    </script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
    </script>
  </body>
</html>
