<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy 101</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Galaxy is a scientific workflow, data integration, and da..." />
        <meta property="og:title" content="Galaxy Training: Galaxy 101" />
        <meta property="og:description" content="Galaxy is a scientific workflow, data integration, and da..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/introduction" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Introduction to Galaxy Analyses
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-101%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-101%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-101%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-101%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-101%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/introduction/tutorials/galaxy-intro-101/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Galaxy 101",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "introduction / galaxy-intro-101 / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Galaxy 101' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "Galaxy 101",
    "description": "Galaxy 101",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html",
  "timeRequired": "PT1H30M",
  "description": "Galaxy 101\\nThe questions this  addresses are:\n - Which coding exon has the highest number of single nucleotide polymorphisms (SNPs) on human chromosome 22?\n\n\\nThe objectives are:\n - Familiarize yourself with the basics of Galaxy\n - Learn how to obtain data from external sources\n - Learn how to run tools\n - Learn how histories work\n - Learn how to create a workflow\n - Learn how to share your work\n\n",
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "",
      "name": "Training data for Galaxy 101 tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Clemens Blank"
    },
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Björn Grüning"
    },
    {
      "@type": "Person",
      "name": "Anne Pajon"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Clemens Blank"
    },
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Björn Grüning"
    },
    {
      "@type": "Person",
      "name": "Anne Pajon"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Introduction to Galaxy Analyses",
      "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/"
    }
  ],
  "skillLevel": "Introductory"
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Galaxy 101</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#shiltemann" class="contributor-badge"><img src="https://avatars.githubusercontent.com/shiltemann" alt="Saskia Hiltemann">Saskia Hiltemann</a>, <a href="/training-material/hall-of-fame#nsoranzo" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nsoranzo" alt="Nicola Soranzo">Nicola Soranzo</a>, <a href="/training-material/hall-of-fame#blankclemens" class="contributor-badge"><img src="https://avatars.githubusercontent.com/blankclemens" alt="Clemens Blank">Clemens Blank</a>, <a href="/training-material/hall-of-fame#nekrut" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nekrut" alt="Anton Nekrutenko">Anton Nekrutenko</a>, <a href="/training-material/hall-of-fame#bgruening" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bgruening" alt="Björn Grüning">Björn Grüning</a>, <a href="/training-material/hall-of-fame#pajanne" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pajanne" alt="Anne Pajon">Anne Pajon</a>, <a href="/training-material/hall-of-fame#erasche" class="contributor-badge"><img src="https://avatars.githubusercontent.com/erasche" alt="Helena Rasche">Helena Rasche</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>Which coding exon has the highest number of single nucleotide polymorphisms (SNPs) on human chromosome 22?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Familiarize yourself with the basics of Galaxy</li>
            
            <li>Learn how to obtain data from external sources</li>
            
            <li>Learn how to run tools</li>
            
            <li>Learn how histories work</li>
            
            <li>Learn how to create a workflow</li>
            
            <li>Learn how to share your work</li>
            
            </ul>

            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour 30 minutes</p>
            

            
            <p><strong><i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> Level: </strong>
             Introductory 
<span class="level introductory" title="Introductory Tutorial">
  <i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span>
</span>

</p>
            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    
                    <div>
                        <a class="topic-icon" href="/training-material/topics/introduction/slides/introduction.html" title="Introduction slides">
                            <i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Introduction Slides
                        </a>
                    </div>
                    

                    

                    
                    


                    
                    
                    <div>
                        

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">instances</span> Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/introduction/docker" title="Docker image for this tutorial">
                <i class="fa fa-ship" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>



                    </div>
                    
                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>This practical aims to familiarize you with the Galaxy user interface. It will teach you how to perform basic tasks such as importing data, running tools, working with histories, creating workflows, and sharing your work.</p>

<blockquote class="comment">
  <h3 id="comment-note-results-may-vary"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note: results may vary</h3>

  <p>Your results may be slightly different from the ones presented in this tutorial
due to differing versions of tools, reference data, external databases, or
because of stochastic processes in the algorithms.</p>

</blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will:</p>

<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#locating-exons" id="markdown-toc-locating-exons">Locating Exons</a></li>
      <li><a href="#obtaining-snp-information" id="markdown-toc-obtaining-snp-information">Obtaining SNP information</a></li>
    </ol>
  </li>
  <li><a href="#analysis" id="markdown-toc-analysis">Analysis</a>    <ol>
      <li><a href="#find-exons-with-the-most-snps" id="markdown-toc-find-exons-with-the-most-snps">Find exons with the most SNPs</a></li>
      <li><a href="#count-the-number-of-snps-per-exon" id="markdown-toc-count-the-number-of-snps-per-exon">Count the number of SNPs per exon</a></li>
      <li><a href="#sort-the-exons-by-snps-count" id="markdown-toc-sort-the-exons-by-snps-count">Sort the exons by SNPs count</a></li>
      <li><a href="#select-the-top-five-exons" id="markdown-toc-select-the-top-five-exons">Select the top five exons</a></li>
      <li><a href="#recovering-exon-info" id="markdown-toc-recovering-exon-info">Recovering exon info</a></li>
      <li><a href="#displaying-data-in-ucsc-genome-browser" id="markdown-toc-displaying-data-in-ucsc-genome-browser">Displaying data in UCSC genome browser</a></li>
    </ol>
  </li>
  <li><a href="#galaxy-management" id="markdown-toc-galaxy-management">Galaxy management</a>    <ol>
      <li><a href="#convert-your-analysis-history-into-a-workflow" id="markdown-toc-convert-your-analysis-history-into-a-workflow">Convert your analysis history into a workflow</a></li>
      <li><a href="#the-workflow-editor" id="markdown-toc-the-workflow-editor">The workflow editor</a></li>
      <li><a href="#run-workflow-on-different-data" id="markdown-toc-run-workflow-on-different-data">Run workflow on different data</a></li>
      <li><a href="#share-your-work" id="markdown-toc-share-your-work">Share your work</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<p>We start with the question: In human chromosome 22, which exon has the highest number of single nucleotide polymorphisms (SNPs)?</p>

<blockquote class="comment">
  <h3 id="comment-background"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Background</h3>
  <p>Not everyone has the same background and that’s ok! You may have studied different organisms with different gene models, or you may not be familiar with the biological aspect at all. The biological background is not necessary for following this tutorial, we just like to provide research context for any tutorial. Here are brief explanations of some of the concepts mentioned in this tutorial:</p>

  <ul>
    <li><strong>Nucleotides</strong> are the A, C, T, and Gs that make up DNA.</li>
    <li>A <strong>Chromosomes</strong> can be thought of as a very long piece of DNA (string of A, C, T, Gs) Some organisms have chromosomes, in this tutorial we will use Human chromosome number 22.</li>
    <li><strong>Features</strong> are regions of the chromosome that are interesting for one reason or another. Some examples of features include genes, terminators, transcription start sites, and repeat regions.</li>
    <li><strong>Genes</strong> are one kind of interesting feature, a region that will be <a href="https://en.wikipedia.org/wiki/Transcription_(biology)">transcribed</a> into RNA before being <a href="https://en.wikipedia.org/wiki/Translation_(biology)">translated</a> into proteins.</li>
    <li>
      <p><strong>Exons</strong> are part of the <a href="https://en.wikipedia.org/wiki/Eukaryote">eukaryotic</a> gene <a href="https://en.wikipedia.org/wiki/Gene#Structure_and_function">model</a>. A gene is a complex structure with numerous segments that are removed before the the gene is transcribed into an RNA. The exons are the coding regions, the regions that will appear in the final protein product.</p>

      <figure id="figure-1"><img src="../../images/DNA_exons_introns.png" alt="Eukaryotic gene model" /><figcaption><span class="figcaption-prefix">Figure 1:</span> An original piece of DNA containing introns and exons has the introns cut out before the exons are joined together to form the mRNA. (Image from <a href="https://commons.wikimedia.org/wiki/File:DNA_exons_introns.gif">WikiMedia</a>, under the Public Domain)</figcaption></figure>
    </li>
    <li><strong>SNP</strong> stands for <a href="https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism">single-nucleotide polymorphism</a> and is pronounced like the English word “snip”. SNPs are single nucleotide differences between a sequenced individual compared to some reference sequence; where one individual might have an A, another could have a C in that position. Databases of SNPs have been created for many organisms and they include any single nucleotide deviation from the reference sequence which appears in some percentage of the population (e.g. &gt;1%). These are often especially interesting to geneticists as the causes of certain inheritable diseases.</li>
  </ul>
</blockquote>

<p>You may be familiar with the <a href="https://genome.ucsc.edu/">UCSC Genome Browser</a> or another resource like it, and know that you can find the data there.
But even with your data in hand, you stll have the question: “how do I actually compute this?” There is really a straightforward answer: <strong>Galaxy</strong>. So let’s try it…</p>

<p>Browse to your favourite <a href="https://galaxyproject.org/use/">Galaxy instance</a> and log in or register. The Galaxy interface consists of three main parts:</p>

<ol>
  <li>The available tools are listed on the left</li>
  <li>Your analysis history is recorded on the right</li>
  <li>The central panel will let you run analyses and view outputs</li>
</ol>

<p><img src="../../images/galaxy_interface.png" alt="Galaxy interface" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-history"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create history</h3>

  <ol>
    <li>
      <p>Make sure you start from an empty analysis history.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p><strong>Rename your history</strong> to be meaningful and easy to find. You can do this by clicking on the title of the history (which by default is <em>Unnamed history</em>) and typing <strong>Galaxy 101</strong> as the name. Do  not forget to hit the <kbd>Enter</kbd> key on your keyboard to save it.</p>

      <p><img src="../../../../shared/images/rename_history.png" alt="Rename the history" /></p>
    </li>
  </ol>

</blockquote>

<h2 id="locating-exons">Locating Exons</h2>

<p>First we need to get some data into our history. You can upload files from your computer, or Galaxy can also fetch data directly from external sources. We know UCSC has exon locations for humans and we can use Galaxy to import the data for chromosome 22, directly from the UCSC table browser.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-from-ucsc"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload from UCSC</h3>

  <ol>
    <li>
      <p><strong>UCSC Main</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> - table browser:</p>

      <p>In the tool menu, navigate to <code class="highlighter-rouge">Get Data -&gt; UCSC Main - table browser</code></p>

      <p><img src="../../images/101_01.png" alt="`UCSC Main table browser` menu entry" /></p>

      <p>You will be taken to the <strong>UCSC table browser</strong>, which looks something like this:</p>

      <p><img src="../../images/101_02.png" alt="`UCSC table browser` tool, first screen for exons" /></p>

      <p>Now set the following parameters:</p>
      <ul>
        <li><em>“clade”</em>: <code class="highlighter-rouge">Mammal</code></li>
        <li><em>“genome”</em>: <code class="highlighter-rouge">Human</code></li>
        <li><em>“assembly”</em>: <code class="highlighter-rouge">Dec. 2013 (GRCh38/hg38)</code></li>
        <li><em>“group”</em>: <code class="highlighter-rouge">Genes and Gene Predictions</code></li>
        <li><em>“track”</em>: <code class="highlighter-rouge">GENCODE v29</code></li>
        <li><em>“table”</em>: <code class="highlighter-rouge">knownGene</code></li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“region”</em> should be changed to <code class="highlighter-rouge">position</code> with value <code class="highlighter-rouge">chr22</code></li>
        <li><em>“output format”</em> should be changed to <code class="highlighter-rouge">BED - browser extensible data</code></li>
        <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Send output to”</em> should have the option <code class="highlighter-rouge">Galaxy</code> checked</li>
      </ul>
    </li>
    <li>
      <p>Click on the <strong>get output</strong> button and you will see the next screen:</p>

      <p><img src="../../images/101_03.png" alt="`UCSC table browser` tool, second screen for exons" /></p>

      <p>Change <strong>Create one BED record per</strong> to <code class="highlighter-rouge">Coding Exons</code> and then click on the <strong>Send query to Galaxy</strong> button.</p>

      <blockquote class="comment">
        <h3 id="comment-comment"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
        <p>After this you will see your first history item in Galaxy’s right panel. It will go through
the gray (preparing/queued) and yellow (running) states to become green (success):</p>

        <p><img src="../../images/101_04.png" alt="`UCSC Main on Human: knownGene` dataset is green" /></p>
      </blockquote>
    </li>
    <li>
      <p>When the dataset is green, click on the <i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon to <strong>view the contents</strong> of the file. It should look something like this:</p>

      <p><img src="../../images/101_exons.png" alt="Contents of the `UCSC Main on Human: knownGene` dataset" /></p>

      <p>Each line represents an exon, the first three columns are the genomic location, and the fourth column contains the name of the exon.</p>
    </li>
    <li>
      <p>Let’s rename our dataset to something more recognizable.</p>
      <ul>
        <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
        <li>In the central panel, change the <strong>Name</strong> field to <code class="highlighter-rouge">Exons</code></li>
        <li>Click the <strong>Save</strong> button</li>
      </ul>

      <p>Your history should now look something like this:</p>

      <p><img src="../../images/101_rename.png" alt="Rename dataset to `Exons`" /></p>
    </li>
  </ol>
</blockquote>

<h2 id="obtaining-snp-information">Obtaining SNP information</h2>

<p>We now have information about the exon locations, but our original question was which exon contains the largest number of SNPs, so let’s get some information about SNP locations from UCSC as well:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-snp-information"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: SNP information</h3>
  <p>Again open the <code class="highlighter-rouge">UCSC Main - table browser</code> tool and set the following parameters</p>

  <ol>
    <li><strong>UCSC Main</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> table browser:
      <ul>
        <li><em>“group”</em> should be changed to <code class="highlighter-rouge">Variation</code></li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“region”</em> should be changed again to <code class="highlighter-rouge">position</code> with value <code class="highlighter-rouge">chr22</code></li>
        <li><em>“output format”</em> should be changed again to <code class="highlighter-rouge">BED - browser extensible data</code></li>
      </ul>

      <p><img src="../../images/101_06.png" alt="`UCSC table browser` tool, first screen for SNPs" /></p>

      <p>The <em>“track”</em> setting shows the version of the SNP database to get. In this example it is version 150, but you may select the latest one. Your results may vary slightly from the ones in this tutorial when you select a different version, but in general it is a good idea to select the latest version, as this will contain the most up-to-date SNP information.</p>
    </li>
    <li>
      <p>Click on the <strong>get output</strong> button to find a form similar to this:</p>

      <p><img src="../../images/101_07.png" alt="`UCSC table browser` tool, second screen for SNPs" /></p>

      <p>Make sure that <em>“Create one BED record per”</em> is set to <code class="highlighter-rouge">Whole Gene</code> (Whole Gene here really means Whole Feature), and click on <strong>Send query to Galaxy</strong>. A second item will appear in your analysis history.</p>
    </li>
    <li>
      <p>Now <strong>rename</strong> your new dataset to <code class="highlighter-rouge">SNPs</code> so we can easily remember what the file contains.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field  to <code class="highlighter-rouge">SNPs</code></li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="analysis">Analysis</h1>

<h2 id="find-exons-with-the-most-snps">Find exons with the most SNPs</h2>

<p>Our objective is to find which exon contains the most SNPs. Therefore we have to intersect the file with the exon locations with the file containing the SNP locations (here “intersect” is just a fancy word for printing SNPs and exons that overlap side-by-side).</p>

<blockquote class="comment">
  <h3 id="comment-search-bar"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Search bar</h3>

  <p>Different Galaxy servers may have tools available under different sections, therefore it is often useful to use the <strong>search bar</strong> at the top of the tool panel to find your tool.</p>

  <p>Additionally different servers may have multiple, similarly named tools which accomplish similar functions. For these tutorials, you should select precisely the one that is described. However, in your real analyses, you’ll need to search among the various options to find the one that works for you.</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-finding-exons"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Finding Exons</h3>

  <ol>
    <li>
      <p><strong>Join</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> the intervals of two datasets side-by-side:</p>

      <p>Enter the word <code class="highlighter-rouge">join</code> in the search bar of the tool panel, and select the
tool named <code class="highlighter-rouge">Join - the intervals of two datasets side-by-side</code></p>

      <ul>
        <li><em>“Join”</em>: Select <code class="highlighter-rouge">Exons</code></li>
        <li><em>“With”</em>: Select <code class="highlighter-rouge">SNPs</code></li>
        <li><em>“with min overlap”</em>: <code class="highlighter-rouge">1</code></li>
        <li><em>“Return”</em>: <code class="highlighter-rouge">Only records that are joined (INNER JOIN)</code>, which means that only matches are included in the output (i.e. only exons with SNPs in it and only SNPs that fall in exons)</li>
      </ul>

      <blockquote class="tip">
        <h3 id="tip-how-do-i-use-this-tool"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> How do I use this tool?</h3>
        <p>All Galaxy tools include documentation. If you scroll down on this page, you will find the help of the tool.</p>
      </blockquote>
    </li>
    <li>Click <strong>Execute</strong></li>
    <li>Wait for the job to finish</li>
    <li>
      <p>View the resulting file (with the <i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon). If everything went okay, you should see a file that looks similar to this:</p>

      <p><img src="../../images/101_joined.png" alt="Contents of the `Join` output dataset" /></p>

      <p>Remember that variations are possible due to using different versions of UCSC databases: as long as you have similar looking columns you did everything right!</p>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-if-things-didnt-work"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> If things didn’t work…</h3>

    <p>Did the Join tool error with a memory failure? Or is this step executing for a long time? Most likely a setting was missed when extracting the data from the UCSC Table Browser. Try again, double checking that:</p>

    <ul>
      <li>For both SNP and EXON: <em>“region”</em> is actually changed to <code class="highlighter-rouge">position</code> with value <code class="highlighter-rouge">chr22</code></li>
      <li>For EXON: <em>“Create one BED record per”</em> <code class="highlighter-rouge">Coding Exons</code> is selected (<em>not</em> <code class="highlighter-rouge">Whole Gene</code> as for the SNP data)</li>
      <li>Carefully inspect the remaining Table Browser settings if these two most common reasons for problems were correct in your tool executions</li>
    </ul>
  </blockquote>

</blockquote>

<p>Let’s take a look at this dataset. The first six columns correspond to the exons, and the last six columns correspond to the SNPs. Column 4 contains the exon IDs, and column 10 contains the SNP IDs. In our screenshot you see that the first lines in the file all have the same exon ID but different SNP IDs, meaning these lines represent different SNPs that all overlap the same exon.</p>

<blockquote class="question">
  <h3 id="question-question"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>
  <p>For the first 3 exons in your file, what is the number of SNPs that fall into that exon?</p>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>At the time of writing, for hg38/GENCODE v29, joined with “Common SNPs(151)”, using <kbd>ctrl-f</kbd> to look for how many times each is used:</p>

    <table>
      <thead>
        <tr>
          <th>Gene</th>
          <th>Occurences</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td><code class="highlighter-rouge">ENST00000252835.5_cds_0_0_chr22_15528159_f</code></td>
          <td>24</td>
        </tr>
        <tr>
          <td><code class="highlighter-rouge">ENST00000643195.1_cds_0_0_chr22_15528192_f</code></td>
          <td>24</td>
        </tr>
        <tr>
          <td><code class="highlighter-rouge">ENST00000343518.10_cds_0_0_chr22_15690078_f</code></td>
          <td>40</td>
        </tr>
      </tbody>
    </table>
  </blockquote>
</blockquote>

<h2 id="count-the-number-of-snps-per-exon">Count the number of SNPs per exon</h2>

<p>Since each line in our file represents a single overlap between SNP and exon, we can find the total number of SNPs in an exon, simply by counting the number of lines that have the same exon ID. So let’s do this for all the exons in our file:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-counting-snps"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Counting SNPs</h3>

  <ol>
    <li>
      <p><strong>Group</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> data by a column and perform aggregate operation on other columns:</p>

      <ul>
        <li><em>“Select data”</em>: select the output dataset from <strong>Join</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Group by column”</em>: <code class="highlighter-rouge">Column: 4</code> (the column with the exon IDs)</li>
        <li><em>“Insert Operation”</em>:
          <ul>
            <li>1: Operation
              <ul>
                <li><em>“Type”</em>: <code class="highlighter-rouge">Count</code></li>
                <li><em>“On column”</em>: <code class="highlighter-rouge">Column: 4</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <p><img src="../../images/101_13.png" alt="Settings for the `Group` tool" /></p>
    </li>
    <li>
      <p>Click <strong>Execute</strong> to perform the grouping. Your new output dataset will look something like this:</p>

      <p><img src="../../images/101_14.png" alt="Contents of the `Group` output dataset" /></p>
    </li>
  </ol>

</blockquote>

<p>This file contains only two columns. The first contains the exon IDs, and the second the number of times that exon ID appeared in the file - in other words, how many SNPs were present in that exon.</p>

<blockquote class="question">
  <h3 id="question-question-1"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>
  <p>How many exons are there in total in your file?</p>

  <blockquote>
    <blockquote class="solution">
      <h3 id="solution-solution-1"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <p>Each line now represents a different exon, so you can see the answer to this when you expand the history item, as in the image above. The exact number you see for your dataset may be slightly different due to the updates to the exon and SNPs information in UCSC.</p>

    </blockquote>
  </blockquote>
</blockquote>

<h2 id="sort-the-exons-by-snps-count">Sort the exons by SNPs count</h2>

<p>Now that we have a list of all exons, and the number of SNPs they contain, we would like to know which exon has the <em>highest number</em> of SNPs. We can do this by sorting the file on the second column.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-sorting"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Sorting</h3>

  <ol>
    <li>
      <p><strong>Sort</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> data in ascending or descending order:</p>

      <ul>
        <li><em>“Sort Query”</em>: Output from <strong>Group</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Column Selections”</em>:
          <ul>
            <li>1: Column Selections
              <ul>
                <li><em>“on column”</em>: <code class="highlighter-rouge">Column: 2</code></li>
                <li><em>“in”</em>: <code class="highlighter-rouge">Descending order</code></li>
                <li><em>“Flavor”</em>: <code class="highlighter-rouge">Fast numeric sort (-n)</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Execute</strong></p>
    </li>
    <li>
      <p>Examine the output file.</p>

      <p><img src="../../images/101_16.png" alt="Contents of the `Sort` output dataset" /></p>

      <p>The file should look similar to before sorting, but now the exons with the highest number of SNPs are at the top.</p>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-question-2"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>
  <p>Which exon has the highest number of SNPs in your file?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>When this tutorial was last updated, <code class="highlighter-rouge">ENST00000343518.10_cds_0_0_chr22_15690078_f</code> had 40 SNPs.
Keep in mind this may depend on your settings when getting the data from UCSC.</p>
  </blockquote>
</blockquote>

<h2 id="select-the-top-five-exons">Select the top five exons</h2>

<p>Let’s say we want a list with just the top-5 exons with highest number of SNPs.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-select-first"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Select first</h3>

  <ol>
    <li>
      <p><strong>Select first</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> lines from a dataset:</p>

      <ul>
        <li><em>“Select first”</em>: <code class="highlighter-rouge">5</code></li>
        <li><em>“from”</em>: The output from <strong>Sort</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Execute</strong></p>
    </li>
    <li>
      <p>Examine the output file, this should contain only the first 5 lines of the previous dataset.</p>

      <p><img src="../../images/101_first_5.png" alt="Contents of the `Select first` output dataset" /></p>
    </li>
  </ol>
</blockquote>

<h2 id="recovering-exon-info">Recovering exon info</h2>

<p>Congratulations! You have now determined which exons on chromosome 22 have the highest number of SNPs, but what else can we learn about them? One way to learn more about a genetic location is to view it in a genome browser. However, in the process of getting our answer, we have lost information about the location of these exons on the chromosome. But fear not, Galaxy saves all of your data, so we can recover this information quite easily.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-compare-two-datasets"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Compare two Datasets</h3>

  <ol>
    <li>
      <p><strong>Compare two Datasets</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to find common or distinct rows:</p>

      <ul>
        <li><em>“Compare”</em>: <code class="highlighter-rouge">Exons</code></li>
        <li><em>“Using column”</em>: <code class="highlighter-rouge">Column: 4</code></li>
        <li><em>“against”</em>: the output from <strong>Select first</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Using column”</em>: <code class="highlighter-rouge">Column: 1</code></li>
        <li><em>“to find”</em>: <code class="highlighter-rouge">Matching rows of 1st dataset</code></li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong></li>
    <li>
      <p>Examine your output file. It should contain the locations of your top 5 exons:</p>

      <p><img src="../../images/101_19.png" alt="Contents of the `Compare two Datasets` output dataset" /></p>
    </li>
  </ol>
</blockquote>

<h2 id="displaying-data-in-ucsc-genome-browser">Displaying data in UCSC genome browser</h2>

<p>A good way to learn about these exons is to look at their genomic surrounding. This can be done by using genome browsers. Galaxy can launch a genome browser such as IGV on your local machine, and it can connect to online genome browsers as well. An example of such an online genome browser is the UCSC genome browser.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-ucsc-genome-browser"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: UCSC genome browser</h3>

  <ol>
    <li>
      <p>First, check that the <strong>database</strong> of your latest history dataset is <code class="highlighter-rouge">hg38</code>. If not, click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> pencil icon and modify the <strong>Database/Build:</strong> field to <code class="highlighter-rouge">Human Dec. 2013 (GRCh38/hg38) (hg38)</code>.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-databasebuild-dbkey"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing Database/Build (dbkey)</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Database/Build</strong> field</li>
          <li>Select your desired database key from the dropdown list: <code class="highlighter-rouge">hg38</code></li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>To <strong>visualize the data in UCSC genome browser</strong>, click on <code class="highlighter-rouge">display at UCSC main</code> option visible when you expand the history item.</p>

      <p><img src="../../images/101_displayucsc.png" alt="`display at UCSC main` link" /></p>

      <p>This will upload the data to UCSC as custom track. To see your data look at the <code class="highlighter-rouge">User Track</code> near the top. You can enter the coordinates of one of your exons at the top to jump to that location.</p>

      <p><img src="../../images/101_21.png" alt="`User Track` shown in the UCSC genome browser" /></p>
    </li>
  </ol>
</blockquote>

<p>UCSC provides a large number of tracks that can help you get a sense of your genomic area, it contains common SNPs, repeats, genes, and much more (scroll down to find all possible tracks).</p>

<h1 id="galaxy-management">Galaxy management</h1>

<p>In Galaxy your analyses live in histories such as your current one. Histories can be very large, and you can have as many histories as you want. You can control your histories (switching, copying, sharing, creating a fresh history, etc.) in the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>History Options</strong> menu on the top of the history panel:</p>

<p><img src="../../images/history_options_menu.png" alt="History options menu" /></p>

<p>If you create a new history, your current history does not disappear. If you would like to list all of your histories just use the <i class="fa fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> Multi-history view, or choose <code class="highlighter-rouge">Saved Histories</code> from the history options menu and you will see a list of all your histories in the central panel:</p>

<p><img src="../../images/101_23.png" alt="`Saved Histories` list" /></p>

<p>An alternative overview of your histories can be accessed by clicking on the <i class="fa fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> <strong>View all histories</strong> button at top of your history panel.</p>

<p><img src="../../images/101_history-overview.png" alt="Overview of histories with their datasets" /></p>

<p>Here you see a more detailed view of each history, and can perform the same operations as the other view:</p>
<ul>
  <li>switching between different histories,</li>
  <li>deleting a history,</li>
  <li>purging it (permanently deleting it, this action cannot be reversed)</li>
  <li>copying histories</li>
</ul>

<p>One difference is that this view lets you copy datasets between your histories by dragging and dropping them.</p>

<p>You can always return to your analysis view by clicking on <strong>Analyze Data</strong> in the top menu bar.</p>

<h2 id="convert-your-analysis-history-into-a-workflow">Convert your analysis history into a workflow</h2>

<p>When you look carefully at your history, you can see that it contains all steps of our analysis, from the beginning to the end. By building this history we have actually built a complete record of our analysis with Galaxy preserving all parameter settings applied at every step. But when you receive new data, or a new report is requested, it would be tedious to do each step over again. Wouldn’t it be nice to just convert this history into a workflow that we will be able to execute again and again?</p>

<p>Galaxy makes this very easy with the <code class="highlighter-rouge">Extract workflow</code> option. This means any time you want to build a workflow, you can just perform the steps once manually, and then convert it to a workflow, so that next time it will be a lot less work to do the same analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-workflow"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract workflow</h3>

  <ol>
    <li>
      <p><strong>Clean up</strong> your history. If you had any failed jobs (red), please remove those datasets from your history by clicking on the <code class="highlighter-rouge">x</code> button. This will make the creation of a workflow easier.</p>
    </li>
    <li>
      <p>Go to the history <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> History Options menu and select the <code class="highlighter-rouge">Extract Workflow</code> option.</p>

      <p><img src="../../images/history_menu_extract_workflow.png" alt="`Extract Workflow` entry in the history options menu" /></p>

      <p>The central panel will change as shown below and you will be able to choose which steps to include/exclude and how to name the newly created workflow.</p>

      <p><img src="../../images/101_25.png" alt="Selection of steps for `Extract Workflow` from history" /></p>
    </li>
    <li>
      <p><strong>Rename</strong> the workflow to something descriptive, for example <code class="highlighter-rouge">Find exons with the highest number of interactions</code>.</p>

      <p>While we created this workflow initially to analyse SNPs, if we had similarly formatted datasets we could use this workflow to find those interactions as well.</p>

      <p>If there are any steps that shouldn’t be included in the workflow, you can <strong>uncheck</strong> them.</p>
    </li>
    <li>
      <p>Click on the <strong>Create Workflow</strong> button near the top.</p>

      <p>You will get a message that the workflow was created. But where did it go?</p>
    </li>
    <li>
      <p>Click on <strong>Workflow</strong> in the top menu of Galaxy. Here you have a list of all your workflows. Your newly created workflow should be listed at the top:</p>

      <p><img src="../../images/101_26.png" alt="`Your workflows` list" /></p>
    </li>
  </ol>
</blockquote>

<h2 id="the-workflow-editor">The workflow editor</h2>

<p>We can examine the workflow in Galaxy’s workflow editor. Here you can view/change the parameter settings of each step, add and remove tools, and connect an output from one tool to the input of another, all in an easy and graphical manner. You can also use this editor to build workflows from scratch.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-workflow-1"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract workflow</h3>

  <ol>
    <li>
      <p>Click on the triangle to the right of your workflow name.</p>

      <p><img src="../../images/101_27.png" alt="`Edit` option in the menu for the latest workflow" /></p>
    </li>
    <li>
      <p>Select <strong>Edit</strong> to launch the workflow editor. You should see something like this:</p>

      <p><img src="../../images/101_28.png" alt="Workflow editor" /></p>

      <p>When you click on a workflow step, you will get a view of all the parameter settings for that tool on the right-hand side of your screen.</p>

      <blockquote class="tip">
        <h3 id="tip-hiding-intermediate-steps"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Hiding intermediate steps</h3>
        <p>When a workflow is executed, the user is usually primarily interested in the final product and not in all intermediate steps. By default all the outputs of a workflow will be shown, but we can explicitly tell Galaxy which outputs to show and which to hide for a given workflow. This behaviour is controlled by the little asterisk next to every output dataset:
<img src="../../../../shared/images/workflow_editor_mark_output.png" alt="Asterisk for `out_file1` in the `Select First` tool" /></p>

        <p>If you click on this asterisk for any of the output datasets, then <em>only</em> files with an asterisk will be shown, and all outputs without an asterisk will be hidden. (Note that clicking <em>all</em> outputs has the same effect as clicking <em>none</em> of the outputs, in both cases all the datasets will be shown.)</p>
      </blockquote>
    </li>
    <li>
      <p>Click the <strong>asterisk</strong> for <code class="highlighter-rouge">out_file1</code> in the <code class="highlighter-rouge">Select First</code> and <code class="highlighter-rouge">Compare two Datasets</code> tools.</p>

      <p>Now, when we run the workflow, we will only see the final two outputs, i.e. the table with the top-5 exons and their SNP counts, and the file with exons ready for viewing in a genome browser. Once you have done this, you will notice that the <strong>minimap</strong> at the bottom-right corner of your screen will have a colour-coded view of your workflow, with orange boxes representing a tool with an output that will be shown.</p>

      <p><img src="../../images/101_31.png" alt="Workflow minimap" /></p>

      <p>In the image above, you see that the top input dataset (with the blue border) connects to the first input of the <code class="highlighter-rouge">Join</code> tool, so this corresponds to the exon data.</p>
    </li>
    <li>
      <p>The box named <code class="highlighter-rouge">Exons</code> is named ok, but we want to change <code class="highlighter-rouge">SNPs</code> since this workflow is not specific to SNPs</p>

      <ul>
        <li><strong>Click</strong> on the box corresponding to the <code class="highlighter-rouge">SNPs</code> input dataset</li>
        <li>change the <strong>Label</strong> to <code class="highlighter-rouge">Features</code> on the right-hand side of your screen.</li>
      </ul>
    </li>
    <li>
      <p>Let’s also <strong>rename the outputs</strong>:</p>

      <ul>
        <li>Click on the <code class="highlighter-rouge">Select first</code> tool in the workflow editor</li>
        <li>In the menu on the right click on <code class="highlighter-rouge">Configure Output: 'out_file1'</code>
          <ul>
            <li>Under <code class="highlighter-rouge">Rename dataset</code>, and enter a descriptive name for the output dataset like <code class="highlighter-rouge">Top 5 exon IDs</code></li>
          </ul>
        </li>
      </ul>

      <p><img src="../../images/101_34.png" alt="Rename the output of the `Select first` tool step" /></p>
    </li>
    <li>
      <p><strong>Repeat</strong> this for the output of the <code class="highlighter-rouge">Compare two Datasets</code> tool, naming it <code class="highlighter-rouge">Top 5 exons</code></p>
    </li>
    <li>
      <p><strong>Save your workflow</strong> (important!) by clicking on the <i class="fa fa fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span> icon at the top right of the screen.</p>

      <p><img src="../../images/workflow_editor_save.png" alt="Save option in the workflow editor menu" /></p>
    </li>
    <li>
      <p><strong>Return</strong> to the analysis view by clicking on <strong>Analyze Data</strong> at the top menu bar.</p>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-comments"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>We could <strong>validate</strong> our newly built workflow by running it on the same input datasets that we used at the start of this tutorial, in order to make sure we do obtain the same results.</p>
</blockquote>

<h2 id="run-workflow-on-different-data">Run workflow on different data</h2>

<p>Now that we have built our workflow, let’s use it on some different data. For example, let’s find out which exons have the highest number of repeat elements.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-run-workflow"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run workflow</h3>

  <ol>
    <li>
      <p>Create a <strong>new history</strong> and give it a name.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history-1"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>We will need the list of exons again. We don’t have to get this from UCSC again, we can just <strong>copy</strong> it from our previous history. The easiest way to do this is to go to the <i class="fa fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> history overview. Here you can just drag and drop datasets from one history to another.</p>

      <p><img src="../../images/101_copydataset.png" alt="Drag and drop of `Exons` dataset in the history overview" /></p>
    </li>
    <li>
      <p>Click <strong>Analyze Data</strong> at the top to return to the main analysis window</p>
    </li>
    <li>We wanted to know something about the repetitive elements per exon. We get this data from UCSC.
      <ul>
        <li><em>“assembly”</em>: <code class="highlighter-rouge">Dec. 2013 (GRCh38/hg38)</code></li>
        <li><em>“group”</em> parameter should be changed to <code class="highlighter-rouge">Repeats</code></li>
        <li><em>“position”</em>: <code class="highlighter-rouge">chr22</code></li>
        <li>leave the rest of the settings to the defaults</li>
      </ul>

      <p>Click on <strong>get output</strong> and then <strong>Send query to Galaxy</strong> on the next screen.</p>
    </li>
    <li>
      <p>Open the <strong>workflow menu</strong> (top menu bar). Find the workflow you made in the previous section, and select the option <code class="highlighter-rouge">Run</code>.</p>

      <p><img src="../../images/101_37.png" alt="`Run` option in the workflow menu" /></p>

      <p>The central panel will change to allow you to configure and launch the workflow.</p>
    </li>
    <li>
      <p>Select appropriate datasets for the inputs as shown below, then scroll down and click <code class="highlighter-rouge">Run workflow</code>.</p>

      <p><img src="../../images/101_38.png" alt="Settings for running the workflow" /></p>

      <blockquote class="comment">
        <h3 id="comment-potential-workflow-issues"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Potential workflow issues</h3>
        <p>Galaxy validates the workflow inputs to ensure they’re correct. It may show a validation error at the start, until you select <code class="highlighter-rouge">Exons</code> for the Exons input, and your repeats for the Features input.</p>
      </blockquote>
    </li>
    <li>
      <p>Once the workflow has started, you will initially be able to see all its steps, but the unimportant intermediates will disappear after they complete successfully:</p>

      <p><img src="../../images/101_39.png" alt="Datasets appearing in the history" /></p>
    </li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-comment-1"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
  <p>Because most intermediate steps of the workflow were hidden, once it is finished you will only see the final two datasets. If we want to view the intermediate files after all, we can unhide all hidden datasets by clicking on “hidden”, just below the history name, or by selecting <code class="highlighter-rouge">Unhide Hidden Datasets</code> from the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> history options menu.</p>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <p>Which exon had the highest number of repeats? How many repeats were there?</p>
</blockquote>

<h2 id="share-your-work">Share your work</h2>

<p>One of the most important features of Galaxy comes at the end of an analysis. When you have published striking findings, it is important that other researchers are able to reproduce your in-silico experiment. Galaxy enables users to easily share their workflows and histories with others.</p>

<p>To share a history, click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon in the history panel and select <code class="highlighter-rouge">Share or Publish</code>. On this page you can do 3 things:</p>

<ol>
  <li><strong>Make History Accessible via Link</strong>. This generates a link that you can give out to others. Anybody with this link will be able to view your history.</li>
  <li><strong>Make History Accessible and Publish</strong>. This will not only create a link, but will also publish your history. This means your history will be listed under <code class="highlighter-rouge">Shared Data → Histories</code> in the top menu.</li>
  <li><strong>Share with a user</strong>. This will share the history only with specific users on the Galaxy instance.</li>
</ol>

<blockquote class="comment">
  <h3 id="comment-permissions"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Permissions</h3>
  <p>Different servers have different default permission settings. Some servers create all of your datasets completely private to you, while others make them accessible if you know the secret ID.</p>

  <p>Be sure to select <strong>Also make all objects within the History accessible</strong> whenever you make a history accessible via link, otherwise whomever you send your link to might not be able to see your history.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-share-history-and-workflow"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Share history and workflow</h3>

  <ol>
    <li>Share one of your histories with your neighbour.</li>
    <li>See if you can do the same with your workflow!</li>
    <li>Find the history and/or workflow shared by your neighbour. Histories shared with specific users can be accessed by those users in their <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> history menu under <code class="highlighter-rouge">Histories shared with me</code>.</li>
  </ol>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p><i class="fa fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> Well done! You have just performed your first analysis in Galaxy. You also created a workflow from your analysis so you can easily repeat the exact same analysis on other datasets. Additionally you shared your results and methods with others.</p>



                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>Galaxy provides an easy-to-use graphical user interface for often complex command-line tools</li>
                            
                            <li>Galaxy keeps a full record of your analysis in a history</li>
                            
                            <li>Workflows enable you to repeat your analysis on different data</li>
                            
                            <li>Galaxy can connect to external sources for data import and visualization purposes</li>
                            
                            <li>Galaxy provides ways to share your results and methods with others</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

        <hr>
        <br>

        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Galaxy 101 (Introduction to Galaxy Analyses)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=Galaxy 101 (Introduction to Galaxy Analyses)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Saskia Hiltemann, Nicola Soranzo, Clemens Blank, Anton Nekrutenko, Björn Grüning, Anne Pajon, Helena Rasche)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/introduction/tutorials/galaxy-intro-101/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
