<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>RAD-Seq de-novo data analysis</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Analyses of sequences" />
        <meta property="og:title" content="Galaxy Training: RAD-Seq de-novo data analysis" />
        <meta property="og:description" content="Analyses of sequences" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/sequence-analysis" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Sequence analysis
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fde-novo-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fde-novo-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fde-novo-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fde-novo-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fde-novo-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/sequence-analysis/tutorials/de-novo-rad-seq/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "RAD-Seq de-novo data analysis",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "sequence-analysis / de-novo-rad-seq / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'RAD-Seq de-novo data analysis' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "RAD-Seq de-novo data analysis",
    "description": "RAD-Seq de-novo data analysis",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/de-novo-rad-seq/tutorial.html",
  "timeRequired": "PT8H",
  "description": "RAD-Seq de-novo data analysis\\nThe questions this  addresses are:\n - How to analyze RAD sequencing data without a reference genome for a population genomics study?\n\n\\nThe objectives are:\n - Analysis of RAD sequencing data without a reference genome\n - SNP calling from RAD sequencing data\n - Calculate population genomics statistics from RAD sequencing data\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.1134547",
      "name": "Training data for RAD-Seq de-novo data analysis tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Yvan Le Bras"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Yvan Le Bras"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Sequence analysis",
      "description": "Analyses of sequences",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>RAD-Seq de-novo data analysis</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#yvanlebras" class="contributor-badge"><img src="https://avatars.githubusercontent.com/yvanlebras" alt="Yvan Le Bras">Yvan Le Bras</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>How to analyze RAD sequencing data without a reference genome for a population genomics study?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Analysis of RAD sequencing data without a reference genome</li>
            
            <li>SNP calling from RAD sequencing data</li>
            
            <li>Calculate population genomics statistics from RAD sequencing data</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 8 hours</p>
            

            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    

                    
                    <div>
                        
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1134547">
    <i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>


                    </div>
                    

                    
                    <div>
                        
    <a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/tutorials/de-novo-rad-seq/workflows/" title="Workflows" alt="RAD-Seq de-novo data analysis workflows">
        <i class="fa fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>


                    </div>
                    
                    
                    <div>
                        
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/tutorials/de-novo-rad-seq/tours/">
    <i class="fa fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>


                    </div>
                    


                    
                    
                    <div>
                        

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">instances</span> Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/docker" title="Docker image for this tutorial">
                <i class="fa fa-ship" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>



                    </div>
                    
                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In the study of <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000862">Hohenlohe <em>et al.</em> 2010</a>, a genome scan of nucleotide diversity and differentiation in natural populations of threespine stickleback <em>Gasterosteus aculeatus</em> was conducted. Authors used Illumina-sequenced RAD tags to identify and type over 45,000 single nucleotide polymorphisms (SNPs) in each of 100 individuals from two oceanic and three freshwater populations.</p>

<p><img src="../../images/RAD4_Population_Genomics/Hohenlohe_et_al_2010.png" alt="Abstract of the paper on Population Genomics" /></p>

<p>We here proposed to re-analyze these data at least until the population genomics statistics calculation step using STACKS pipeline. Existing <em>Gasterosteus aculeatus</em> draft genome will not be used here so the analysis will be performed de novo. In a de novo RAD-seq data analysis, the reads are aligned one on each other to create stacks and then clustered to build loci. A reference approach can also be conducted (see <a href="/training-material/topics/sequence-analysis/tutorials/ref-based-rad-seq/tutorial.html">ref_based tutorial</a>, allowing to work on existing assembled loci).</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#demultiplexing-reads" id="markdown-toc-demultiplexing-reads">Demultiplexing reads</a></li>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
    </ol>
  </li>
  <li><a href="#snp-calling-from-radtags" id="markdown-toc-snp-calling-from-radtags">SNP calling from radtags</a></li>
  <li><a href="#calculate-population-genomics-statistics" id="markdown-toc-calculate-population-genomics-statistics">Calculate population genomics statistics</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<h2 id="data-upload">Data upload</h2>

<p>The original data is available at NCBI SRA ENA under accession number <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR034310">SRR034310</a> as part of the NCBI SRA ENA study accession number <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP001747">SRP001747</a>.</p>

<p>We will look at the first run SRR034310 out of seven which includes 16 samples from 2 populations, 8 from Bear Paw (freshwater) and 8 from Rabbit Slough (oceanic). We will download the reads directly from SRA and the remaining data (i.e reference genome, population map file, and barcodes file) from <a href="https://zenodo.org/record/1134547#.WkugUjfjJXI">Zenodo</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>Create a <strong>new history</strong> for this RAD-seq exercise. If you are not inspired, you can name it <em>STACKS RAD: population genomics without reference genome</em> for example…
      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li><strong>EBI SRA</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>Select the Run from the results of the search for <code class="highlighter-rouge">SRR034310</code> (which will present you 1 Experiment (SRX015877) and 1 Run (SRR034310)).</li>
        <li>Click the link in the column <strong>FASTQ files (Galaxy)</strong> of the results table</li>
        <li>This will redirect to the Galaxy website and start the download.</li>
      </ul>
    </li>
    <li>Upload remaining training data from Zenodo:
      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1134547/files/Barcode_SRR034310.txt
https://zenodo.org/record/1134547/files/Details_Barcode_Population_SRR034310.txt
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the Galaxy Upload Manager (<i class="fa fa fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, Galaxy uses the URL as the name, so rename the files with a more useful name.</p>
      </blockquote>
    </li>
    <li>
      <p>Make sure your fasta files are of datatype <code class="highlighter-rouge">fastqsanger</code></p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fa fa fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="highlighter-rouge">fastqsanger</code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The sequences are raw sequences from the sequencing machine, without any pretreatments. They need to be demultiplexed. To do so, we can use the Process Radtags tool from STACKS.</p>

<h2 id="demultiplexing-reads">Demultiplexing reads</h2>

<p>For demultiplexing, we use the Process Radtags tool from <a href="https://www.g3journal.org/content/1/3/171.full">STACKS</a> .</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-demultiplexing-reads"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Demultiplexing reads</h3>

  <ol>
    <li><strong>Process Radtags</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to demultiplex the reads:
      <ul>
        <li><em>“Single-end or paired-end reads files”</em>: <code class="highlighter-rouge">Single-end files</code></li>
        <li><em>“singles-end reads infile(s)”</em>: <code class="highlighter-rouge">SRR034310.fastq(.gz)</code></li>
        <li><em>“Barcode file”</em>: <code class="highlighter-rouge">Barcodes_SRR034310.tabular</code></li>
        <li><em>“Number of enzymes”</em>: <code class="highlighter-rouge">One</code></li>
        <li><em>“Enzyme”</em>: <code class="highlighter-rouge">sbfI</code></li>
        <li><em>“Capture discarded reads to a file</em>”: <code class="highlighter-rouge">Yes</code></li>
        <li><em>“Output format:”</em> <code class="highlighter-rouge">fastq</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How many reads where on the original dataset?</li>
          <li>How many are kept?</li>
          <li>Can you try to explain the reason why we loose a lot of reads here?</li>
          <li>What kind of information does this result give concerning the upcoming data analysis and the barcodes design in general?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>The informations can be found in the results log file:</p>

          <ol>
            <li>8895289 total reads</li>
            <li>8139531 retained reads</li>
            <li>There are no sequences filtered because of low quality. This is because radtags didn’t apply quality related filtering since the corresponding advanced option (Discard reads with low quality scores) has not been enabled. So here, all not retained sequences are removed because of an ambiguous barcode (<code class="highlighter-rouge">626265</code>) or an ambiguous RAD-Tag (<code class="highlighter-rouge">129493</code>). This means that some barcodes are not exactly what was specified on the barcode file and that sometimes, no SbfI restriction enzyme site was found. This can be due to some sequencing problems but here, this is also due to the addition, in the original sequencing library, of RAD-seq samples from another study. This strategy is often used to avoid having too much sequences beginning with the exact same nucleotide sequence which may cause Illumina related issues during sequencing and cluster analysis</li>
            <li>Sequencing quality is essential! Each time your sequencing quality decreases, you loose data and thus essential biological information!</li>
          </ol>

          <p>In addition to the overall statistics the numbers of retained and removed reads are also given for each bar code sequence.</p>
        </blockquote>
      </blockquote>

      <p>Next we will play around with the parameters a bit, to see the impact they have on the results.</p>
    </li>
    <li><strong>Process Radtags</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Re-Run playing with parameters
      <ul>
        <li>Section <strong>Advanced</strong>
          <ul>
            <li><em>“Discard reads with low quality scores”</em>: <code class="highlighter-rouge">Yes</code></li>
            <li><em>“score limit”</em>: <code class="highlighter-rouge">20</code> (for example; play with this)</li>
            <li><em>“Set the size of the sliding window as a fraction of the read length, between 0 and 1”</em>: <code class="highlighter-rouge">0.30</code> (for example; play with this)</li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <p>This sliding window parameter allows notably the user to deal with the declining quality at the 3’ end of reads.</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on</h3>

  <p>You can use the <code class="highlighter-rouge">Charts</code> functionality through the Visualize button reachable on the <code class="highlighter-rouge">Radtags logs</code> file you just generated.</p>

  <p>If like me you don’t have payed attention to the organization of you file for the graphical representation you obtain a non optimal bars diagram with a not intelligent X-axis ordering. There is a lot of different manner to fix this. You can use a copy/paste “bidouille” or you can use Galaxy tools to manipulate the <code class="highlighter-rouge">radtags logs</code> file to generate a better graph. For example, you can use <code class="highlighter-rouge">Select lines that match an expression</code> tool to select rows then use the <code class="highlighter-rouge">Concatenate datasets tail-to-head</code> tool to reorganize these lines in a new file.
Then we generate a graphical display of the changes:</p>

  <p>First we cut the interesting lines of each <code class="highlighter-rouge">result.log with Stacks: process radtags</code></p>

  <ol>
    <li><strong>Select lines that match an expression</strong> applying <code class="highlighter-rouge">^R1.fq.gz</code> on the log files and then</li>
    <li><strong>Replace Text in entire line</strong> on the resulting data sets finding <code class="highlighter-rouge">R1.fq.gz</code> and replacing with <code class="highlighter-rouge">NoScoreLimit</code> or <code class="highlighter-rouge">Score10</code> or <code class="highlighter-rouge">Score20</code> depending of the case</li>
    <li><strong>Select lines that match an expression</strong> applying <code class="highlighter-rouge">File\tRetained Reads</code> on one of the log file to obtain you futur header</li>
    <li><strong>Replace Text in entire line</strong> on the resulting data set finding <code class="highlighter-rouge">File</code> and replacing with <code class="highlighter-rouge">#</code> to have a best display</li>
    <li><strong>Concatenate datasets tail-to-head</strong> on the resulting data sets stating from the header you just created</li>
    <li>Finally <strong>Convert delimiters to TAB</strong> on the resulting data set converting all Tabs to be sure having a well formatted tabular file at the end</li>
  </ol>

  <p>Alternatively just copy/paste these lines on the Galaxy upload tool using Paste/fetch data section and modifying the File header by sample and filename by Score 10 / Score 20 and Lowquality for example… Before Starting the upload, you can select the <code class="highlighter-rouge">Convert spaces to tabs</code> option through the <code class="highlighter-rouge">Upload configuration</code> wheel. If you did not pay attention to the order you can just sort the file using the first column.</p>
</blockquote>

<p>You can use the <code class="highlighter-rouge">Charts</code> functionality through the Visualize button to plot the data.</p>

<p>And you obtain a file like this one, ready to generate a beautiful and smart bar stacked!</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#		Retained Reads	Low Quality	Ambiguous Barcodes	Ambiguous RAD-Tag	Total
NoScoreLimit	8139531		0		626265			129493			8895289
Score10		7373160		766371		626265			129493			8895289
Score20		2980543		5158988		626265			129493			8895289
</code></pre></div></div>

<p><img src="../../images/RAD4_Population_Genomics/Process_radtags_charts_end.PNG" alt="The chart on the sorted file" width="50%" /></p>

<p>You can further test using a filter like <code class="highlighter-rouge">clean data, remove any read with an uncalled base</code> and see that here, this has only little impact on the number of retained reads.</p>

<p>The demultiplexed sequences are raw sequences from the sequencing machine, without any pretreatments. They need to be controlled for their quality.</p>

<p>We propose to continue the tutorial using the dataset collection containing the demultiplexed reads obtained with Process Radtag execution made with a quality score of 10 and with the <code class="highlighter-rouge">Discard reads with low quality scores</code> parameter set to Yes (so containing 7373160 retained reads).</p>

<h2 id="quality-control">Quality control</h2>

<p>For quality control, we use similar tools as described in <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">NGS-QC tutorial</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control</h3>

  <ol>
    <li>
      <p><strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run FastQC on FastQ files to control the quality of the reads. Warning! Don’t forget you are working on data collections….</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is the read length?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>The read length is 32 bp</p>
        </blockquote>
      </blockquote>
    </li>
    <li>
      <p><strong>MultiQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run MultiQC on FastQC results to better see quality information over samples.</p>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/RAD4_Population_Genomics/multiqc.PNG" alt="MultiQC output" /></p>

<h1 id="snp-calling-from-radtags">SNP calling from radtags</h1>

<p>Run <code class="highlighter-rouge">Stacks: De novo map</code> Galaxy tool. This program will run ustacks, cstacks, and sstacks on the members of the population, accounting for the alignments of each read.</p>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>Information on denovo_map.pl and its parameters can be found online: https://creskolab.uoregon.edu/stacks/comp/denovo_map.php.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-stacks-de-novo-map"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Stacks: De novo map</h3>

  <ol>
    <li>
      <p>First we will create a population map file with the following contents:</p>

      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sample_CCCC	1
sample_CCAA	1
sample_CCTT	1
sample_CCGG	1
sample_CACA	1
sample_CAAC	1
sample_CATG	1
sample_CAGT	1
sample_CTCT	2
sample_CTAG	2
sample_CTTC	2
sample_CTGA	2
sample_GGGG	2
sample_GGAA	2
sample_GGTT	2
sample_GGCC	2
</code></pre></div>      </div>
      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-file"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new file</h3>

        <ul>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the file contents into the text field</p>
          </li>
          <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li>
        </ul>
      </blockquote>
    </li>
    <li><strong>Stacks: De novo map</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: with the following parameters:
      <ul>
        <li><em>“Select your usage”</em>: <code class="highlighter-rouge">Population</code></li>
        <li><em>“Files containing an individual sample from a population”</em>: <code class="highlighter-rouge">The demultiplexed reads (collection)</code></li>
        <li><em>“Specify a population map”</em>: <code class="highlighter-rouge">the population map you just created</code></li>
        <li>Section <strong>Assembly options</strong>
          <ul>
            <li><em>“Minimum number of identical raw reads required to create a stack”</em>: <code class="highlighter-rouge">3</code></li>
            <li><em>“Number of mismatches allowed between loci when building the catalog”</em>: <code class="highlighter-rouge">3</code></li>
            <li><em>“Remove, or break up, highly repetitive RAD-Tags in the ustacks program”</em>: <code class="highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Once Stacks has completed running, investigate the output files: <code class="highlighter-rouge">result.log</code> and <code class="highlighter-rouge">catalog.</code> (snps, alleles and tags).</p>

      <p><img src="../../images/RAD4_Population_Genomics/denovo/denovo_out.png" alt="The output of de novo map" /></p>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-data-formatting"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Data formatting</h3>

    <p>If you are using a file presenting population information and individual name in a different manner than expected by STACKS, you can use Galaxy tools like <code class="highlighter-rouge">Replace Text</code> (for example to replace <code class="highlighter-rouge">Rabbit Slough</code> by a population number like <code class="highlighter-rouge">2</code>, <code class="highlighter-rouge">Add column</code> (for example to add <code class="highlighter-rouge">sample_</code>) or <code class="highlighter-rouge">Cut columns from a table</code> (to put the new <code class="highlighter-rouge">sample_</code> column af the first place) and finally <code class="highlighter-rouge">Regex replace</code> (replacing <code class="highlighter-rouge">(sample_)\t</code> by <code class="highlighter-rouge">\1</code>) to generate it…</p>
  </blockquote>

</blockquote>

<h1 id="calculate-population-genomics-statistics">Calculate population genomics statistics</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-calculate-population-genomics-statistics"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Calculate population genomics statistics</h3>

  <ol>
    <li><strong>Stacks: populations</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>Section <strong>Data filtering options</strong>
          <ul>
            <li><em>“Minimum percentage of individuals in a population required to process a locus for that population”</em>: <code class="highlighter-rouge">0.75</code></li>
          </ul>
        </li>
        <li>Section <strong>Output options</strong> (VCF and Structure) and enabling SNP and haplotype-based F statistics calculation.
          <ul>
            <li><em>“Output results in Variant Call Format (VCF)”</em>: <code class="highlighter-rouge">Yes</code></li>
            <li><em>“Output results in Structure Format”</em>: <code class="highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>“Enable SNP and haplotype-based F statistics”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>

      <p><img src="../../images/RAD4_Population_Genomics/denovo/populations_log.png" alt="The output of the populations tool" /></p>
    </li>
    <li>Now look at the output in the file <code class="highlighter-rouge">batch_1.sumstats</code> nammed <code class="highlighter-rouge">SNP and Haplotype-based F statistics with Stacks: populations ...</code> on your history. There are a large number of statistics calculated at each SNP, so use Galaxy tools like filter, cut, and sort to focus on some.</li>
  </ol>

  <blockquote class="question">
    <h3 id="question-question"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

    <ol>
      <li>What is the maximum value of FST’ at any SNP? Don’t hesitate to look at the <a href="http://catchenlab.life.illinois.edu/stacks/manual/#files">STACKS manual</a> to find where this parameter is</li>
      <li>What is the meaning of this FST’ value compared a a classical FST one? Once again, the <a href="http://catchenlab.life.illinois.edu/stacks/manual/#files">STACKS manual</a> can help you ;)</li>
      <li>How many SNPs reach this FST’ value?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution-2"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <ol>
        <li>1</li>
        <li>FST’ is a haplotype measure of FST that is scaled to the theoretical maximum FST value at this locus. Depending on how many haplotypes there are, it may not be possible to reach an FST of 1, so this method will scale the value to 1</li>
        <li>114</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>
<blockquote>
  <p>You can now for xample filter this dataset to only keep FST’=1 loci for further analysis…</p>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we have analyzed real RAD sequencing data to extract useful information, such as which loci are candidate regarding the genetic differentiation between freshwater and oceanic Stickelback populations. To answer these questions, we analyzed RAD sequence datasets using a de novo RAD-seq data analysis approach. This approach can be sum up with the following scheme:</p>

<p><img src="../../images/denovo_based_workflow.PNG" alt="The de novo tutorial workflow" /></p>



                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/sequence-analysis#references">here</a>.</p>
                    

                    

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

        <hr>
        <br>

        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=RAD-Seq de-novo data analysis (Sequence analysis)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=RAD-Seq de-novo data analysis (Sequence analysis)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Yvan Le Bras)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/tutorials/de-novo-rad-seq/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
