<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>From peaks to genes</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Galaxy is a scientific workflow, data integration, and da..." />
        <meta property="og:title" content="Galaxy Training: From peaks to genes" />
        <meta property="og:description" content="Galaxy is a scientific workflow, data integration, and da..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/introduction" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Introduction to Galaxy Analyses
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-peaks2genes%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-peaks2genes%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-peaks2genes%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-peaks2genes%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-peaks2genes%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "From peaks to genes",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "introduction / galaxy-intro-peaks2genes / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'From peaks to genes' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "From peaks to genes",
    "description": "From peaks to genes",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html",
  "timeRequired": "PT3H",
  "description": "From peaks to genes\\nThe questions this  addresses are:\n - How to use Galaxy?\n - How to get from peak regions to a list of gene names?\n\n\\nThe objectives are:\n - Familiarize yourself with the basics of Galaxy\n - Learn how to obtain data from external sources\n - Learn how to run tools\n - Learn how histories work\n - Learn how to create a workflow\n - Learn how to share your work\n\n",
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.1025586",
      "name": "Training data for From peaks to genes tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Anne Pajon"
    },
    {
      "@type": "Person",
      "name": "Clemens Blank"
    },
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Björn Grüning"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Dilmurat Yusuf"
    },
    {
      "@type": "Person",
      "name": "Sarah Peter"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Anne Pajon"
    },
    {
      "@type": "Person",
      "name": "Clemens Blank"
    },
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Björn Grüning"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Dilmurat Yusuf"
    },
    {
      "@type": "Person",
      "name": "Sarah Peter"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Introduction to Galaxy Analyses",
      "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/"
    }
  ],
  "skillLevel": "Introductory"
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>From peaks to genes</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#pajanne" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pajanne" alt="Anne Pajon">Anne Pajon</a>, <a href="/training-material/hall-of-fame#blankclemens" class="contributor-badge"><img src="https://avatars.githubusercontent.com/blankclemens" alt="Clemens Blank">Clemens Blank</a>, <a href="/training-material/hall-of-fame#bebatut" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bebatut" alt="Bérénice Batut">Bérénice Batut</a>, <a href="/training-material/hall-of-fame#bgruening" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bgruening" alt="Björn Grüning">Björn Grüning</a>, <a href="/training-material/hall-of-fame#nsoranzo" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nsoranzo" alt="Nicola Soranzo">Nicola Soranzo</a>, <a href="/training-material/hall-of-fame#dyusuf" class="contributor-badge"><img src="https://avatars.githubusercontent.com/dyusuf" alt="Dilmurat Yusuf">Dilmurat Yusuf</a>, <a href="/training-material/hall-of-fame#sarah-peter" class="contributor-badge"><img src="https://avatars.githubusercontent.com/sarah-peter" alt="Sarah Peter">Sarah Peter</a>, <a href="/training-material/hall-of-fame#erasche" class="contributor-badge"><img src="https://avatars.githubusercontent.com/erasche" alt="Helena Rasche">Helena Rasche</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>How to use Galaxy?</li>
            
            <li>How to get from peak regions to a list of gene names?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Familiarize yourself with the basics of Galaxy</li>
            
            <li>Learn how to obtain data from external sources</li>
            
            <li>Learn how to run tools</li>
            
            <li>Learn how histories work</li>
            
            <li>Learn how to create a workflow</li>
            
            <li>Learn how to share your work</li>
            
            </ul>

            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 3 hours</p>
            

            
            <p><strong><i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> Level: </strong>
             Introductory 
<span class="level introductory" title="Introductory Tutorial">
  <i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span>
</span>

</p>
            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    
                    <div>
                        <a class="topic-icon" href="/training-material/topics/introduction/slides/introduction.html" title="Introduction slides">
                            <i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Introduction Slides
                        </a>
                    </div>
                    

                    
                    <div>
                        
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1025586">
    <i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>


                    </div>
                    

                    
                    


                    
                    
                    <div>
                        

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">instances</span> Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/introduction/docker" title="Docker image for this tutorial">
                <i class="fa fa-ship" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>



                    </div>
                    
                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>We stumbled upon a paper (<a class="citation" href="#Li2012">Li <i>et al.</i> 2012</a>) called <em>“The histone acetyltransferase MOF is a key regulator of the embryonic stem cell core transcriptional network”</em>. The paper contains the analysis of possible target genes of an interesting protein called Mof. The targets were obtained by ChIP-seq in mice and the raw data is available through <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE37268">GEO</a>.
However, the list of genes is neither in the supplement of the paper, nor part of the GEO submission.
The closest thing we could find is a file in GEO containing a list of the regions where the signal is significantly enriched (so called <em>peaks</em>):</p>

<table>
  <tbody>
    <tr>
      <td>1</td>
      <td>3660676</td>
      <td>3661050</td>
      <td>375</td>
      <td>210</td>
      <td>62.0876250438913</td>
      <td>-2.00329386666667</td>
    </tr>
    <tr>
      <td>1</td>
      <td>3661326</td>
      <td>3661500</td>
      <td>175</td>
      <td>102</td>
      <td>28.2950833625942</td>
      <td>-0.695557142857143</td>
    </tr>
    <tr>
      <td>1</td>
      <td>3661976</td>
      <td>3662325</td>
      <td>350</td>
      <td>275</td>
      <td>48.3062708406486</td>
      <td>-1.29391285714286</td>
    </tr>
    <tr>
      <td>1</td>
      <td>3984926</td>
      <td>3985075</td>
      <td>150</td>
      <td>93</td>
      <td>34.1879823073944</td>
      <td>-0.816992</td>
    </tr>
    <tr>
      <td>1</td>
      <td>4424801</td>
      <td>4424900</td>
      <td>100</td>
      <td>70</td>
      <td>26.8023246007435</td>
      <td>-0.66282</td>
    </tr>
  </tbody>
</table>

<p><strong>Table 1</strong> Subsample of the available file</p>

<p>The goal of this exercise is to <strong>turn this list of genomic regions into a list of possible target genes</strong>.</p>

<blockquote class="comment">
  <h3 id="comment-note-results-may-vary"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note: results may vary</h3>

  <p>Your results may be slightly different from the ones presented in this tutorial
due to differing versions of tools, reference data, external databases, or
because of stochastic processes in the algorithms.</p>

</blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
    </ol>
  </li>
  <li><a href="#part-1-naive-approach" id="markdown-toc-part-1-naive-approach">Part 1: Naive approach</a>    <ol>
      <li><a href="#file-preparation" id="markdown-toc-file-preparation">File preparation</a></li>
      <li><a href="#analysis" id="markdown-toc-analysis">Analysis</a></li>
      <li><a href="#visualization" id="markdown-toc-visualization">Visualization</a></li>
      <li><a href="#extracting-workflow" id="markdown-toc-extracting-workflow">Extracting workflow</a></li>
    </ol>
  </li>
  <li><a href="#part-2-more-sophisticated-approach" id="markdown-toc-part-2-more-sophisticated-approach">Part 2: More sophisticated approach</a>    <ol>
      <li><a href="#preparation" id="markdown-toc-preparation">Preparation</a></li>
      <li><a href="#create-peak-summit-file" id="markdown-toc-create-peak-summit-file">Create peak summit file</a></li>
      <li><a href="#get-gene-names" id="markdown-toc-get-gene-names">Get gene names</a></li>
      <li><a href="#repeat-workflow" id="markdown-toc-repeat-workflow">Repeat workflow</a></li>
    </ol>
  </li>
  <li><a href="#share-your-work" id="markdown-toc-share-your-work">Share your work</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-open-galaxy"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Open Galaxy</h3>
  <ol>
    <li>Browse to a Galaxy instance: the one recommended by your instructor or one in the list <strong>Galaxy instance</strong> on the head of this page</li>
    <li>
      <p>Log in or register (top panel)</p>

      <p><img src="../../images/login_register.png" alt="Login or Register on the top panel" /></p>
    </li>
  </ol>
</blockquote>

<p>The Galaxy interface consist of three main parts. The available tools are listed on the left, your analysis history is recorded on the right, and the central panel will show the tools and datasets.</p>

<figure id="figure-1"><img src="../../images/galaxy_interface.png" alt="Galaxy interface" /><figcaption><span class="figcaption-prefix">Figure 1:</span> The Galaxy interface</figcaption></figure>

<p>Let’s start with a fresh history.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-history"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create history</h3>

  <ol>
    <li>
      <p>Make sure you have an empty analysis history.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p><strong>Rename your history</strong> to make it easy to recognize</p>

      <blockquote class="tip">
        <h3 id="tip-rename-a-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Rename a history</h3>

        <ul>
          <li>
            <p>Click on the title of the history (by default the title is <code class="highlighter-rouge">Unnamed history</code>)</p>

            <p><img src="../../../../shared/images/rename_history.png" alt="Renaming history" /></p>
          </li>
          <li>Type <code class="highlighter-rouge">Galaxy Introduction</code> as the name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ul>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="data-upload">Data upload</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>Download the list of peak regions (the file <a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE37268&amp;format=file&amp;file=GSE37268%5Fmof3%2Eout%2Ehpeak%2Etxt%2Egz"><code class="highlighter-rouge">GSE37268_mof3.out.hpeak.txt.gz</code></a>) from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE37268">GEO</a> to your computer</li>
    <li>
      <p>Click on the upload button in the upper left ot the interface</p>

      <p><img src="../../../galaxy-data-manipulation/images/upload_button.png" alt="Upload icon" /></p>
    </li>
    <li>Press <strong>Choose local file</strong> and search for your file on your computer</li>
    <li>Select <code class="highlighter-rouge">interval</code> as <strong>Type</strong></li>
    <li>Press <strong>Start</strong></li>
    <li>Press <strong>Close</strong></li>
    <li>
      <p>Wait for the upload to finish. Galaxy will automatically unpack the file.</p>
    </li>
    <li>
      <p>After this you will see your first history item in Galaxy’s right pane. It will go through
the gray (preparing/queued) and yellow (running) states to become green (success):</p>

      <p><img src="../../images/intro_01.png" alt="History section" /></p>

      <p>Directly uploading files is not the only way to get data into Galaxy</p>
      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the Galaxy Upload Manager (<i class="fa fa fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Change <strong>Type</strong> from “Auto-detect” to <code class="highlighter-rouge">interval</code></p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, Galaxy uses the URL as the name, so rename the files with a more useful name.</p>
      </blockquote>

      <blockquote class="tip">
        <h3 id="tip-tip-importing-data-to-galaxy"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data to Galaxy</h3>
        <p>There are <a href="/training-material/topics/galaxy-data-manipulation/tutorials/get-data/slides.html">more options</a> for advanced users.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-interval-file-format"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Interval file format</h3>
  <p><strong>Interval</strong> format is a Galaxy format for representing genomic intervals. It is tab-separated, but has the added requirement that three of the columns must be:</p>
  <ul>
    <li>chromosome ID</li>
    <li>start position (0-based)</li>
    <li>end position (end-exclusive)</li>
  </ul>

  <p>An optional strand column can also be specified, and an initial header row can be used to label the columns, which do not have to be in any special order. Unlike BED format (see below) arbitrary additional columns can also be present.</p>

  <p>You can find more information about formats that can be used in Galaxy at the <a href="https://usegalaxy.org/static/formatHelp.html">Galaxy Data Formats page</a>.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-and-edit-attributes-of-a-file"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect and edit attributes of a file</h3>

  <ol>
    <li>
      <p>Click on the file in the history panel</p>

      <p>Some meta-information (e.g. format, reference database) about the file and the header of the file are then displayed, along with the number of lines in the file (48,647):</p>

      <p><img src="../../images/input_expanded_file.png" alt="Expanded file in history" /></p>
    </li>
    <li>
      <p>Click on the <i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon (<strong>View data</strong>) in your dataset in the history</p>

      <p>The content of the file is displayed in the central panel</p>
    </li>
    <li>
      <p>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (pencil) icon (<strong>Edit attributes</strong>) in your dataset in the history</p>

      <p>A form to edit dataset attributes is displayed in the central panel</p>
    </li>
    <li>Search for <code class="highlighter-rouge">mm9</code> in <strong>Database/Build</strong> attribute and select <code class="highlighter-rouge">Mouse July 2007 (NCBI37/mm9)</code> (the paper tells us the peaks are from <code class="highlighter-rouge">mm9</code>)</li>
    <li>Click on <strong>Save</strong> on the top</li>
    <li>Add a tag called <code class="highlighter-rouge">#peaks</code> to the dataset to make it easier to track in the history
      <blockquote class="tip">

        <h3 id="tip-tip-adding-a-tag"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Adding a tag</h3>
        <ul>
          <li>Click on the dataset</li>
          <li>Click on <i class="fa fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <strong>Edit dataset tags</strong></li>
          <li>
            <p>Add a tag starting with <code class="highlighter-rouge">#</code></p>

            <p>Tags starting with <code class="highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>
          </li>
          <li>Check that the tag is appearing below the dataset name</li>
        </ul>

      </blockquote>

      <p>The dataset should now look like below in the history</p>

      <p><img src="../../images/input_tagged_file.png" alt="Peaks file" width="250px" height="300px" /></p>
    </li>
  </ol>

</blockquote>

<p>In order to find the related genes to these peak regions,
we also need a list of genes in mice, which we can obtain from UCSC.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-from-ucsc"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload from UCSC</h3>

  <ol>
    <li>
      <p>Search for <code class="highlighter-rouge">UCSC Main</code> in the tool search bar (top left)</p>

      <p><img src="../../images/101_01.png" alt="UCSC Main tool in tools section" /></p>
    </li>
    <li>
      <p>Click on <code class="highlighter-rouge">UCSC Main</code> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

      <p>You will be taken to the <strong>UCSC table browser</strong>, which looks something like this:</p>

      <p><img src="../../images/intro_02.png" alt="UCSC table browser interface" /></p>
    </li>
    <li>Set the following options:
      <ul>
        <li><em>“clade”</em>: <code class="highlighter-rouge">Mammal</code></li>
        <li><em>“genome”</em>: <code class="highlighter-rouge">Mouse</code></li>
        <li><em>“assembly”</em>: <code class="highlighter-rouge">July 2007 (NCBI37/mm9)</code></li>
        <li><em>“group”</em>: <code class="highlighter-rouge">Genes and Gene Predictions</code></li>
        <li><em>“track”</em>: <code class="highlighter-rouge">RefSeq Genes</code></li>
        <li><em>“table”</em>: <code class="highlighter-rouge">refGene</code></li>
        <li><em>“region”</em>: <code class="highlighter-rouge">genome</code></li>
        <li><em>“output format”</em>: <code class="highlighter-rouge">BED - browser extensible data</code></li>
        <li><em>“Send output to”</em>: <code class="highlighter-rouge">Galaxy</code> (only)</li>
      </ul>
    </li>
    <li>
      <p>Click on the <strong>get output</strong> button</p>

      <p>You will see the next screen:</p>

      <p><img src="../../images/intro_03.png" alt="Output settings" /></p>
    </li>
    <li>Make sure that <em>“Create one BED record per”</em> is set to <code class="highlighter-rouge">Whole Gene</code></li>
    <li>Click on the <strong>Send Query to Galaxy</strong> button</li>
    <li>Wait for the upload to finish</li>
    <li>
      <p>Rename our dataset to something more recognizable like <code class="highlighter-rouge">Genes</code></p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field  to <code class="highlighter-rouge">Genes</code></li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>Add a tag called <code class="highlighter-rouge">#genes</code> to the dataset to make it easier to track in the history</li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-bed-file-format"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> BED file format</h3>
  <p>The <strong>BED - Browser Extensible Data</strong> format provides a flexible way to encode gene regions. BED lines have three required fields:</p>
  <ul>
    <li>chromosome ID</li>
    <li>start position (0-based)</li>
    <li>end position (end-exclusive)</li>
  </ul>

  <p>There can be up to and nine additional optional fields, but the number of fields per line must be consistent throughout any single set of data.</p>

  <p>You can find more information about it at <a href="https://genome.ucsc.edu/FAQ/FAQformat#format1">UCSC</a> including a description of the optional fields.</p>
</blockquote>

<p>Now we have collected all the data we need to start our analysis.</p>

<h1 id="part-1-naive-approach">Part 1: Naive approach</h1>

<p>We will first use a “naive” approach to try to identify the genes that the peak regions are associated with. We will identify genes that overlap at least 1bp with the peak regions.</p>

<h2 id="file-preparation">File preparation</h2>

<p>Let’s have a look at our files to see what we actually have here.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-view-file-content"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: View file content</h3>

  <ol>
    <li>
      <p>Click on the <i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon (<strong>View data</strong>) of the peak file to view the content of it</p>

      <p>It should look like this:</p>

      <p><img src="../../images/intro_04.png" alt="Contents of the peak file" /></p>
    </li>
    <li>
      <p>View the content of the regions of the genes from UCSC</p>

      <p><img src="../../images/intro_05.png" alt="Contents of UCSC file" /></p>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>While the file from UCSC has labels for the columns, the peak file does not. Can you guess what the columns stand for?</p>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>This peak file is not in any standard format and just by looking at it, we cannot find out what the numbers in the different columns mean. In the paper the authors mention that they used the peak caller <a href="https://www.ncbi.nlm.nih.gov/pubmed/20598134">HPeak</a>.</p>

    <p>By looking at the HPeak manual we can find out that the columns contain the following information:</p>

    <ul>
      <li>chromosome name by number</li>
      <li>start coordinate</li>
      <li>end coordinate</li>
      <li>length</li>
      <li>location within the peak that has the highest hypothetical DNA fragment coverage (summit)</li>
      <li>not relevant</li>
      <li>not relevant</li>
    </ul>

  </blockquote>
</blockquote>

<p>In order to compare the two files, we have to make sure that the chromosome names follow the same format.
As we can see, the peak file lacks <code class="highlighter-rouge">chr</code> before any chromosome number. But what happens with chromosome 20 and 21? Will it be X and Y instead? Let’s check:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-view-end-of-file"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: View end of file</h3>

  <ol>
    <li>Search for <strong>Select last</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool and run <strong>Select last lines from a dataset (tail)</strong> with the following settings:
      <ul>
        <li><em>“Text file”</em>: our peak file <code class="highlighter-rouge">GSE37268_mof3.out.hpeak.txt.gz</code></li>
        <li><em>“Operation”</em>: <code class="highlighter-rouge">Keep last lines</code></li>
        <li><em>“Number of lines”</em>: Choose a value, e.g. <code class="highlighter-rouge">100</code></li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong></li>
    <li>Wait for the job to finish</li>
    <li>
      <p>Inspect the file through the <i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon (<strong>View data</strong>)</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How are the chromosomes named?</li>
          <li>How are the chromosomes X and Y named?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>The chromosomes are just given by their number. In the gene file from UCSC, they started with <code class="highlighter-rouge">chr</code></li>
            <li>The chromosomes X and Y are named 20 and 21</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>In order to convert the chromosome names we have therefore two things to do:</p>

<ol>
  <li>add <code class="highlighter-rouge">chr</code></li>
  <li>change 20 and 21 to X and Y</li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-adjust-chromosome-names"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adjust chromosome names</h3>

  <ol>
    <li><strong>Replace Text</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>Replace Text in a specific column</strong> with the following settings:
      <ul>
        <li><em>“File to process”</em>: our peak file <code class="highlighter-rouge">GSE37268_mof3.out.hpeak.txt.gz</code></li>
        <li><em>“in column”</em>: <code class="highlighter-rouge">Column:1</code></li>
        <li>
          <p><em>“Find pattern”</em>: <code class="highlighter-rouge">[0-9]+</code></p>

          <p>This will look for numerical digits</p>
        </li>
        <li>
          <p><em>“Replace with”</em>: <code class="highlighter-rouge">chr&amp;</code></p>

          <p><code class="highlighter-rouge">&amp;</code> is a placeholder for the find result of the pattern search</p>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename your output file <code class="highlighter-rouge">chr prefix added</code>.</p>
    </li>
    <li><strong>Replace Text</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Let’s rerun the tool with
      <ul>
        <li><em>“File to process”</em>: the output from the last run, <code class="highlighter-rouge">chr prefix added</code></li>
        <li><em>“in column”</em>: <code class="highlighter-rouge">Column:1</code></li>
        <li><em>“Find pattern”</em>: <code class="highlighter-rouge">chr20</code></li>
        <li><em>“Replace with”</em>: <code class="highlighter-rouge">chrX</code></li>
      </ul>

      <blockquote class="tip">
        <h3 id="tip-tip-rerunning-a-tool"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Rerunning a tool</h3>

        <ul>
          <li>Expand the dataset information</li>
          <li>Press the <i class="fa fa-refresh" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> icon (<strong>Run this job again</strong>)</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Rename your output file <code class="highlighter-rouge">chrX fixed</code></p>
    </li>
    <li><strong>Replace Text</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Rerun this tool to do the same for chromosome Y
      <ul>
        <li><em>“File to process”</em>: <code class="highlighter-rouge">chrX fixed</code>, the output from the last run</li>
        <li><em>“in column”</em>: <code class="highlighter-rouge">Column:1</code></li>
        <li><em>“Find pattern”</em>: <code class="highlighter-rouge">chr21</code></li>
        <li><em>“Replace with”</em>: <code class="highlighter-rouge">chrY</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the latest file through the <i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon. Have we been successful?</p>

      <p>We have quite a few files now and need to take care to select the correct ones at each step.</p>

      <blockquote class="question">
        <h3 id="question-questions-2"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>How many regions are in our output file? You can click the name of the output to expand it and see the number.</p>

        <blockquote class="solution">
          <h3 id="solution-solution-2"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>It should be equal to the number of regions in your first file, <code class="highlighter-rouge">GSE37268_mof3.out.hpeak.txt.gz</code>: 48,647
If yours says 100 regions, then you have run it on the <code class="highlighter-rouge">Tail</code> file and need to re-run the steps.</p>
        </blockquote>
      </blockquote>
    </li>
    <li>Rename the file to something more recognizable, e.g. <code class="highlighter-rouge">Peak regions</code></li>
  </ol>
</blockquote>

<h2 id="analysis">Analysis</h2>

<p>Our goal is to compare the 2 region files (the genes file and the peak file from the publication)
to know which peaks are related to which genes. If you only want to know which peaks are located <strong>inside</strong> genes (within the gene body) you
can skip the next step. Otherwise, it might be reasonable to include the <strong>promoter</strong> region of the genes into the comparison, e.g. because
you want to include transcriptions factors in ChIP-seq experiments. There is no strict definition for promoter region but 2kb upstream of the TSS (start of region) is commonly used. We’ll use the <strong>Get Flanks</strong> tool to get regions 2kb bases upstream of the start of the gene to 10kb bases downstream of the start (12kb in length). To do this we tell the Get Flanks tool we want regions upstream of the start, with an offset of 10kb, that are 12kb in length, as shown in the diagram below.</p>

<p><img src="../../images/intro_get_flanks.png" alt="Get Flanks" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-add-promoter-region-to-gene-records"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Add promoter region to gene records</h3>

  <ol>
    <li><strong>Get Flanks</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>Get flanks returns flanking region/s for every gene</strong> with the following settings:
      <ul>
        <li><em>“Select data”</em>: <code class="highlighter-rouge">Genes</code> file from UCSC</li>
        <li><em>“Region”</em>: <code class="highlighter-rouge">Around Start</code></li>
        <li><em>“Location of the flanking region/s”</em>: <code class="highlighter-rouge">Upstream</code></li>
        <li><em>“Offset”</em>: <code class="highlighter-rouge">10000</code></li>
        <li><em>“Length of the flanking region(s)”</em>: <code class="highlighter-rouge">12000</code></li>
      </ul>

      <p>This tool returns flanking regions for every gene</p>
    </li>
    <li>
      <p>Compare the rows of the resulting BED file with the input to find out how the start and end positions changed</p>

      <blockquote class="tip">
        <h3 id="tip-tip-inspecting-several-files-using-the-scratchbook"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Inspecting several files using the scratchbook</h3>

        <ul>
          <li>
            <p>Click <strong>Enable/Disable Scratchbook</strong> on the top panel</p>

            <p><img src="../../images/intro_scratchbook_enable.png" alt="Enable/Disable Scratchbook" /></p>
          </li>
          <li>Click on the <i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon of the files to inspect</li>
          <li>
            <p>Click on <strong>Show/Hide Scratchbook</strong></p>

            <p><img src="../../images/intro_scratchbook_show_hide.png" alt="Show/Hide Scratchbook" /></p>
          </li>
        </ul>
      </blockquote>
    </li>
    <li>Rename your dataset to reflect your findings (<code class="highlighter-rouge">Promoter regions</code>)</li>
  </ol>
</blockquote>

<p>The output is regions that start from 2kb upstream of the TSS and include 10kb downstream. For input regions on the positive strand e.g. <code class="highlighter-rouge">chr1 134212701 134230065</code> this gives <code class="highlighter-rouge">chr1 134210701 134222701</code>. For regions on the negative strand e.g. <code class="highlighter-rouge">chr1 8349819 9289958</code> this gives <code class="highlighter-rouge">chr1  9279958 9291958</code>.</p>

<p>You might have noticed that the UCSC file is in <code class="highlighter-rouge">BED</code> format and has a database associated to it. That’s what we want for our peak file as well. The <strong>Intersect</strong> tool we will use can automatically convert interval files to BED format but we’ll convert our interval file explicitly here to show how this can be achieved with Galaxy.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-change-format-and-database"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Change format and database</h3>

  <ol>
    <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (pencil) icon in the history entry of our peak region file</li>
    <li>Switch to the <strong>Convert</strong> tab</li>
    <li>Select <code class="highlighter-rouge">Convert Genomic Intervals to BED</code></li>
    <li>Press <strong>Convert datatype</strong></li>
    <li>Check that the “Database/Build” is <code class="highlighter-rouge">mm9</code> (the database build for mice used in the paper)</li>
    <li>Again rename the file to something more recognizable, e.g. <code class="highlighter-rouge">Peak regions BED</code></li>
  </ol>
</blockquote>

<p>It’s time to find the overlapping intervals (finally!). To do that, we want to extract the genes which overlap/intersect with our peaks.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-find-overlaps"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Find Overlaps</h3>

  <ol>
    <li><strong>Intersect</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>Intersect the intervals of two datasets</strong> with the following settings:
      <ul>
        <li><em>“Return”</em>: <code class="highlighter-rouge">Overlapping Intervals</code></li>
        <li><em>“of”</em>: the UCSC file with promoter regions (<code class="highlighter-rouge">Promoter regions</code>)</li>
        <li><em>“that intersect”</em>: our peak region file from <strong>Replace</strong> (<code class="highlighter-rouge">Peak regions BED</code>)</li>
        <li><em>“for at least”</em>: <code class="highlighter-rouge">1</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comments"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
        <p>The order of the inputs is important! We want to end up with a list of <strong>genes</strong>, so the corresponding dataset with the gene information needs to be the first input (<code class="highlighter-rouge">Promoter regions</code>).</p>
      </blockquote>
      <p><img src="../../images/intro_overlapping_genes.png" alt="Genes overlapping peaks" /></p>
    </li>
  </ol>
</blockquote>

<p>We now have the list of genes (column 4) overlapping with the peak regions, similar to shown above.</p>

<p>To get a better overview of the genes we obtained, we want to look at their distribution across the different chromosomes.
We will group the table by chromosome and count the number of genes with peaks on each chromosome</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-count-genes-on-different-chromosomes"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Count genes on different chromosomes</h3>

  <ol>
    <li><strong>Group</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>Group data by a column and perform aggregate operation on other columns</strong> with the following settings:
      <ul>
        <li><em>“Select data”</em> to the result of the intersection</li>
        <li><em>“Group by column”</em>:<code class="highlighter-rouge">Column 1</code></li>
        <li>Press <strong>Insert Operation</strong> and choose:
          <ul>
            <li><em>“Type”</em>: <code class="highlighter-rouge">Count</code></li>
            <li><em>“On column”</em>: <code class="highlighter-rouge">Column 1</code></li>
            <li><em>“Round result to nearest integer?”</em>: <code class="highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-3"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>Which chromosome contained the highest number of target genes?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-3"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>The result varies with different settings, for example, the annotation may change due to updates at UCSC. If you followed step by step, with the same annotation, it should be chromosome 11 with 1992 genes. Note that for reproducibility, you should keep all input data used because Galaxy can store all parameters but inputs may change e.g. the annotation from UCSC.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="visualization">Visualization</h2>

<p>Since we have some nice data, let’s draw a barchart out of it!</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-draw-barchart"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Draw barchart</h3>

  <ol>
    <li>Click on <i class="fa fa-bar-chart" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> (visualize) icon on the output from the <strong>Group</strong> tool</li>
    <li>Select <code class="highlighter-rouge">Bar diagram</code></li>
    <li>Choose a title at <strong>Provide a title</strong>, e.g. <code class="highlighter-rouge">Gene counts per chromosome</code></li>
    <li>Switch to the <i class="fa fa fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Select data</strong> tab and play around with the settings</li>
    <li>
      <p>When you are happy, click the <i class="fa fa fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span> <strong>Save</strong> visualization in the top right of the <em>main panel</em></p>

      <p>This will store it to your saved visualisations where you can later view, download, or share it with others.</p>
    </li>
  </ol>

</blockquote>

<h2 id="extracting-workflow">Extracting workflow</h2>

<p>When you look carefully at your history, you can see that it contains all steps of our analysis, from the beginning to the end. By building this history we have actually built a complete record of our analysis with Galaxy preserving all parameter settings applied at every step.
Wouldn’t it be nice to just convert this history into a workflow that we’ll be able to execute again and again?</p>

<p>Galaxy makes this very simple with the <code class="highlighter-rouge">Extract workflow</code> option. This means that any time you want to build a workflow, you can just perform it manually once, and then convert it to a workflow, so that next time it will be a lot less work to do the same analysis. It also allows you to easily share or publish your analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-workflow"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract workflow</h3>

  <ol>
    <li>
      <p><strong>Clean up</strong> your history</p>

      <p>If you had any failed jobs (red), please remove those datasets from your history by clicking on the <i class="fa fa-times" aria-hidden="true"></i><span class="visually-hidden">galaxy-cross</span> (cross) button. This will make the creation of a workflow easier.</p>
    </li>
    <li>Go to the history <strong>History options</strong> (<i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (gear) icon)</li>
    <li>
      <p>Select the <code class="highlighter-rouge">Extract Workflow</code> option.</p>

      <p><img src="../../images/history_menu_extract_workflow.png" alt="Extracting workflow in history menu" /></p>

      <p>The central panel will change and you will be able to choose which steps to include/exclude and how to name the newly created workflow.</p>
    </li>
    <li>
      <p><strong>Uncheck</strong> any steps that shouldn’t be included in the workflow</p>

      <p>Since we did some steps which where specific to our custom peak file, we might want to exclude:</p>
      <ul>
        <li><strong>Select last</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>all <strong>Replace Text</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> steps</li>
        <li><strong>Convert Genomic Intervals to BED</strong></li>
        <li><strong>Get flanks</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li>
      <p>Rename the workflow to something descriptive, e.g. <code class="highlighter-rouge">From peaks to genes</code></p>
    </li>
    <li>
      <p>Click on the <strong>Create Workflow</strong> button near the top.</p>

      <p>You will get a message that the workflow was created. But where did it go?</p>
    </li>
    <li>
      <p>Click on <strong>Workflow</strong> in the top menu of Galaxy</p>

      <p>Here you have a list of all your workflows</p>
    </li>
    <li>
      <p>Select the newly generated workflow and click on <strong>Edit</strong></p>

      <p>You should see something similar to this:</p>

      <p><img src="../../images/intro_06.png" alt="Editing workflow interface" /></p>

      <blockquote class="comment">
        <h3 id="comment-the-workflow-editor"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> The workflow editor</h3>
        <p>We can examine the workflow in Galaxy’s workflow editor. Here you can view/change the parameter settings of each step, add and remove tools, and connect an output from one tool to the input of another, all in an easy and graphical manner. You can also use this editor to build workflows from scratch.</p>
      </blockquote>

      <p>Although we have our two inputs in the workflow they are missing their connection to the first tool (<strong>Intersect</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>), because we didn’t carry over some of the intermediate steps.</p>
    </li>
    <li>Connect each input dataset to the <strong>Intersect</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool by dragging the arrow pointing outwards on the right of its box (which denotes an output) to an arrow on the left of the <strong>Intersect</strong> box pointing inwards (which denotes an input)</li>
    <li>Rename the input datasets to <code class="highlighter-rouge">Reference regions</code> and <code class="highlighter-rouge">Peak regions</code></li>
    <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (gear) icon at the top right</li>
    <li>Press <strong>Auto Re-layout</strong> to clean up our view
<img src="../../images/intro_07.png" alt="Auto re-layouting" /></li>
    <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (gear) icon at the top right</li>
    <li>Press <strong>Save</strong> to save your changes</li>
  </ol>

  <blockquote class="tip">
    <h3 id="tip-tip-hiding-intermediate-steps"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Hiding intermediate steps</h3>
    <p>When a workflow is executed, the user is usually primarily interested in the final product and not in all intermediate steps. By default all the outputs of a workflow will be shown, but we can explicitly tell Galaxy which output to show and which to hide for a given workflow. This behaviour is controlled by the little asterisk next to every output dataset:</p>

    <p><img src="../../../../shared/images/workflow_editor_mark_output.png" alt="Workflow editor mark output" /></p>

    <p>If you click on this asterisk for any of the output datasets, then <em>only</em> files with an asterisk will be shown, and all outputs without an asterisk will be hidden (Note that clicking <em>all</em> outputs has the same effect as clicking <em>none</em> of the outputs, in both cases all the datasets will be shown).</p>
  </blockquote>

</blockquote>

<p>Now it’s time to reuse our workflow for a more sophisticated approach.</p>

<h1 id="part-2-more-sophisticated-approach">Part 2: More sophisticated approach</h1>

<p>In part 1 we used an overlap definition of 1 bp (default setting) to identify genes associated with the peak regions. However, the peaks could be broad, so instead, in order to get a more meaningful definition, we could identify the genes that overlap where most of the reads are concentrated, the <strong>peak summit</strong>. We will use the information on the position of the peak summit contained in the original peak file and check for overlap of the summits with genes.</p>

<h2 id="preparation">Preparation</h2>

<p>We again need our peak file, but we’d like to work in a clean history. Instead of uploading it twice, we can copy it to a new history.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-copy-history-items"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Copy history items</h3>

  <ol>
    <li>
      <p>Create a new history and give it a new name like <code class="highlighter-rouge">Galaxy Introduction Part 2</code></p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history-1"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Click on the <strong>View all histories</strong> (<i class="fa fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> icon) at the top right of your history</p>

      <p>You should see both of your histories side-by-side now</p>
    </li>
    <li>Drag and drop the edited peak file (<code class="highlighter-rouge">Peak regions</code>, after the replace steps), which contains the summit information, to your new history.</li>
    <li>Click on <strong>Analyze Data</strong> in the top menu bar to go back to your analysis window</li>
  </ol>

</blockquote>

<h2 id="create-peak-summit-file">Create peak summit file</h2>

<p>We need to generate a new BED file from the original peak file that contains the positions of the peak summits. The start of the summit is the start of the peak (column 2) plus the location within the peak that has the highest hypothetical DNA fragment coverage (column 5). As the end of the peak region, we will simply define <code class="highlighter-rouge">start + 1</code>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-peak-summit-file"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create peak summit file</h3>

  <ol>
    <li><strong>Compute an expression on every row</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Add expression”</em>: <code class="highlighter-rouge">c2+c5</code></li>
        <li><em>“as a new column to”</em>: our peak file <code class="highlighter-rouge">Peak regions</code> (the interval format file)</li>
        <li><em>“Round result?”</em>: <code class="highlighter-rouge">YES</code></li>
      </ul>
    </li>
  </ol>

  <p>This will create an 8th column in our table, which we will use in our next step:</p>

  <ol>
    <li>
      <p>Rename the output <code class="highlighter-rouge">Peak regions new column</code></p>
    </li>
    <li><strong>Compute an expression on every row</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: rerun this tool on the last result with:
      <ul>
        <li><em>“Add expression”</em>: <code class="highlighter-rouge">c8+1</code></li>
        <li><em>“as a new column to”</em>: the <code class="highlighter-rouge">Peak regions new column</code> file we just created</li>
        <li><em>“Round result?”</em>: <code class="highlighter-rouge">YES</code></li>
      </ul>
    </li>
    <li>Rename this file <code class="highlighter-rouge">Peak summit regions</code></li>
  </ol>
</blockquote>

<p>Now we cut out just the chromosome plus the start and end of the summit:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-cut-out-columns"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Cut out columns</h3>
  <ol>
    <li><strong>Cut</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>Cut columns from a table</strong> with the following settings:
      <ul>
        <li><em>“Cut columns”</em>: <code class="highlighter-rouge">c1,c8,c9</code></li>
        <li><em>“Delimited by Tab”</em>: <code class="highlighter-rouge">Tab</code></li>
        <li><em>“From”</em>: <code class="highlighter-rouge">Peak summit regions</code></li>
      </ul>

      <p>The output from <strong>Cut</strong> will be in <code class="highlighter-rouge">tabular</code> format.</p>
    </li>
    <li>
      <p>Change the format to <code class="highlighter-rouge">interval</code> (use the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span>) since that’s what the tool <strong>Intersect</strong> expects.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fa fa fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="highlighter-rouge">interval</code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>

      <p>The output should look like below:</p>

      <p><img src="../../images/intro_summits.png" alt="Peak summits" width="200px" /></p>
    </li>
  </ol>
</blockquote>

<h2 id="get-gene-names">Get gene names</h2>

<p>The RefSeq genes we downloaded from UCSC did only contain the RefSeq identifiers, but not the gene names. To get a list of gene names in the end, we use another BED file from the Data Libraries.</p>

<blockquote class="comment">
  <h3 id="comment-comments-1"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>There are several ways to get the gene names in, if you need to do it yourself. One way is to retrieve a mapping through Biomart and then join the two files (<strong>Join two Datasets side by side on a specified field</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>). Another is to get the full RefSeq table from UCSC and manually convert it to BED format.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-1"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>From <a href="https://zenodo.org/record/1025586">Zenodo</a> or from the data library, import the file <code class="highlighter-rouge">mm9.RefSeq_genes_from_UCSC.bed</code></p>

      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1025586/files/mm9.RefSeq_genes_from_UCSC.bed
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links-1"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the Galaxy Upload Manager (<i class="fa fa fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Change <strong>Genome</strong> to <code class="highlighter-rouge">mm9</code></p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, Galaxy uses the URL as the name, so rename the files with a more useful name.</p>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Click on “Training data” and then “Introduction - From peaks to genes”</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>

      <p>As default, Galaxy takes the link as name, so rename them.</p>
    </li>
    <li>
      <p>Inspect the file content to check if it contains gene names.
It should look similar to below:
<img src="../../images/intro_gene_names.png" alt="Gene names" /></p>
    </li>
    <li>Rename it <code class="highlighter-rouge">mm9.RefSeq_genes</code></li>
    <li>Apply the tag <code class="highlighter-rouge">#genes</code></li>
  </ol>

</blockquote>

<h2 id="repeat-workflow">Repeat workflow</h2>

<p>It’s time to reuse the workflow we created earlier.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-run-a-workflow"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run a workflow</h3>
  <ol>
    <li>Open the workflow menu (top menu bar)</li>
    <li>Find the workflow you made in the previous section, and select the option <strong>Run</strong></li>
    <li>Choose as inputs our <code class="highlighter-rouge">mm9.RefSeq_genes</code> (<code class="highlighter-rouge">#genes</code>) BED file and the result of the <strong>Cut</strong> tool (<code class="highlighter-rouge">#peaks</code>)</li>
    <li>
      <p>Click <strong>Run workflow</strong></p>

      <p>The outputs should appear in the history but it might take some time until they are finished.</p>
    </li>
  </ol>

</blockquote>

<p>We used our workflow to rerun our analysis with the peak summits. The <strong>Group</strong> tool again produced a list containing the number of genes found in each chromosome.
But wouldn’t it be more interesting to know the number of peaks in each unique gene? Let’s rerun the workflow with different settings!</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-run-a-workflow-with-changed-settings"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run a workflow with changed settings</h3>
  <ol>
    <li>Open the workflow menu (top menu bar)</li>
    <li>Find the workflow you made in the previous section, and select the option <strong>Run</strong></li>
    <li>Choose as inputs our <code class="highlighter-rouge">mm9.RefSeq_genes</code> (<code class="highlighter-rouge">#genes</code>) BED file and the result of the <strong>Cut</strong> tool (<code class="highlighter-rouge">#peaks</code>)</li>
    <li>Click on the title of the <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Group</strong> tool to expand the options.</li>
    <li>Change the following settings by clicking on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (pencil) icon on the left:
      <ul>
        <li><em>“Group by column”</em>: <code class="highlighter-rouge">7</code></li>
        <li>In <em>“Operation”</em>:
          <ul>
            <li><em>“On column”</em>: <code class="highlighter-rouge">7</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click <strong>Run workflow</strong></li>
  </ol>
</blockquote>

<p>Congratulations! You should have a file with all the unique gene names and a count on how many peaks they contained.</p>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>The list of unique genes is not sorted. Try to sort it on your own!</p>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>You can use the tool “Sort data in ascending or descending order” on column 2 and “fast numeric sort”.</p>
  </blockquote>
</blockquote>

<h1 id="share-your-work">Share your work</h1>

<p>One of the most important features of Galaxy comes at the end of an analysis. When you have published striking findings, it is important that other researchers are able to reproduce your in-silico experiment. Galaxy enables users to easily share their workflows and histories with others.</p>

<p>To share a history, click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (gear) symbol in the history pane and select <code class="highlighter-rouge">Share or Publish</code>. On this page you can do 3 things:</p>

<ol>
  <li>
    <p><strong>Make accessible via Link</strong></p>

    <p>This generates a link that you can give out to others. Anybody with this link will be able to view your history.</p>
  </li>
  <li>
    <p><strong>Publish History</strong></p>

    <p>This will not only create a link, but will also publish your history. This means your history will be listed under <code class="highlighter-rouge">Shared Data → Published Histories</code> in the top menu.</p>
  </li>
  <li>
    <p><strong>Share with Individual Users</strong></p>

    <p>This will share the history only with specific users on the Galaxy instance.</p>
  </li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-share-history-and-workflow"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Share history and workflow</h3>

  <ol>
    <li>Share one of your histories with your neighbour</li>
    <li>See if you can do the same with your workflow!</li>
    <li>
      <p>Find the history and/or workflow shared by your neighbour</p>

      <p>Histories shared with specific users can be accessed by those users in their history menu (<i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (gear) icon) under <code class="highlighter-rouge">Histories shared with me</code>.</p>
    </li>
  </ol>

</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p><i class="fa fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> You have just performed your first analysis in Galaxy. You also created a workflow from your analysis so you can easily repeat the exact same analysis on other datasets. Additionally you shared your results and methods with others.</p>



                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>Galaxy provides an easy-to-use graphical user interface for often complex commandline tools</li>
                            
                            <li>Galaxy keeps a full record of your analysis in a history</li>
                            
                            <li>Workflows enable you to repeat your analysis on different data</li>
                            
                            <li>Galaxy can connect to external sources for data import and visualization purposes</li>
                            
                            <li>Galaxy provides ways to share your results and methods with others</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Li2012">Li, X., L. Li, R. Pandey, J. S. Byun, K. Gardner <i>et al.</i>, 2012 <b>The Histone Acetyltransferase MOF Is a Key Regulator of the Embryonic Stem Cell Core Transcriptional Network</b>. Cell Stem Cell 11: 163–178.</span>
    
    
    <a href="https://doi.org/10.1016/j.stem.2012.04.023">10.1016/j.stem.2012.04.023</a>
    
    

</li></ol>
                    

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

        <hr>
        <br>

        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=From peaks to genes (Introduction to Galaxy Analyses)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=From peaks to genes (Introduction to Galaxy Analyses)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Anne Pajon, Clemens Blank, Bérénice Batut, Björn Grüning, Nicola Soranzo, Dilmurat Yusuf, Sarah Peter, Helena Rasche)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
