<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Peptide and Protein ID using OpenMS tools</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:title" content="Galaxy Training: Peptide and Protein ID using OpenMS tools" />
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Proteomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fprotein-id-oms%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fprotein-id-oms%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fprotein-id-oms%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fprotein-id-oms%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fprotein-id-oms%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/proteomics/tutorials/protein-id-oms/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Peptide and Protein ID using OpenMS tools",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "proteomics / protein-id-oms / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Peptide and Protein ID using OpenMS tools' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "Peptide and Protein ID using OpenMS tools",
    "description": "Peptide and Protein ID using OpenMS tools",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html",
  "timeRequired": "PT45M",
  "description": "Peptide and Protein ID using OpenMS tools\\nThe questions this  addresses are:\n - How to convert LC-MS/MS raw files?\n - How to identify peptides?\n - How to identify proteins?\n - How to evaluate the results?\n\n\\nThe objectives are:\n - Protein identification from LC-MS/MS raw files.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html",
      "name": "Peptide and Protein Quantification via Stable Isotope Labelling (SIL)",
      "description": "Hands-on for 'Peptide and Protein Quantification via Stable Isotope Labelling (SIL)' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/546301",
      "name": "Training data for Peptide and Protein ID using OpenMS tools tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Florian Christoph Sigloch"
    },
    {
      "@type": "Person",
      "name": "Björn Grüning"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Florian Christoph Sigloch"
    },
    {
      "@type": "Person",
      "name": "Björn Grüning"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://galaxyproject.github.io//training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Peptide and Protein ID using OpenMS tools</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#stortebecker" class="contributor-badge"><img src="https://avatars.githubusercontent.com/stortebecker" alt="Florian Christoph Sigloch">Florian Christoph Sigloch</a>, <a href="/training-material/hall-of-fame#bgruening" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bgruening" alt="Björn Grüning">Björn Grüning</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>How to convert LC-MS/MS raw files?</li>
            
            <li>How to identify peptides?</li>
            
            <li>How to identify proteins?</li>
            
            <li>How to evaluate the results?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Protein identification from LC-MS/MS raw files.</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Peptide and Protein Quantification via Stable Isotope Labelling (SIL):
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 45 minutes</p>
            

            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    

                    
                    <div>
                        
<a class="topic-icon" href="https://zenodo.org/record/546301">
    <i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>


                    </div>
                    

                    
                    <div>
                        
    <a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/protein-id-oms/workflows/" title="Workflows" alt="Peptide and Protein ID using OpenMS tools workflows">
        <i class="fa fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>


                    </div>
                    
                    
                    <div>
                        
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/protein-id-oms/tours/">
    <i class="fa fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>


                    </div>
                    


                    
                    
                    <div>
                        

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">instances</span> Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fa fa-ship" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>



                    </div>
                    
                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Identifying the proteins contained in a sample is an important step in any proteomic experiment. However, in most settings, proteins are digested to peptides prior to LC-MS/MS analysis. In this so-called “bottom-up” procedure, only peptide masses are measured. Therefore, protein identification cannot be performed directly from raw data, but is a multi-step process:</p>

<ol>
  <li>Raw data preparations</li>
  <li>Peptide-to-Spectrum matching</li>
  <li>Peptide inference</li>
  <li>Protein inference</li>
</ol>

<p>A plethora of different software solutions exists for each step. In this tutorial, we will use <strong><em>msconvert</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>  for raw data conversion and tools from the <a href="https://openms.de">OpenMS software suite</a> for all other steps. We will use one peptide search engine at first and later on show how to expand the workflow for using multiple search engines. Protein inference will be performed with the Fido algorithm (<a href="https://www.ncbi.nlm.nih.gov/pubmed/20712337">Serang et al, JPR, (2010)</a>).
This tutorial covers peptide and protein <strong>identification</strong> only, but you may use the output of this tutorial for the <a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">tutorial on protein quantitation</a>.</p>

<p>For an alternative protein ID workflow using the <a href="https://compomics.com/">Compomics</a> tools <a href="https://compomics.github.io/projects/searchgui.html">SearchGUI</a> and <a href="https://compomics.github.io/projects/peptide-shaker.html">PeptideShaker</a>, please consult <a href="/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html">this tutorial</a>.
The latter tutorial does not allow to continue with the tutorial on protein quantitation.</p>

<h1 class="no_toc" id="input-data">Input data</h1>

<p>As an example dataset, we will use an LC-MS/MS analysis of HeLa cell lysate published
in <a href="https://www.ncbi.nlm.nih.gov/pubmed/24678044">Vaudel et al., 2014, Proteomics</a>. Detailed information
about the dataset can be found on <a href="https://www.ebi.ac.uk/pride/archive/projects/PXD000674">PRIDE</a>.
For step 2 we will use a validated human Uniprot FASTA database with appended decoys.</p>

<p>If you already completed the tutorial on <a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">Database Handling</a>
you can use the constructed database including decoys.
You can find a prepared database, as well as the input LC-MS/MS data in different file formats on <a href="https://zenodo.org/record/796184">Zenodo</a>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#preparing-raw-data" id="markdown-toc-preparing-raw-data">Preparing Raw Data</a></li>
  <li><a href="#peptide-identification" id="markdown-toc-peptide-identification">Peptide Identification</a></li>
  <li><a href="#peptide-fdr-filtering" id="markdown-toc-peptide-fdr-filtering">Peptide FDR filtering</a></li>
  <li><a href="#protein-inference" id="markdown-toc-protein-inference">Protein Inference</a></li>
  <li><a href="#analysis-of-contaminants" id="markdown-toc-analysis-of-contaminants">Analysis of Contaminants</a></li>
  <li><a href="#using-multiple-search-engines" id="markdown-toc-using-multiple-search-engines">Using multiple search engines</a></li>
  <li><a href="#premade-workflow" id="markdown-toc-premade-workflow">Premade Workflow</a></li>
  <li><a href="#further-reading" id="markdown-toc-further-reading">Further Reading</a></li>
</ol>

</blockquote>

<h1 id="preparing-raw-data">Preparing Raw Data</h1>

<p>Raw data conversion is the first step of any proteomic data analysis. The most common converter is MSConvert from the <a href="http://proteowizard.sourceforge.net/">ProteoWizard software suite</a>, the format to convert to is mzML.
Due to licensing reasons, MSConvert runs only on windows systems and will not work on most Galaxy servers.</p>

<p>Depending on your machine settings, raw data will be generated either in profile mode or centroid mode. For most peptide search engines, the MS2 data have to be converted to centroid mode, a process called “peak picking” or “centroiding”.
Machine vendors offer algorithms to extract peaks from profile raw data. Those are integrated in <strong><em>msconvert</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and can be run in parallel to the mzML conversion.
However, the OpenMS tool <strong><em>PeakPickerHiRes</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is reported to generate slightly better results (<a href="https://www.ncbi.nlm.nih.gov/pubmed/17094243">Lange et al., 2006, Pac Symp Biocomput</a>) and is therefore recommended for quantitative studies (<a href="https://www.ncbi.nlm.nih.gov/pubmed/19953549">Vaudel et al., 2010, Proteomics</a>).
If your data were generated on a low resolution mass spectrometer, use <strong>PeakPickerWavelet</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> instead.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-file-conversion-and-peak-picking"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-On: File Conversion and Peak Picking</h3>

  <p>We provide the <a href="https://zenodo.org/record/796184">input data</a> in the original <code class="highlighter-rouge">raw</code> format and also already converted to <code class="highlighter-rouge">mzML</code>. If <strong>msconvert</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> does not run on your Galaxy instance, please download the preconverted <code class="highlighter-rouge">mzML</code> as an input and continue with step 5 of the following hands-on training.</p>

  <ol>
    <li>
      <p>Create a new history for this Peptide and Protein ID exercise.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Load one of the example datasets into your history from Zenodo</p>

      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/892005/files/qExactive01819.raw
https://zenodo.org/record/892005/files/qExactive01819_profile.mzml
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the Galaxy Upload Manager (<i class="fa fa fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, Galaxy uses the URL as the name, so rename the files with a more useful name.</p>
      </blockquote>
    </li>
    <li>
      <p>Rename the dataset to something meaningful</p>
    </li>
    <li>Run <strong>msconvert</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the training <code class="highlighter-rouge">raw</code> file to convert to the <code class="highlighter-rouge">mzML</code> format
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input unrefined MS data”</em>: imported <code class="highlighter-rouge">raw</code> file</li>
        <li><em>“Do you agree to the vendor licenses”</em>: set to <code class="highlighter-rouge">Yes</code></li>
        <li><em>“Output Type”</em>: set to <code class="highlighter-rouge">mzML</code></li>
      </ul>
    </li>
    <li>Run <strong>PeakPickerHiRes</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the resulting file
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> *“input profile data file”: output of <strong>msconvert</strong> or <code class="highlighter-rouge">mzML</code> file</li>
        <li>In <em>“param_algorithm_ms_levels”</em>
          <ul>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click on <em>“Insert param_algorithm_ms_levels”</em></li>
            <li>In <em>“1: param_algorithm_ms_levels”</em>
              <ul>
                <li>
                  <p><em>“List of MS levels for which the peak picking is applied”</em>: <code class="highlighter-rouge">2</code></p>

                  <p>Peak picking will only be performed on MS2 level.</p>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-comment-local-use-of-msconvert"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Local Use of MSConvert</h3>
    <p>The vendor libraries used by MSConvert are only licensed for Windows systems and are therefore rarely implemented in Galaxy instances. If <strong>msconvert</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is not available in your Galaxy instance, please install the software on a Windows computer and run the conversion locally. You can find a detailed description of the necessary steps <a href="https://compomics.com/bioinformatics-for-proteomics/identification/">here</a> (“Peak List Generation”). Afterwards, upload the resulting mzML file to your Galaxy history.</p>
  </blockquote>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-comment-ms2-peak-picking-during-data-acquisition"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: MS2 peak picking during data acquisition</h3>
  <p>MS2 peaks are often acquired in centroided mode in first place. The profile data are converted to centroided mode already during data acquisition, resulting in MS2-centroided <code class="highlighter-rouge">raw</code> files. If your MS2 data are already centroided, simply omit the peak picking step.</p>
</blockquote>

<h1 id="peptide-identification">Peptide Identification</h1>

<p>MS/MS experiments identify peptides by isolating them and subsequently colliding them with a gas for fragmentation. This method generates a spectrum of peptide fragment masses for each isolated peptide - an MS2 spectrum.
To find out the peptide sequences, the MS2 spectrum is compared to a theoretical spectrum generated from a protein database. This step is called peptide-to-spectrum (also: spectrum-to-sequence) matching. Accordingly, a peptide that is successfully matched to a sequence is termed PSM (Peptide-Spectrum-Match). There can be multiple PSMs per peptide, if the peptide was fragmented several times.</p>

<p>Different peptide search engines have been developed to fulfill the matching procedure. Here, we will use the search engine <a href="https://www.ncbi.nlm.nih.gov/pubmed/14976030">X!Tandem</a>. OpenMS provides “adapters” (wrappers) for several other peptide search engines, like MSGF+ or OMSSA. You may replace the XTandemAdapter by another search engine of your choice.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-peptide-identification"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-On: Peptide Identification</h3>

  <ol>
    <li>Copy the prepared protein database (Human database including cRAP contaminants and decoys) from the tutorial <a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">Database Handling</a> into your current history by using the multiple history view</li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-you-did-not-run-the-database-handling-first"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> You did not run the Database Handling first?</h3>
    <p>You can upload the ready-made database from Zenodo</p>

    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/892005/files/Human_database_including_decoys_%28cRAP_added%29.fasta
</code></pre></div>    </div>
  </blockquote>

  <ol>
    <li>Run the tool <strong>XTandemAdapter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file containing MS2 spectra”</em>: MS2-centroided mzML file</li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA file”</em>: FASTA protein database</li>
        <li><em>“Fragment mass error”</em>: <code class="highlighter-rouge">10</code></li>
        <li><em>“Fragment monoisotopic mass error units”</em>: <code class="highlighter-rouge">ppm</code></li>
      </ul>
      <ul>
        <li>In <em>“param_fixed_modifications”</em>
          <ul>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click on <em>“Insert param_fixed_modifications”</em></li>
            <li>In <em>“1: param_fixed_modifications”</em>
              <ul>
                <li><em>“Fixed modifications”</em>: <code class="highlighter-rouge">Carbamidomethyl (C)</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“param_variable_modifications”</em>
          <ul>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click on <em>“Insert param_variable_modifications”</em></li>
            <li>In <em>“1: param_variable_modifications”</em>
              <ul>
                <li><em>“Variable modifications”</em>: <code class="highlighter-rouge">Oxidation (M)</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run the tool <strong>FileInfo</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: <strong>XTandem</strong> output</li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-settings-for-labelled-data"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Settings for labelled data</h3>
  <p>Several common quantitation methods are based on labels e.g. SILAC, TMT, iTRAQ, Dimethyl. Those labels must be specified in the search engine as a (variable) modification. See also <a href="/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">Peptide and Protein Quantification via Stable Isotope Labelling</a>.</p>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-comment-advanced-search-engine-parameters"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Advanced Search Engine Parameters</h3>
  <p>The OpenMS adapters do not always allow to set every option of the underlying search engine. If an option is missing, you may also run the search engine locally or by using a Galaxy wrapper. Afterwards, convert the search engine output to the OpenMS format <code class="highlighter-rouge">idXML</code> by running <strong>IDFileConverter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

  <p>The search engine X!Tandem features some more advanced options than the ones reflected in the <strong>XTandemAdapter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. If you need those advanced options, the <strong>XTandemAdapter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> allows for the optional input of a classic X!Tandem parameter file. Upload your parameter file to the history and use it as an input in the field <code class="highlighter-rouge">Default X!Tandem configuration file</code>. You may also set the option <em>“ignore_adapter_param”</em> to <code class="highlighter-rouge">Yes</code> to overwrite all options set by the GUI.</p>
</blockquote>

<h1 id="peptide-fdr-filtering">Peptide FDR filtering</h1>

<p>The next step of peptide identification is to decide which PSMs will be used for protein inference. Measured MS2 spectra never perfectly fit the theoretical spectra. Therefore, peptide search engines calculate a score which indicates how well the measured MS2 spectrum was fitting the theoretical spectrum. How do we decide which PSMs are likely true and which are false?</p>

<p>In proteomics, this decision is typically done by calculating false discovery rates (FDRs). Remember that the database we were using for peptide-to-spectrum matching consisted not only of true proteins, but also the same number of “fake entries”, the so-called decoys. Those decoys can now be used to estimate the number of false identifications in the list of PSMs.
The calculation is based on a simple assumption: for every decoy peptide identified with a given score, we expect one false positive with at least the same score.
The false discovery rate is therefore defined as the number of false discoveries (decoy hits) divided by the number of false and correct discoveries (both target and decoy hits) at a given score threshold.</p>

<p>We will calculate peptide posterior error probabilities (PEPs), because they are needed for the protein inference algorithm used by OpenMS. To calculate FDRs, we first have to annotate the identified peptides to determine which of them are decoys. This is done with the tool <strong>PeptideIndexer</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. This tool allows the annotation of protein names and calculates the protein coverage based on the identified peptides. We will then filter PSMs for 1 % FDR and set the score back to PEP.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-peptide-fdr-filtering"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-On: Peptide FDR filtering</h3>

  <ol>
    <li>Run <strong>IDPosteriorErrorProbability</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: <strong>XTandemAdapter</strong> output</li>
        <li><em>“If set scores will be calculated as ‘1 - ErrorProbabilities’ and can be interpreted as probabilities for correct identifications”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Run <strong>PeptideIndexer</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input idXML file containing the identifications”</em>: output of <strong>IDPosteriorErrorProbability</strong></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input sequence database in FASTA format”</em>: FASTA protein database</li>
        <li><em>“If set, the protein sequences are stored as well”</em>: <code class="highlighter-rouge">Yes</code></li>
        <li><em>“If set, the protein description is stored as well”</em>: <code class="highlighter-rouge">Yes</code></li>
        <li><em>“Specificity of the enzyme”</em>: <code class="highlighter-rouge">none</code></li>
      </ul>
    </li>
    <li>Run <strong>FalseDiscoveryRate</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Identifications from searching a target-decoy database”</em>: output of <strong>PeptideIndexer</strong></li>
        <li><em>“Perform FDR calculation on protein level”</em>: <code class="highlighter-rouge">false</code></li>
        <li><em>“Filter PSMs based on q-value”</em>: <code class="highlighter-rouge">0.01</code></li>
        <li><em>“If ‘true’ decoy peptides will be written to output file, too”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Run <strong>IDScoreSwitcher</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: output of <strong>FalseDiscoveryRate</strong></li>
        <li><em>“Name of the meta value to use as the new score”</em>: <code class="highlighter-rouge">Posterior Probability_score</code></li>
        <li><em>“Orientation of the new score”</em>: <code class="highlighter-rouge">higher_better</code></li>
      </ul>
    </li>
    <li>Run <strong>FileInfo</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to get basic information about the identified peptides
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: <strong>IDScoreSwitcher</strong> output</li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions:</h3>
  <ol>
    <li>How many peptides were identified?</li>
    <li>How many peptides with oxidized methionine were identified?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>You should have identified 3366 non-redundant peptide hits.</li>
      <li>798 peptides contain an oxidized methionine.</li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="protein-inference">Protein Inference</h1>
<p>In bottom-up proteomics, it is necessary to combine the identified peptides to proteins. This is not a trivial task, as proteins are redundant to some degree. Thus, not every peptide can be assigned to only one protein.
The OpenMS suite implemented the <a href="https://www.ncbi.nlm.nih.gov/pubmed/20712337">Fido</a> algorithm for protein inference. Fido uses a Bayesian probabilistic model to group and score proteins based on peptide-spectrum matches.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-protein-inference"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-On: Protein inference</h3>

  <ol>
    <li>Run <strong>FidoAdapter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input: identification results”</em>: output of <strong>IDScoreSwitcher</strong></li>
        <li><em>“Post-process Fido output with greedy resolution of shared peptides based on the protein probabilities”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Run <strong>FalseDiscoveryRate</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Identifications from searching a target-decoy database”</em>: output of <strong>FidoAdapter</strong></li>
        <li><em>“Perform FDR calculation on PSM level”</em>: <code class="highlighter-rouge">false</code></li>
      </ul>
    </li>
    <li>Run <strong>IDFilter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: output of <strong>FalseDiscoveryRate</strong></li>
        <li><em>“The score which should be reached by a protein hit to be kept”</em>: <code class="highlighter-rouge">0.01</code></li>
      </ul>
    </li>
    <li>Run <strong>FileInfo</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to get basic information about the identified proteins
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: output of <strong>IDFilter</strong></li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-comment-greedy-group-resolution"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: “Greedy” Group Resolution</h3>
  <p>Protein groups are reported, when an identified peptide maps to multiple proteins in the used database <a href="https://www.ncbi.nlm.nih.gov/pubmed/16009968">Nesvizhskii and Aebersold (2005)</a>. Some peptides may map to different protein groups and can therefore not be used for protein quantitation. The option <code class="highlighter-rouge">-greedy_group_resolution</code> solves this problem by assigning peptides only to the one most probable protein group, thus enabling to quantify proteins based not only on unique, but also on shared peptides. This usually leads to a much higher number of quantified proteins. However it will introduce noise in the FCs when a peptide was indeed shared by different proteins and the quantity of this peptide was a weighted sum of contributions. The greedy group resolution is similar to Occam’s razor.</p>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions:</h3>
  <p>How many proteins were finally identified?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>You should have identified 1255 proteins.</p>
  </blockquote>
</blockquote>

<h1 id="analysis-of-contaminants">Analysis of Contaminants</h1>

<p>The FASTA database used for the peptide to spectrum matching contained some entries that were not expected to stem from the HeLa cell lysate, but are common contaminations in LC-MS/MS samples. The main reason to add those is to avoid false assignment of contaminant spectra to other proteins.
It also enables you to check for contaminations in your samples.</p>

<p><strong>CAVE:</strong> When analyzing human samples, many proteins that are common contaminants may also stem from the sample. Therefore, human contaminants do not have to be excluded from further analysis, but you should keep in mind that the source of these proteins is unclear.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-analysis-of-contaminants"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-On: Analysis of Contaminants</h3>

  <ol>
    <li>Run <strong>TextExporter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to convert the idXML output to a human-readable tabular file.
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <strong>IDFilter</strong> output</li>
      </ul>
    </li>
    <li>Run <strong>Select lines that match an expression</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <strong>TextExporter</strong></li>
        <li><em>“that”</em>: <code class="highlighter-rouge">Matching</code></li>
        <li><em>“the pattern”</em>: <code class="highlighter-rouge">CONTAMINANT</code></li>
      </ul>
    </li>
    <li>Run <strong>Select lines that match an expression</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to remove all non human proteins (e.g. bovine)
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <strong>TextExporter</strong></li>
        <li><em>“that”</em>: <code class="highlighter-rouge">Matching</code></li>
        <li><em>“the pattern”</em>: <code class="highlighter-rouge">HUMAN</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <ol>
    <li>Which contaminants did you identify? Where do these contaminations likely come from?</li>
    <li>What other sources of contaminants exist?</li>
    <li>How many false positives do we expect in our list?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>TRY1_BOVIN is bovine trypsin. It was used to degrade the proteins to peptides. ALBU_BOVIN is bovine serum albumin. It is added to cell culture medium in high amounts. Also, eight human proteins are listed, these are commonly introduced during sample preparation. As we were analyzing a human sample, it is not neccessary to remove these proteins, as they may as well originate from the HeLa cells.</li>
      <li>Contaminants often stem from the experimenter, these are typically keratins or other high-abundant human proteins. Basically any protein present in the room of the mass spectrometer might get into the ion source, if it is airborne. As an example, sheep keratins are sometimes found in proteomic samples, stemming from clothing made of sheep wool.</li>
      <li>As we were allowing for a false discovery rate of 1 %, we would expect 13 (1255/0.01) false positive proteins in our list.</li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="using-multiple-search-engines">Using multiple search engines</h1>

<p>It is generally recommended to use more than one peptide search engine and use the combined results for peptide inference (<a href="https://www.ncbi.nlm.nih.gov/pubmed/23720762">Shteynberg et al., 2013, Mol. Cell. Proteomics</a>).
By comparing results of multiple search engines, you may improve the <em>sensitivity</em> (when accepting peptides that were found by only one of the engines), the <em>specificity</em> (when accepting only peptides that were found by all of the search engines) or <em>both</em> (when using n&gt;2 search engines and accept peptides found by a fraction of the (e.g. n-1) search engines).</p>

<p>Here, we will use the OpenMS tool <a href="https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/latest/html/TOPP_ConsensusID.html">ConsensusID</a> to combine the search engine results.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-multiple-search-engines"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-On: Multiple search engines</h3>

  <ol>
    <li>Run <strong>MSGFPlusAdapter</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: the MS2-centroided mzML</li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Protein sequence database</em>”: the FASTA protein database</li>
        <li><em>“Precursor monoisotopic mass tolerance”</em>: <code class="highlighter-rouge">10.0</code></li>
        <li><em>“Instrument that generated the data”</em>: <code class="highlighter-rouge">Q_Exactive</code></li>
        <li>In <em>“param_fixed_modifications”</em>
          <ul>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click on <em>“Insert param_fixed_modifications”</em></li>
            <li>In <em>“1: param_fixed_modifications”</em>
              <ul>
                <li><em>“Fixed modifications”</em>: <code class="highlighter-rouge">Carbamidomethyl (C)</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“param_variable_modifications”</em>
          <ul>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click on <em>“Insert param_variable_modifications”</em></li>
            <li>In <em>“1: param_variable_modifications”</em>
              <ul>
                <li><em>“Variable modifications”</em>: <code class="highlighter-rouge">Oxidation (M)</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <strong>FileInfo</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to get basic information about the identified proteins
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: <strong>MSGFPlusAdapter</strong> output</li>
      </ul>
    </li>
    <li>Run <strong>IDPosteriorErrorProbability</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: <strong>MSGFPlusAdapter</strong> output</li>
        <li><em>“If set scores will be calculated as ‘1 - ErrorProbabilities’ and can be interpreted as probabilities for correct identifications”</em>: <code class="highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li>Run <strong>IDMerger</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><em>“Reduce collectons”</em>: <code class="highlighter-rouge">reduce collections by aggregating single files of multiple collections</code>
          <ul>
            <li>In <em>“inputs”</em>
              <ul>
                <li>In <em>“1: Repeat”</em>
                  <ul>
                    <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input files separated by blanks”</em>: output of <strong>IDScoreSwitcher</strong> based on <strong>XTandemAdapter</strong></li>
                  </ul>
                </li>
                <li>In <em>“2: Repeat”</em>
                  <ul>
                    <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input files separated by blanks”</em>: output of <strong>IDScoreSwitcher</strong> based on <strong>MSGFPlusAdapter</strong></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <strong>ConsensusID</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: <strong>IDMerger</strong> output</li>
      </ul>
    </li>
    <li>Run <strong>PeptideIndexer</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input idXML file containing the identifications”</em>: <strong>ConsensusID</strong> output</li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input sequence database in FASTA format”</em>: FASTA protein database</li>
        <li><em>“Specificity of the enzyme”</em>: <code class="highlighter-rouge">none</code></li>
      </ul>
    </li>
    <li>Run <strong>FalseDiscoveryRate</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Identifications from searching a target-decoy database”</em>: output of <strong>PeptideIndexer</strong></li>
        <li><em>“Perform FDR calculation on protein level”</em>: <code class="highlighter-rouge">false</code></li>
        <li><em>“Filter PSMs based on q-value”</em>: <code class="highlighter-rouge">0.01</code></li>
        <li><em>“If ‘true’ decoy peptides will be written to output file, too”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Run <strong>IDScoreSwitcher</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <strong>FalseDiscoveryRate</strong> output</li>
        <li><em>“Name of the meta value to use as the new score”</em>: <code class="highlighter-rouge">Posterior Error Probability_score</code></li>
        <li><em>“Orientation of the new score”</em>: <code class="highlighter-rouge">lower_better</code></li>
        <li><em>“Name to use as the type of the new score”</em>: <code class="highlighter-rouge">Posterior Error Probability</code></li>
      </ul>
    </li>
    <li>Run <strong>FileInfo</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to get basic information about the identified peptides
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: <strong>IDScoreSwitcher</strong> output</li>
      </ul>
    </li>
    <li>Proceed with the protein inference as described <a href="#protein-inference">above</a></li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions:</h3>
  <ol>
    <li>How many PSMs could be matched with XTandem and MSGFPlus alone? How many peptides were identified?</li>
    <li>How many PSMs could be matched after combining the results with ConsensusID? How many peptides were identified?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>After FDR-filtering, XTandem matched 3,552 PSMs (2,616 unique peptides) and MSGFPlus matched 4,292 PSMs (2,991 peptides).</li>
      <li>Combining the results with ConsensusID leads to matching of 4,299 PSMs (3,041 unique peptides).</li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="premade-workflow">Premade Workflow</h1>

<p>A premade workflow for this tutorial can be found <a href="/training-material/topics/proteomics/tutorials/protein-id-oms/workflows/workflow.ga">here</a>.</p>

<p>A premade workflow using the search engines XTandem and MSGF+ can be found <a href="/training-material/topics/proteomics/tutorials/protein-id-oms/workflows/workflow_two-search-engines.ga">here</a>.</p>

<h1 id="further-reading">Further Reading</h1>

<ul>
  <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/16009968">Protein inference</a></li>
  <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/20712337">Fido publication</a></li>
  <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/27498275">Evaluation of protein inference algorithms</a></li>
</ul>



                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>LC-MS/MS raw files have to be converted to mzML before using GalaxyP on most GalaxyP servers.</li>
                            
                            <li>OpenMS provides many tools for proteomic analysis and guarantees compatibility by using open file formats.</li>
                            
                            <li>OpenMS provides several thirdparty search engines and Fido for protein inference.</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                    

                    

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

        <hr>
        <br>

        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Peptide and Protein ID using OpenMS tools (Proteomics)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=Peptide and Protein ID using OpenMS tools (Proteomics)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Florian Christoph Sigloch, Björn Grüning)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/protein-id-oms/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
