<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Proteogenomics 2: Database Search</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:title" content="Galaxy Training: Proteogenomics 2: Database Search" />
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Proteomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Proteogenomics 2: Database Search",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "proteomics / proteogenomics-dbsearch / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Proteogenomics 2: Database Search' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "Proteogenomics 2: Database Search",
    "description": "Proteogenomics 2: Database Search",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html",
  "timeRequired": "PT15M",
  "description": "Proteogenomics 2: Database Search\\nThe questions this  addresses are:\n - How to identify variant proteoforms in MS data by searching with the customized Protein database?\n\n\\nThe objectives are:\n - A proteogenomic data analysis of mass spectrometry data to identify and visualize variant peptides.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html",
      "name": "Proteogenomics 1: Database Creation",
      "description": "Hands-on for 'Proteogenomics 1: Database Creation' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.1489208",
      "name": "Training data for Proteogenomics 2: Database Search tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "James Johnson"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "James Johnson"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://galaxyproject.github.io//training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Proteogenomics 2: Database Search</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#subinamehta" class="contributor-badge"><img src="https://avatars.githubusercontent.com/subinamehta" alt="Subina Mehta">Subina Mehta</a>, <a href="/training-material/hall-of-fame#timothygriffin" class="contributor-badge"><img src="https://avatars.githubusercontent.com/timothygriffin" alt="Timothy J. Griffin">Timothy J. Griffin</a>, <a href="/training-material/hall-of-fame#pratikdjagtap" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pratikdjagtap" alt="Pratik Jagtap">Pratik Jagtap</a>, <a href="/training-material/hall-of-fame#jraysajulga" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jraysajulga" alt="Ray Sajulga">Ray Sajulga</a>, <a href="/training-material/hall-of-fame#jj-umn" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jj-umn" alt="James Johnson">James Johnson</a>, <a href="/training-material/hall-of-fame#pravs3683" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pravs3683" alt="Praveen Kumar">Praveen Kumar</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>How to identify variant proteoforms in MS data by searching with the customized Protein database?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>A proteogenomic data analysis of mass spectrometry data to identify and visualize variant peptides.</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Proteogenomics 1: Database Creation:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 15 minutes</p>
            

            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    

                    
                    <div>
                        
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1489208">
    <i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>


                    </div>
                    

                    
                    <div>
                        
    <a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/proteogenomics-dbsearch/workflows/" title="Workflows" alt="Proteogenomics 2: Database Search workflows">
        <i class="fa fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>


                    </div>
                    
                    


                    
                    
                    <div>
                        

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden">instances</span> Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fa fa-ship" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>



                    </div>
                    
                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In this tutorial, we perform proteogenomic database searching using the Mass Spectrometry data. The inputs for performing the proteogenomic database searching are the peaklist MGF files and the FASTA database file. The FASTA database is obtained by running the first workflow “Uniprot_cRAP_SAV_indel_translatedbed.FASTA”. The second workflow focuses on performing database search of the peak list files (MGFs).</p>

<p><img src="../../images/second_workflow.png" alt="Workflow" /></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>
<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a></li>
  <li><a href="#match-peptide-sequences" id="markdown-toc-match-peptide-sequences">Match peptide sequences</a>    <ol>
      <li><a href="#searchgui" id="markdown-toc-searchgui">SearchGUI</a></li>
      <li><a href="#peptideshaker" id="markdown-toc-peptideshaker">PeptideShaker</a></li>
    </ol>
  </li>
  <li><a href="#create-database" id="markdown-toc-create-database">Create database</a></li>
  <li><a href="#remove-known-peptides" id="markdown-toc-remove-known-peptides">Remove known peptides</a>    <ol>
      <li><a href="#merged-uniprot-and-crap-database" id="markdown-toc-merged-uniprot-and-crap-database">Merged Uniprot and cRAP database</a></li>
      <li><a href="#text-manipulation-steps" id="markdown-toc-text-manipulation-steps">Text Manipulation steps</a></li>
      <li><a href="#query-tabular" id="markdown-toc-query-tabular">Query Tabular</a></li>
    </ol>
  </li>
  <li><a href="#blastp-basic-local-alignment-search-tool--proteins" id="markdown-toc-blastp-basic-local-alignment-search-tool--proteins">BLASTP (Basic Local Alignment Search Tool- proteins)</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<blockquote class="no_toc hands_on">
  <h3 id="hands_on-hands-on-data-upload-and-organization"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: data upload and organization</h3>

  <ol>
    <li>Create a <strong>new history</strong> and name it something meaningful (e.g. <em>Proteogenomics DB search</em>)</li>
    <li>Import the four MGF MS/MS files and the FASTA sequence file from Zenodo.<a href="https://doi.org/10.5281/zenodo.1489208"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.1489208.svg" alt="DOI" /></a>
      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1489208/files/Mo_Tai_Trimmed_mgfs__Mo_Tai_iTRAQ_f4.mgf
https://zenodo.org/record/1489208/files/Mo_Tai_Trimmed_mgfs__Mo_Tai_iTRAQ_f5.mgf
https://zenodo.org/record/1489208/files/Mo_Tai_Trimmed_mgfs__Mo_Tai_iTRAQ_f8.mgf
https://zenodo.org/record/1489208/files/Mo_Tai_Trimmed_mgfs__Mo_Tai_iTRAQ_f9.mgf
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the Galaxy Upload Manager (<i class="fa fa fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, Galaxy uses the URL as the name, so rename the files with a more useful name.</p>
      </blockquote>
    </li>
    <li>Rename the datasets to something more recognizable (strip the URL prefix)</li>
    <li>
      <p>Build a <strong>Dataset list</strong> for the three MGF files, name it something recognizable, for example <em>MGF files</em></p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-dataset-collection"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a dataset collection</h3>

        <ul>
          <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="../../../galaxy-data-manipulation/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>
          <li>Check all the datasets in your history you would like to include</li>
          <li>
            <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>

            <p><img src="/training-material/topics/galaxy-data-manipulation/images/buildList.png" alt="build list collection menu item" width="15%" /></p>
          </li>
          <li>Enter a name for your collection</li>
          <li>Click <strong>Create List</strong> to build your collection</li>
          <li>Click on the checkmark icon at the top of your history again</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="match-peptide-sequences">Match peptide sequences</h1>

<p>The search database labelled <code class="highlighter-rouge">Uniprot_cRAP_SAV_indel_translatedbed.FASTA</code> is the input database that
will be used to match MS/MS to peptide sequences via a sequence database search.</p>

<p>For this, the sequence database-searching program called <a href="https://compomics.github.io/projects/searchgui.html">SearchGUI</a> will be used.The generated dataset collection of the three <em>MGF files</em> in the history is used as the MS/MS input. We will walk through a number of these settings in order to utilize SearchGUI on these example MGF files.</p>

<h2 id="searchgui">SearchGUI</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-searchgui"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: SearchGUI</h3>

  <ol>
    <li><strong>SearchGUI</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Protein Database”</em>: <code class="highlighter-rouge">Uniprot_cRAP_SAV_indel_translatedbed.FASTA</code> (Or however you named the <code class="highlighter-rouge">FASTA</code> file)</li>
        <li>
          <p><i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input Peak lists (mgf)”</em>: <code class="highlighter-rouge">MGF files</code> dataset collection.</p>

          <blockquote class="tip">
            <h3 id="tip-tip-select-dataset-collections-as-input"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Select dataset collections as input</h3>

            <ul>
              <li>
                <p>Click the <strong>Dataset collection</strong> icon on the left of the input field:</p>
              </li>
              <li>
                <p>Select the appropriate dataset collection from the list</p>
              </li>
            </ul>
          </blockquote>
        </li>
        <li>Section <strong>Search Engine Options</strong>:
          <ul>
            <li>
              <p><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“B-Search Engines”</em>: <code class="highlighter-rouge">X!Tandem</code></p>

              <blockquote class="comment">
                <h3 id="comment-comment"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment:</h3>
                <p>The section <strong>Search Engine Options</strong> contains a selection of sequence database searching
   programs that are available in SearchGUI. Any combination of these programs can be used for
   generating PSMs from MS/MS data. For the purpose of this tutorial, <strong>X!Tandem</strong> we will be
   used.</p>
              </blockquote>
            </li>
          </ul>
        </li>
        <li>Section <strong>Protein Digestion Options</strong>:
          <ul>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Enzyme”</em>: <code class="highlighter-rouge">Trypsin</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Maximum Missed Cleavages”</em>: <code class="highlighter-rouge">2</code></li>
          </ul>
        </li>
        <li>Section <strong>Precursor options</strong>:
          <ul>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Precursor Ion Tolerance Units”</em>: <code class="highlighter-rouge">Parts per million (ppm)</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Precursor Ion Tolerance”</em>:<code class="highlighter-rouge"> 10</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Fragment Tolerance (Daltons)”</em>: <code class="highlighter-rouge">0.05</code> (this is high resolution MS/MS data)</li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Minimum charge”</em>:<code class="highlighter-rouge">2</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Maximum charge”</em>:<code class="highlighter-rouge">6</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Forward Ion”</em>: <code class="highlighter-rouge">b</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Reverse Ion”</em>:<code class="highlighter-rouge"> y</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Minimum Precursor Isotope”</em> :<code class="highlighter-rouge">0</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Maximum Precursor Isotope”</em> :<code class="highlighter-rouge">1</code></li>
          </ul>
        </li>
        <li>Section <strong>Protein Modification Options</strong>:
          <ul>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Fixed Modifications”</em>: <code class="highlighter-rouge">Carbamidomethylation of C, ITRAQ-4Plex of K, ITRAQ-4Plex of Ntermini</code></li>
            <li>
              <p><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Variable modifications”</em>: <code class="highlighter-rouge">Oxidation of M, ITRAQ-4Plex of Y</code></p>

              <blockquote class="tip">
                <h3 id="tip-tip-search-for-options"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Search for options</h3>
                <p>For selection lists, typing the first few letters in the window will filter the
available options.</p>
              </blockquote>
            </li>
          </ul>
        </li>
        <li>Section <strong>Advanced Options</strong>:
          <ul>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>*“X!Tandem Options”</em>: <code class="highlighter-rouge">Advanced</code>
              <ul>
                <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>*“X!Tandem: Quick Acetyl”</em>: <code class="highlighter-rouge">No</code></li>
                <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>*“X!Tandem: Quick Pyrolidone”</em>: <code class="highlighter-rouge">No</code></li>
                <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>*“X!Tandem: Protein stP Bias”</em>: <code class="highlighter-rouge">No</code></li>
                <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>*“X!Tandem: Maximum Valid Expectation Value”</em>: <code class="highlighter-rouge">100</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Leave everything else as default</li>
        <li>Click <strong>Execute</strong></li>
      </ul>
    </li>
  </ol>

  <p>Once the database search is completed, the SearchGUI tool will output a file (called a
SearchGUI archive file) that will serve as an input for the next section, PeptideShaker.</p>

  <blockquote class="comment">
    <h3 id="comment-comment-1"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment:</h3>
    <p>Note that sequence databases used for proteogenomics are usually much larger than
the excerpt used in this tutorial. When using large databases, the peptide identification
step can take much more time for computation. In proteogenomics, choosing the optimal
database is a crucial step of your workflow.</p>
  </blockquote>

</blockquote>

<h2 id="peptideshaker">PeptideShaker</h2>

<p><a href="https://compomics.github.io/projects/peptide-shaker.html">PeptideShaker</a> is a post-processing software tool that
processes data from the SearchGUI software tool. It serves to organize the Peptide-Spectral
Matches (PSMs) generated from SearchGUI processing and is contained in the SearchGUI archive.
It provides an assessment of confidence of the data, inferring proteins identified from the
matched peptide sequences and generates outputs that can be visualized by users to interpret
results. PeptideShaker has been wrapped in Galaxy to work in combination with SearchGUI
outputs.</p>

<blockquote class="comment">
  <h3 id="comment-comment-file-formats"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: File Formats</h3>
  <p>There are a number of choices for different data files that can be generated using
  PeptideShaker. A compressed file can be made containing all information needed to
  view them results in the standalone PeptideShaker viewer. A <code class="highlighter-rouge">mzidentML</code> file can
  be generated that contains all peptide sequence matching information and can be
  utilized by compatible downstream software. Other outputs are focused on the inferred
  proteins identified from the PSMs, as well as phosphorylation reports, relevant if
  a phosphoproteomics experiment has been undertaken.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-peptideshaker"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: PeptideShaker</h3>

  <ol>
    <li><strong>PeptideShaker</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Compressed SearchGUI results”</em>: The SearchGUI archive file</li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Specify Advanced PeptideShaker Processing Options”</em>: <code class="highlighter-rouge">Default Processing Options</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Specify Advanced Filtering Options”</em>: <code class="highlighter-rouge">Default Filtering Options</code></li>
        <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Include the protein sequences in mzIdentML”</em>: <code class="highlighter-rouge">No</code></li>
        <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Output options”</em>: Select the <code class="highlighter-rouge">PSM Report</code> (Peptide-Spectral Match) and the <code class="highlighter-rouge">Certificate of Analysis</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-certificate-of-analysis"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Certificate of Analysis</h3>

        <p>The “Certificate of Analysis” provides details on all the parameters
used by both SearchGUI and PeptideShaker in the analysis. This can be downloaded from the
Galaxy instance to your local computer in a text file if desired.</p>

      </blockquote>
    </li>
    <li>Inspect the resulting files after they turned green with the <strong>View data</strong> icon</li>
  </ol>

</blockquote>

<p>A number of new items will appear in your history, each corresponding to the outputs selected
in the PeptideShaker parameters. Most relevant for this tutorial is the PSM report:</p>

<p>Scrolling at the bottom to the left will show the sequence for the PSM that matched to these
peptide entries. Column 3 is the sequence matched for each PSM entry. Every PSM is a
new row in the tabular output.</p>

<p>A number of new items will appear in your History, each corresponding to the outputs selected in the PeptideShaker parameters. The Peptide Shaker’s PSM report is used as an input for the BlastP analysis. Before performing BlastP analysis. The Query Tabular tool and few test manipulation tools are used to remove spectra that belongs to the reference proteins. The output tabular file “Peptides_for_Blast-P_analysis” will contain only those spectra that did not belong to any known proteins.</p>

<h1 id="create-database">Create database</h1>

<p>The mzidentml output from the Peptide shaker is converted into an sqlite database file by using the mz to sqlite tool. This sqlite output is used to open the Multi-omics visualization platform, wherein you can view the spectra of the peptides using Lorikeet parameters. To open the MVP viewer, click on the “Visualize in MVP Application” icon ( this will pop-open the interactive multi-omics viewer in a new window/tab)</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mz-to-sqlite"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: mz to sqlite</h3>

  <p>This tool extracts mzidentml and its associated proteomics datasets into a sqlite db</p>

  <ol>
    <li><strong>mz to sqlite</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Proteomics identification files”</em>: <code class="highlighter-rouge">PeptideShaker_mzidentml</code></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Proteomics Spectrum files”</em>: <code class="highlighter-rouge">Mo_Tai_MGFs</code></li>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Proteomics Search Database Fasta”</em>: <code class="highlighter-rouge">Uniprot_cRAP_SAV_indel_translatedbed.FASTA</code></li>
      </ul>

      <p><img src="../../images/mz2sqlite.png" alt="mz2sqlite" width="20%" /></p>
    </li>
  </ol>

</blockquote>

<p>The next step is to remove known peptides from the list of PSM’s that we acquired from the Peptide Shaker results. For that we need to perform some text manipulation steps to extract list of known peptides from the UniProt and cRAP database.</p>

<h1 id="remove-known-peptides">Remove known peptides</h1>

<h2 id="merged-uniprot-and-crap-database">Merged Uniprot and cRAP database</h2>

<p>The file named “Trimmed_ref_500_Uniprot_cRAP.fasta” is the trimmed version of Uniprot and cRAP database merged Fasta files.</p>

<p>This Fasta file will be subjected to few text manipulation steps in order to get the tabular file for the known peptides. The first step is to convert this FASTA file to tabular in order to proceed with text manipulation.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on--hands-on-fasta-to-tabular"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span>  Hands-on: FASTA to Tabular</h3>

  <ol>
    <li>Run <strong>FASTA to Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Data input ‘input’ (fasta)”</em>: <code class="highlighter-rouge">Trimmed_ref_500_Uniprot_cRAP.fasta</code></li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“How many columns to divide title string into?”</em>: <code class="highlighter-rouge">2</code></li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“How many title characters to keep?”</em>: <code class="highlighter-rouge">0</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The resultant tabular file will go through a series of text manipulation steps to make it suitable for
input to the Query Tabular tool.</p>

<h2 id="text-manipulation-steps">Text Manipulation steps</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-text-manipulation"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Text Manipulation</h3>

  <ol>
    <li><strong>Cut</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Cut Columns”</em>: <code class="highlighter-rouge">c1</code></li>
        <li><em>“Delimited by”</em>: <code class="highlighter-rouge">Tab</code></li>
      </ul>

      <p>Upon completion of this step you will have extracted C1 (column 1) from the input tabular file</p>
    </li>
    <li><strong>Convert</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Convert all”</em>: <code class="highlighter-rouge">Whitespaces</code></li>
        <li><em>“in Dataset”</em> : <code class="highlighter-rouge">Data input 'input' (txt)</code></li>
        <li><em>“Strip leading and trailing whitespaces”</em>: <code class="highlighter-rouge">Yes</code></li>
        <li><em>“Condense consecutive delimiters in one TAB”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>

      <p>This step will convert all the white spaces into different tabular column.</p>
    </li>
    <li><strong>Cut</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Cut Columns”</em>: <code class="highlighter-rouge">c2</code></li>
        <li><em>“Delimited by”</em>: <code class="highlighter-rouge">Pipe</code></li>
      </ul>

      <p>This step will extract information in column2 separated by Pipe</p>
    </li>
    <li><strong>Convert</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Convert all”</em>: <code class="highlighter-rouge">Dots</code></li>
        <li><em>“in Dataset”</em> : <code class="highlighter-rouge">Data input 'input' (txt)</code></li>
        <li><em>“Strip leading and trailing whitespaces”</em>: <code class="highlighter-rouge">Yes</code></li>
        <li><em>“Condense consecutive delimiters in one TAB”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>

      <p>This step will convert all the dots into different tabular column.</p>
    </li>
    <li><strong>Group</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Select data”</em>: <code class="highlighter-rouge">input from above</code></li>
        <li><em>“Group by column”</em>: <code class="highlighter-rouge">1</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Now that we have the list of known peptides, the query tabular tool is used to move these reference pepides from the PSM report.</p>

<h2 id="query-tabular">Query Tabular</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-query-tabular"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Query Tabular</h3>

  <ol>
    <li><strong>Query Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</li>
  </ol>

  <ul>
    <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong> (a):
      <ul>
        <li>Section <strong>Table Options</strong>:
          <ul>
            <li><em>“Tabular Dataset for Table”</em>: Uniprot</li>
            <li><em>“Use first line as column names”</em> : <code class="highlighter-rouge">No</code></li>
            <li><em>“Specify Column Names (comma-separated list)”</em>:<code class="highlighter-rouge">prot</code></li>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Table Index</strong>:
              <ul>
                <li><em>“Table Index”</em>: <code class="highlighter-rouge">No</code></li>
                <li><em>“Index on Columns”</em>: <code class="highlighter-rouge">Prot</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong> (b):
      <ul>
        <li>Section <strong>Filter Dataset Input</strong>:
          <ul>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
              <ul>
                <li><em>“Filter by”</em>:  <code class="highlighter-rouge">skip leading lines</code></li>
                <li><em>“Skip lines”</em>: <code class="highlighter-rouge">1</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Section <strong>Table Options</strong>:
          <ul>
            <li><em>“Specify Name for Table”</em>: <code class="highlighter-rouge">psms</code></li>
            <li><em>“Use first line as column names”</em> : <code class="highlighter-rouge">No</code></li>
            <li><em>“Specify Column Names (comma-separated list)”</em>:<code class="highlighter-rouge">id,Proteins,Sequence</code></li>
            <li><em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">Yes</code></li>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Table Index</strong>
              <ul>
                <li><em>“Table Index”</em>: <code class="highlighter-rouge">No</code></li>
                <li><em>“Index on Columns”</em>: <code class="highlighter-rouge">id</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong> (c):</p>

      <ul>
        <li>Section <strong>Filter Dataset Input</strong>
          <ul>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
              <ul>
                <li><em>“Filter by”</em>:  <code class="highlighter-rouge">skip leading lines</code></li>
                <li><em>“Skip lines”</em>: <code class="highlighter-rouge">1</code></li>
              </ul>
            </li>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
              <ul>
                <li><em>“Filter by”</em>:  <code class="highlighter-rouge">select columns</code></li>
                <li><em>“Enter column numbers to keep”</em>: <code class="highlighter-rouge">1,2</code></li>
              </ul>
            </li>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
              <ul>
                <li><em>“Filter by”</em>:  <code class="highlighter-rouge">normalize list columns,replicate rows for each item in the list</code></li>
                <li><em>“Enter column numbers to normalize”</em>: <code class="highlighter-rouge">2</code></li>
                <li><em>“List item delimiter in column”</em>: <code class="highlighter-rouge">,</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Section <strong>Table Options</strong>:
          <ul>
            <li><em>“Specify Name for Table”</em>: <code class="highlighter-rouge">prots</code></li>
            <li><em>“Use first line as column names”</em> : <code class="highlighter-rouge">No</code></li>
            <li><em>“Specify Column Names (comma-separated list)”</em>:<code class="highlighter-rouge">id,prot</code></li>
            <li><em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">Yes</code></li>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Table Index</strong>:
              <ul>
                <li><em>“Table Index”</em>: <code class="highlighter-rouge">No</code></li>
                <li><em>“Index on Columns”</em>: <code class="highlighter-rouge">prot, id</code></li>
              </ul>
            </li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-comment-2"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment:</h3>

            <p>By default, table columns will be named: c1,c2,c3,…,cn (column names for a table must be unique).
You can override the default names by entering a comma separated list of names, e.g. <code class="highlighter-rouge">,name1,,,name2</code>
would rename the second and fifth columns.
Check your input file to find the settings which best fits your needs.</p>

          </blockquote>
        </li>
        <li>
          <p><em>“Save the sqlite database in your history”</em>: <code class="highlighter-rouge">No</code></p>

          <blockquote class="comment">
            <h3 id="comment-querying-an-sqlite-database"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Querying an SQLite Database</h3>
            <p><strong>Query Tabular</strong> can also use an existing SQLite database.
Activating <code class="highlighter-rouge">Save the sqlite database in your history</code>
will store the generated database in the history, allowing to reuse it directly.</p>
          </blockquote>
        </li>
        <li><em>“SQL Query to generate tabular output”</em>:
          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT psms.*
FROM psms
WHERE psms.id NOT IN
(SELECT distinct prots.id
FROM prots JOIN uniprot ON prots.prot = uniprot.prot)
ORDER BY psms.id
</code></pre></div>          </div>
        </li>
        <li><em>“include query result column headers”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong> and inspect the query results file after it turned green.</li>
  </ul>

</blockquote>

<p>The output from this step is that the resultant peptides would be those which doesn’t belong in the Uniprot or cRAP database. The query tabular tool is used again to create a tabular output containing peptides ready for Blast P analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-query-tabular-1"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Query Tabular</h3>

  <p><strong>Query Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: with the following parameters:</p>

  <ul>
    <li>Section <strong>Filter Dataset Input</strong>
      <ul>
        <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Filter Tabular Input Lines”</em>
          <ul>
            <li><em>“Filter by”</em>:  <code class="highlighter-rouge">skip leading lines</code></li>
            <li><em>“Skip lines”</em>: <code class="highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Section <strong>Table Options</strong>:
      <ul>
        <li><em>“Specify Name for Table”</em>: <code class="highlighter-rouge">psm</code></li>
        <li><em>“Use first line as column names”</em> : <code class="highlighter-rouge">No</code></li>
        <li><em>“Specify Column Names (comma-separated list)”</em>:<code class="highlighter-rouge">id,Proteins,Sequence</code></li>
        <li><em>“Only load the columns you have named into database”</em>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li><em>“SQL Query to generate tabular output”</em>:
      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT Sequence || ' PSM=' || group_concat(id,',') || ' length='
|| length(Sequence) as "ID",Sequence
FROM  psm
WHERE length(Sequence) &gt;6
AND length(Sequence) &lt;= 30
GROUP BY Sequence
ORDER BY length(Sequence),Sequence
</code></pre></div>      </div>
    </li>
    <li>
      <p><em>“include query result column headers”</em>: <code class="highlighter-rouge">Yes</code></p>
    </li>
    <li>Click <strong>Execute</strong> and inspect the query results file after it turned green.</li>
  </ul>

  <p><img src="../../images/QT_output.png" alt="QT" /></p>

</blockquote>

<h1 id="blastp-basic-local-alignment-search-tool--proteins">BLASTP (Basic Local Alignment Search Tool- proteins)</h1>

<p><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> is a web based tool used to compare biological sequences. BlastP, matches protein sequences against a protein database. More specifically, it looks at the amino acid sequence of proteins and can detect and evaluate the amount of differences between say, an experimentally derived sequence and all known amino acid sequences from a database. It can then find the most similar sequences and allow for identification of known proteins or for identification of potential peptides associated with novel proteoforms.</p>

<p>BlastP search is carried out with the PSM report (output from PeptideShaker). Before, BlastP analysis the “Peptides_for_Blast-P_analysis” is first converted from Tabular format to FASTA file format which can be easily read by the BlastP algorithm. This is done with the help of “Tabular to FASTA” conversion tool.
The short BlastP uses parameters for short peptide sequences (8-30 aas). Please use the rerun option to look at the parameters used.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-tabular-to-fasta"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Tabular to FASTA</h3>
  <ol>
    <li><strong>Tabular to FASTA</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: with the following parameters:
      <ul>
        <li><em>“Title column”</em>: <code class="highlighter-rouge">1</code></li>
        <li><em>“Sequence Column”</em>:<code class="highlighter-rouge">2</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output FASTA file is going to be subjected to BLAST-P analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-ncbi-blast-blastp"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: NCBI BLAST+ blastp</h3>

  <ol>
    <li><strong>NCBI BLAST+ blastp</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>BLASTP</strong> with:
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Protein query sequence(s)”</em>: <code class="highlighter-rouge">Data input 'query' (fasta)</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Subject database/sequences”</em>: <code class="highlighter-rouge">Locally installed BLAST database</code>
          <ul>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Protein Blast database”</em>: Select <code class="highlighter-rouge">nr_mouse_current</code></li>
          </ul>
        </li>
        <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Type of BLAST”</em>: <code class="highlighter-rouge">blastp-short - BLASTP optimized for queries shorter than 30 residues</code></li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Set expectation value cutoff”</em>:<code class="highlighter-rouge">200000.0</code></li>
        <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Output format”</em>: <code class="highlighter-rouge">Tabular(extended 25 columns)</code></li>
        <li><em>“Advanced Options”</em>: <code class="highlighter-rouge">Show Advanced Options</code>
          <ul>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Filter out low complexity regions (with SEG)”</em>: <code class="highlighter-rouge">No</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Scoring matrix and gap costs”</em>: <code class="highlighter-rouge">PAM30</code>
              <ul>
                <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Gap Costs”</em>: <code class="highlighter-rouge">Existence: 9 Extension: 1</code></li>
              </ul>
            </li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Maximum hits to show”</em>: <code class="highlighter-rouge">1</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Maximum number of HSPs (alignments) to keep for any single query-subject pair”</em>:<code class="highlighter-rouge">1</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Word size for wordfinder algorithm”</em>: <code class="highlighter-rouge">2</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Multiple hits window size”</em>: <code class="highlighter-rouge">40</code></li>
            <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Minimum score to add a word to the BLAST lookup table”</em>: <code class="highlighter-rouge">11</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Composition-based statistics”</em>: <code class="highlighter-rouge">0 (No composition)</code></li>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Should the query and subject defline(s) be parsed?”</em>: <code class="highlighter-rouge">No</code></li>
            <li><i class="fa fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Restrict search of database to a given set of ID’s”</em>:<code class="highlighter-rouge">No restriction</code></li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-comment-3"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment:</h3>

            <p>This feature provides a means to exclude ID’s from a BLAST database search.
The expectation values in the BLAST results are based upon the sequences actually
searched, and not on the underlying database. Note this cannot be used when
comparing against a FASTA file.</p>

          </blockquote>
        </li>
        <li><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Minimum query coverage per hsp (percentage, 0 to 100)”</em>: <code class="highlighter-rouge">0</code></li>
        <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Compute locally optimal Smith-Waterman alignments”</em>:<code class="highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Once BlastP search is performed, it provides with a tabular output containing peptides corresponding to novel proteoforms termed as “<strong>Novel peptides</strong>”. Now this output is further processed by comparing the novel peptide output with the PSM report for selecting only distinct peptides which pass these criteria. This could be achieved by proceeding to the novel peptide analysis tutorial.</p>




                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>With SearchGUI and PeptideShaker you can gain access to multiple search engines</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                    

                    

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
        <blockquote class="agenda follow-up">
            <strong class="follow-up"><i class="fa fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
            <ul>
                
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Proteogenomics 3: Novel peptide analysis:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
        </blockquote>
        

        <hr>
        <br>

        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Proteogenomics 2: Database Search (Proteomics)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=Proteogenomics 2: Database Search (Proteomics)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
